<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 A tutorial for using TreeSAPP | Gene-centric mapping of microbial community structure and function</title>
  <meta name="description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="2 A tutorial for using TreeSAPP | Gene-centric mapping of microbial community structure and function" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 A tutorial for using TreeSAPP | Gene-centric mapping of microbial community structure and function" />
  
  <meta name="twitter:description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  

<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="exploring-the-microcosmos.html"/>
<link rel="next" href="acquiring-golden-data.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html"><i class="fa fa-check"></i><b>1</b> Exploring the microcosmos</a><ul>
<li class="chapter" data-level="1.1" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#background"><i class="fa fa-check"></i><b>1.1</b> Background</a></li>
<li class="chapter" data-level="1.2" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#using-multi-omic-data-to-reconstruct-distributed-networks-of-metabolite-exchange"><i class="fa fa-check"></i><b>1.2</b> Using multi-omic data to reconstruct distributed networks of metabolite exchange</a></li>
<li class="chapter" data-level="1.3" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#treesapp"><i class="fa fa-check"></i><b>1.3</b> TreeSAPP</a></li>
</ul></li>
<li class="part"><span><b>II Tutorials</b></span></li>
<li class="chapter" data-level="2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html"><i class="fa fa-check"></i><b>2</b> A tutorial for using TreeSAPP</a><ul>
<li class="chapter" data-level="2.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#introduction-and-goals"><i class="fa fa-check"></i><b>2.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="2.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#a-typical-treesapp-workflow"><i class="fa fa-check"></i><b>2.2</b> A typical TreeSAPP workflow</a></li>
<li class="chapter" data-level="2.3" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#genes-for-creating-reference-packages"><i class="fa fa-check"></i><b>2.3</b> Genes for creating reference packages</a><ul>
<li class="chapter" data-level="2.3.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#xmoa"><i class="fa fa-check"></i><b>2.3.1</b> XmoA</a></li>
<li class="chapter" data-level="2.3.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#group-assigned-genes"><i class="fa fa-check"></i><b>2.3.2</b> Group-assigned genes</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#tools"><i class="fa fa-check"></i><b>2.4</b> Tools</a><ul>
<li class="chapter" data-level="2.4.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#shell"><i class="fa fa-check"></i><b>2.4.1</b> Shell</a></li>
<li class="chapter" data-level="2.4.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#treesapp-1"><i class="fa fa-check"></i><b>2.4.2</b> TreeSAPP</a></li>
<li class="chapter" data-level="2.4.3" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#itol"><i class="fa fa-check"></i><b>2.4.3</b> iTOL</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#data"><i class="fa fa-check"></i><b>2.5</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html"><i class="fa fa-check"></i><b>3</b> Acquiring golden data</a><ul>
<li class="chapter" data-level="3.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#the-golden-standard"><i class="fa fa-check"></i><b>3.1</b> The golden standard</a></li>
<li class="chapter" data-level="3.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#what-databases-can-i-retrieve-sequences-from"><i class="fa fa-check"></i><b>3.2</b> What databases can I retrieve sequences from?</a></li>
<li class="chapter" data-level="3.3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#xmoa-example"><i class="fa fa-check"></i><b>3.3</b> XmoA example</a><ul>
<li class="chapter" data-level="3.3.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#fungene"><i class="fa fa-check"></i><b>3.3.1</b> FunGene</a></li>
<li class="chapter" data-level="3.3.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#eggnog"><i class="fa fa-check"></i><b>3.3.2</b> EggNOG</a></li>
<li class="chapter" data-level="3.3.3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#copying-data-onto-the-server"><i class="fa fa-check"></i><b>3.3.3</b> Copying data onto the server</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#alternative-sources"><i class="fa fa-check"></i><b>3.4</b> Alternative sources</a></li>
<li class="chapter" data-level="3.5" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#data-access-hazards"><i class="fa fa-check"></i><b>3.5</b> Data access hazards</a></li>
<li class="chapter" data-level="3.6" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#validation"><i class="fa fa-check"></i><b>3.6</b> Validation</a><ul>
<li class="chapter" data-level="3.6.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#basic"><i class="fa fa-check"></i><b>3.6.1</b> Basic</a></li>
<li class="chapter" data-level="3.6.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#advanced"><i class="fa fa-check"></i><b>3.6.2</b> Advanced</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="treesapp-create.html"><a href="treesapp-create.html"><i class="fa fa-check"></i><b>4</b> Creating a reference package for TreeSAPP</a><ul>
<li class="chapter" data-level="4.1" data-path="treesapp-create.html"><a href="treesapp-create.html#making-the-reference-package"><i class="fa fa-check"></i><b>4.1</b> Making the reference package</a></li>
<li class="chapter" data-level="4.2" data-path="treesapp-create.html"><a href="treesapp-create.html#testing-the-purity-of-the-reference-package"><i class="fa fa-check"></i><b>4.2</b> Testing the purity of the reference package</a></li>
<li class="chapter" data-level="4.3" data-path="treesapp-create.html"><a href="treesapp-create.html#viewing-the-reference-package-tree-with-itol"><i class="fa fa-check"></i><b>4.3</b> Viewing the reference package tree with iTOL</a><ul>
<li class="chapter" data-level="4.3.1" data-path="treesapp-create.html"><a href="treesapp-create.html#classifying-amino-acid-sequences"><i class="fa fa-check"></i><b>4.3.1</b> Classifying amino acid sequences</a></li>
<li class="chapter" data-level="4.3.2" data-path="treesapp-create.html"><a href="treesapp-create.html#classifying-orfs-predicted-from-genomes"><i class="fa fa-check"></i><b>4.3.2</b> Classifying ORFs predicted from genomes</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="treesapp-create.html"><a href="treesapp-create.html#update-the-reference-package-with-new-sequences"><i class="fa fa-check"></i><b>4.4</b> Update the reference package with new sequences</a><ul>
<li class="chapter" data-level="4.4.1" data-path="treesapp-create.html"><a href="treesapp-create.html#updating-with-publicly-available-sequences"><i class="fa fa-check"></i><b>4.4.1</b> Updating with publicly available sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="treesapp-create.html"><a href="treesapp-create.html#annotate-clades"><i class="fa fa-check"></i><b>4.5</b> Annotate clades</a><ul>
<li class="chapter" data-level="4.5.1" data-path="treesapp-create.html"><a href="treesapp-create.html#updating-with-mag-sequences"><i class="fa fa-check"></i><b>4.5.1</b> Updating with MAG sequences</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="treesapp-create.html"><a href="treesapp-create.html#annotate-clades-1"><i class="fa fa-check"></i><b>4.6</b> Annotate clades</a></li>
<li class="chapter" data-level="4.7" data-path="treesapp-create.html"><a href="treesapp-create.html#reclassify-and-layer-annotations-onto-classification-table"><i class="fa fa-check"></i><b>4.7</b> Reclassify and layer annotations onto classification table</a></li>
<li class="chapter" data-level="4.8" data-path="treesapp-create.html"><a href="treesapp-create.html#use-your-groups-reference-package-to-classify-unkown-sequences-then-update-your-package"><i class="fa fa-check"></i><b>4.8</b> Use your group’s reference package to classify unkown sequences then update your package</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html"><i class="fa fa-check"></i><b>5</b> Calculating relative abundance of meta’omic data</a><ul>
<li class="chapter" data-level="5.1" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#introduction-and-goals-1"><i class="fa fa-check"></i><b>5.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="5.2" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#workflow-description"><i class="fa fa-check"></i><b>5.2</b> Workflow description</a></li>
<li class="chapter" data-level="5.3" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#usage"><i class="fa fa-check"></i><b>5.3</b> Usage</a></li>
<li class="chapter" data-level="5.4" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#outputs"><i class="fa fa-check"></i><b>5.4</b> Outputs</a></li>
<li class="chapter" data-level="5.5" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#tutorial-steps"><i class="fa fa-check"></i><b>5.5</b> Tutorial steps</a></li>
<li class="chapter" data-level="5.6" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#use-your-groups-reference-package-to-calculate-abundances"><i class="fa fa-check"></i><b>5.6</b> Use your group’s reference package to calculate abundances</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html"><i class="fa fa-check"></i><b>A</b> The Saanich Inlet data set</a><ul>
<li class="chapter" data-level="A.1" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#description"><i class="fa fa-check"></i><b>A.1</b> Description</a></li>
<li class="chapter" data-level="A.2" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#data-on-the-server"><i class="fa fa-check"></i><b>A.2</b> Data on the server</a></li>
<li class="chapter" data-level="A.3" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#description-of-input-data"><i class="fa fa-check"></i><b>A.3</b> Description of input data</a></li>
<li class="chapter" data-level="A.4" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#methods"><i class="fa fa-check"></i><b>A.4</b> Methods</a><ul>
<li class="chapter" data-level="A.4.1" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#quality-filtering-of-reads"><i class="fa fa-check"></i><b>A.4.1</b> Quality filtering of reads</a></li>
<li class="chapter" data-level="A.4.2" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#assembly"><i class="fa fa-check"></i><b>A.4.2</b> Assembly</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="shell-cheat-sheet.html"><a href="shell-cheat-sheet.html"><i class="fa fa-check"></i><b>B</b> Shell cheat sheet</a></li>
<li class="chapter" data-level="C" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>C</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Gene-centric mapping of microbial community structure and function</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="a-tutorial-for-using-treesapp" class="section level1">
<h1><span class="header-section-number">2</span> A tutorial for using TreeSAPP</h1>
<div id="introduction-and-goals" class="section level2">
<h2><span class="header-section-number">2.1</span> Introduction and goals</h2>
<p>In this series of tutorials, users will analyze a gene family lacking an existing TreeSAPP reference package (refpkg). You will work through a typical workflow of TreeSAPP both with an example gene (XmoA) to familiarize yourself with the tools, and then you will repeat the steps with a gene assigned to your group for which no reference package exists. You will document your efforts for the new reference package in Problem Set 5.</p>
</div>
<div id="a-typical-treesapp-workflow" class="section level2">
<h2><span class="header-section-number">2.2</span> A typical TreeSAPP workflow</h2>
<ul>
<li><a href="acquiring-golden-data.html#acquiring-golden-data">Acquiring golden data</a>.</li>
<li><a href="treesapp-create.html#treesapp-create">Creating a Reference Package For TreeSAPP</a>.</li>
<li>[Classifying unknown sequences with TreeSAPP and updating a reference package].</li>
<li><a href="calculating-relative-abundance-of-metaomic-data.html#calculating-relative-abundance-of-metaomic-data">Calculating relative abundance of meta’omic data</a>.</li>
</ul>
</div>
<div id="genes-for-creating-reference-packages" class="section level2">
<h2><span class="header-section-number">2.3</span> Genes for creating reference packages</h2>
<div id="xmoa" class="section level3">
<h3><span class="header-section-number">2.3.1</span> XmoA</h3>
<p>The protein family we will be focusing on is that of the copper-containing membrane-bound monooxygenases <span class="citation">(<a href="references.html#ref-Khadka.2018" role="doc-biblioref">5</a>)</span>. This family contains particulate methane monooxygenase (pMMO) and ammonia monooxygenase (AMO) and well be building a reference package for the alpha subunits of these enzymes called XmoA. All students will work through this example individually.</p>
</div>
<div id="group-assigned-genes" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Group-assigned genes</h3>
<p>You will create a reference package and perform the other steps only for the gene that has been assigned to you.</p>
<div style="border: 1px solid #ddd; padding: 5px; overflow-x: scroll; width:100%; ">
<table class=" lightable-paper" style='font-family: "Arial Narrow", arial, helvetica, sans-serif; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
RefPkg
</th>
<th style="text-align:left;">
refpkg_code
</th>
<th style="text-align:left;">
Full Protein Name
</th>
<th style="text-align:left;">
Pathway
</th>
<th style="text-align:left;">
Cycle
</th>
<th style="text-align:left;">
TIGRFAM
</th>
<th style="text-align:left;">
PFam
</th>
<th style="text-align:left;">
EggNOG
</th>
<th style="text-align:left;">
Other Database
</th>
<th style="text-align:left;">
EC number
</th>
<th style="text-align:left;">
Reading
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
cbbS
</td>
<td style="text-align:left;">
C0003
</td>
<td style="text-align:left;">
ribulose bisphosphate carboxylase, type III (Ribulose bisphosphate carboxylase small chain)
</td>
<td style="text-align:left;">
Carbon fixation
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
TIGR03326
</td>
<td style="text-align:left;">
PF00101
</td>
<td style="text-align:left;">
COG4451
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
4.1.1.39
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
ACoA
</td>
<td style="text-align:left;">
C0004
</td>
<td style="text-align:left;">
ACoA_crb (acetylCoA carboxylase)
</td>
<td style="text-align:left;">
Carbon fixation
</td>
<td style="text-align:left;">
C
</td>
<td style="text-align:left;">
TIGR03182
</td>
<td style="text-align:left;">
PF00676
</td>
<td style="text-align:left;">
COG1071
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.2.4.1
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Hzs
</td>
<td style="text-align:left;">
N0701
</td>
<td style="text-align:left;">
hydrazine synthase
</td>
<td style="text-align:left;">
Annamox
</td>
<td style="text-align:left;">
N
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
PF18582
</td>
<td style="text-align:left;">
COG3391
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.7.2.7
</td>
<td style="text-align:left;">
<a href="https://www.nature.com/articles/nature10453" class="uri">https://www.nature.com/articles/nature10453</a> <a href="https://www.frontiersin.org/articles/10.3389/fmars.2019.00027/full" class="uri">https://www.frontiersin.org/articles/10.3389/fmars.2019.00027/full</a>
</td>
</tr>
<tr>
<td style="text-align:left;">
NirB
</td>
<td style="text-align:left;">
N0201
</td>
<td style="text-align:left;">
nitrite reductase (NADH) large subunit
</td>
<td style="text-align:left;">
Dissimilitory nitrate reduction (DNRA)
</td>
<td style="text-align:left;">
N
</td>
<td style="text-align:left;">
TIGR02374
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
COG1251
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.7.1.15
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NirD
</td>
<td style="text-align:left;">
N0202
</td>
<td style="text-align:left;">
nitrite reductase (NADH) small subunit
</td>
<td style="text-align:left;">
Dissimilitory nitrate reduction (DNRA)
</td>
<td style="text-align:left;">
N
</td>
<td style="text-align:left;">
TIGR02378
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
COG2146
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.7.1.15
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NrfA
</td>
<td style="text-align:left;">
N0203
</td>
<td style="text-align:left;">
nitrite reductase (cytochrome c-552)
</td>
<td style="text-align:left;">
Dissimilitory nitrate reduction (DNRA)
</td>
<td style="text-align:left;">
N
</td>
<td style="text-align:left;">
TIGR03152
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
COG3303
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.7.2.2
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
NrfH
</td>
<td style="text-align:left;">
N0204
</td>
<td style="text-align:left;">
cytochrome c nitrite reductase, small subunit
</td>
<td style="text-align:left;">
Dissimilitory nitrate reduction (DNRA)
</td>
<td style="text-align:left;">
N
</td>
<td style="text-align:left;">
TIGR03153
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
COG3005
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.7.2.2
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
APS_red
</td>
<td style="text-align:left;">
S0101
</td>
<td style="text-align:left;">
adenylylsulfate reductase, thioredoxin dependent
</td>
<td style="text-align:left;">
Sulfur metabolism
</td>
<td style="text-align:left;">
S
</td>
<td style="text-align:left;">
TIGR02055
</td>
<td style="text-align:left;">
PF01507
</td>
<td style="text-align:left;">
COG1404
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
1.8.99.2
</td>
<td style="text-align:left;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
BacRho
</td>
<td style="text-align:left;">
P0102
</td>
<td style="text-align:left;">
Bacteriorhodopsin
</td>
<td style="text-align:left;">
Energy production and conversion
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
PF01036
</td>
<td style="text-align:left;">
COG5524
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
NA
</td>
<td style="text-align:left;">
<a href="https://science.sciencemag.org/content/289/5486/1902/tab-pdf" class="uri">https://science.sciencemag.org/content/289/5486/1902/tab-pdf</a> <a href="https://www.nature.com/articles/s41467-018-07840-4" class="uri">https://www.nature.com/articles/s41467-018-07840-4</a>
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="tools" class="section level2">
<h2><span class="header-section-number">2.4</span> Tools</h2>
<div id="shell" class="section level3">
<h3><span class="header-section-number">2.4.1</span> Shell</h3>
<p>Please use this short <a href="shell-cheat-sheet.html#shell-cheat-sheet">Shell cheat sheet</a> for commonly used commands and review previous tutorials on Canvas.</p>
</div>
<div id="treesapp-1" class="section level3">
<h3><span class="header-section-number">2.4.2</span> TreeSAPP</h3>
<p>Tree-based Sensitive and Accurate Phylogenetic Profiler (TreeSAPP) <span class="citation">(<a href="references.html#ref-Morgan-Lang.2020" role="doc-biblioref">3</a>)</span> can be found on <a href="https://github.com/hallamlab/TreeSAPP">GitHub</a> including an excellent <a href="https://github.com/hallamlab/TreeSAPP/wiki">wiki</a> with additional information on each of the <code>treesapp</code> subcommands.</p>
</div>
<div id="itol" class="section level3">
<h3><span class="header-section-number">2.4.3</span> iTOL</h3>
<p>Interactive Tree Of Life (<a href="https://itol.embl.de/">iTOL</a>) <span class="citation">(<a href="references.html#ref-Letunic.2019" role="doc-biblioref">4</a>)</span> is a browser-based tool that allows you to visualize data generated in TreeSAPP as a phylogenetic tree with additional annotations.</p>
</div>
</div>
<div id="data" class="section level2">
<h2><span class="header-section-number">2.5</span> Data</h2>
<p>Other than <a href="the-saanich-inlet-data-set.html#the-saanich-inlet-data-set">The Saanich Inlet data set</a> already located on the server, you will download data from different databases:</p>
<ul>
<li><a href="http://fungene.cme.msu.edu/">FunGene</a>, the functional gene pipeline and repository.</li>
<li><a href="http://eggnog5.embl.de/#/app/home">EggNOG</a></li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="exploring-the-microcosmos.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="acquiring-golden-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
