<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>3 Acquiring reference sequence data | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="3.1 Introduction and goals Reference data in the context of biological sequence analysis are those that come from well-curated open-source databases. These data have been used to create all...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="3 Acquiring reference sequence data | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="3.1 Introduction and goals Reference data in the context of biological sequence analysis are those that come from well-curated open-source databases. These data have been used to create all...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="3 Acquiring reference sequence data | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="3.1 Introduction and goals Reference data in the context of biological sequence analysis are those that come from well-curated open-source databases. These data have been used to create all...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="active" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="references.html"><span class="header-section-number">D</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="acquiring-reference-sequence-data" class="section level1">
<h1>
<span class="header-section-number">3</span> Acquiring reference sequence data<a class="anchor" aria-label="anchor" href="#acquiring-reference-sequence-data"><i class="fas fa-link"></i></a>
</h1>
<div id="introduction-and-goals-1" class="section level2">
<h2>
<span class="header-section-number">3.1</span> Introduction and goals<a class="anchor" aria-label="anchor" href="#introduction-and-goals-1"><i class="fas fa-link"></i></a>
</h2>
<p>Reference data in the context of biological sequence analysis are those that come from well-curated open-source databases. These data have been used to create all TreeSAPP reference packages to date, and they should be used to create all future reference packages. But what does “well-curated” mean, and how can you identify data that would meet this standard?</p>
<p>The goals of this tutorial are:</p>
<ol style="list-style-type: decimal">
<li>Develop a sense of biological sequence database curation quality</li>
<li>Learn how to subset and download sequences from the EggNOG and UniProt databases</li>
</ol>
</div>
<div id="the-golden-standard" class="section level2">
<h2>
<span class="header-section-number">3.2</span> The golden standard<a class="anchor" aria-label="anchor" href="#the-golden-standard"><i class="fas fa-link"></i></a>
</h2>
<p>The field of genomics is blessed with an abundance of reference data. In fact, it is not uncommon for biologists and bioinformaticians alike to gripe over the increased run-time of computational analyses caused by steps that include comparisons to such large reference databases. Not all of these data are, however, of the same quality. The vast majority have been automatically curated using algorithms with known shortcomings, and only a small portion have been manually curated with verified functional characterization. These latter data are the ones we consider well-curated or golden. More details on choosing databases for retrieving sequences can be found on the <a href="https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#step-1-reference-sequences">TreeSAPP wiki</a>.</p>
</div>
<div id="recommended-biological-sequence-databases" class="section level2">
<h2>
<span class="header-section-number">3.3</span> Recommended biological sequence databases<a class="anchor" aria-label="anchor" href="#recommended-biological-sequence-databases"><i class="fas fa-link"></i></a>
</h2>
<p>One of the most comprehensive and reputable biological sequence databases is SwissProt. The protein sequences curated within SwissProt can be accessed and downloaded through the <a href="https://www.uniprot.org/">UniProt Knowledgebase</a> by selecting “Swiss-Prot” under the “UniProtKB” panel on the left of the landing page. You can submit queries for a protein family through their search bar and add filters to the query using their <a href="https://www.uniprot.org/help/advanced_search">advanced search</a> options. For example, you can search for reviewed RuBisCO sequences by entering the query ‘rubisco taxonomy:“Bacteria [2]” AND reviewed:yes’.</p>
<p><a href="http://eggnog5.embl.de/#/app/home">EggNOG</a> is another source of curated orthologous groups. The breadth of families available here is astounding and more often than not they do have the sequences you’re looking for. There are a couple of benefits to accessing data from EggNOG. The first is they use a common identifier format. Using the correct identifier for a protein family, a single search will yield all available amino acid sequences belonging to that protein family or COG. What I’ve found handy is first looking for the Clusters of Orthologous Gene (COG) identifier on NCBI’s <a href="https://www.ncbi.nlm.nih.gov/Structure/cdd/cdd.shtml">Conserved Domains</a> database. Then I’d use that identifier to search for the sequences I’m interested in on EggNOG. The second benefit is the quality of their automated annotation pipeline. With every version of EggNOG, the entire set of protein sequences is clustered and the orthologous groups are defined <em>de novo</em>. This prevents incorrect legacy annotations from propagating to new sequences. Unfortunately, the breadth of taxa covered by EggNOG is limited to <em>just</em> 5,090 organisms. In many cases, this is good enough to build a “seed” reference package, especially if you’re building a reference package of a well-conserved—or housekeeping—gene. Yet, there are plenty of scenarios where the number of sequences downloaded is just too few for your objective, and a more comprehensive reference package is required—more on that below.</p>
<p>In addition to these general and all-encompassing databases, there are smaller curated databases that are independently maintained. These tend to be gene- or function-specific and, as long as they’re still maintained, highly reliable. Perhaps the most notable of these is the <a href="http://www.cazy.org/">Carbohydrate Active enZYmes (CAZy)</a> database. The maintainers of CAZy constantly screen new sequenced from GenBank and sort any hits into different families based on their putative biochemical characterization. <a href="http://fungene.cme.msu.edu/">FunGene</a> has also been used extensively to build reference packages but due to system instability and variability in curation quality this database is used less often. Still, the focus on biogeochemical cycles and convenient organization, export process, and sequence header format make this database a good back-up.</p>
</div>
<div id="xmoa-example" class="section level2">
<h2>
<span class="header-section-number">3.4</span> XmoA example<a class="anchor" aria-label="anchor" href="#xmoa-example"><i class="fas fa-link"></i></a>
</h2>
<p>To show you how to access these data, we will download the alpha subunit sequences of the enzymes particulate methane monooxygenase (PmoA) and ammonia monooxygenase (AmoA) that are used in the <a href="treesapp-create.html#treesapp-create"><code>treesapp create</code> tutorial</a> from the EggNOG and UniProt databases. Since XmoA is not a recognized identifier, but a convenient label that reflects the operon structure (<em>xmoCAB</em>) of homologous oxidoreductase enzymes, we will need to search in databases multiple times.</p>
<div id="eggnog" class="section level3">
<h3>
<span class="header-section-number">3.4.1</span> EggNOG<a class="anchor" aria-label="anchor" href="#eggnog"><i class="fas fa-link"></i></a>
</h3>
<p>Go to the EggNOG website at <a href="http://eggnog5.embl.de/#/app/home" class="uri">http://eggnog5.embl.de/#/app/home</a> and search for either “ENOG5028JPK” or “arCOG08676” in the search bar at the top of the page. These EggNOG ortholog identifiers correspond to PmoA/AmoA from Bacteria and AmoA from Archaea, respectively. In the image below, we’ve searched for “ENOG5028JPK”.</p>
<div class="inline-figure"><img src="images/EggNOG_xmoa_1.png" style="width:100.0%"></div>
<p>Click “Download” at the bottom-left of the panel and select “All <em>XX</em> sequences (FASTA)”, where <em>XX</em> is the number of sequences belonging to the orthologous group (OG). This will open a new window with all sequences from that OG in FASTA format. Download this file to a file on your computer following the format &lt;OG name&gt;_EggNOGv5.faa where &lt;OG name&gt; is the EggNOG ortholog identifier.</p>
<p>Repeat the above steps with the other EggNOG identifier.</p>
</div>
<div id="uniprot" class="section level3">
<h3>
<span class="header-section-number">3.4.2</span> UniProt<a class="anchor" aria-label="anchor" href="#uniprot"><i class="fas fa-link"></i></a>
</h3>
<p>The UniProt database architecture is built upon standard biological attributes, in contrast to the <em>de novo</em> clustering and classification scheme of EggNOG. Protein sequences are therefore searchable using multiple keys covering a range of taxonomic and functional specificities. The most specific is the unique identifier of a single protein sequence (e.g. Q04507 is an AmoA sequence in Nitrosomonas europaea). More broad searchable attributes include protein names, gene names, organism, <a href="https://en.wikipedia.org/wiki/Enzyme_Commission_number">Enzyme Commission (EC)</a> numbers, and identifiers of other databases such as PFam and TIGRFAM.</p>
<p>Here, we will use the gene names AmoA and PmoA to find reference sequences in UniProt. Begin by navigating to <a href="https://www.uniprot.org/">https://www.uniprot.org/</a> and typing “Particulate methane monooxygenase” (including the double quotation marks) into the search bar at the top of the page.</p>
<div id="refining-the-search" class="section level4">
<h4>
<span class="header-section-number">3.4.2.1</span> Refining the search<a class="anchor" aria-label="anchor" href="#refining-the-search"><i class="fas fa-link"></i></a>
</h4>
<p>12,231 sequences were returned by the initial query. In the following figure, we can see many entries were unrelated to pmoA (e.g. ftsH, ffsA, carB).</p>
<p><img src="images/UniProt_pmo.png" style="width:100.0%">
To remove these off-target search results, we will need to expand our search terms and for this we will turn to the “Advanced” drop-down menu on the right side of the search bar. Add “pmoa” for the <em>Gene name [GN]</em> and remove all sequences with unresolved lineages by specifying “uncultured bacterium” for <em>Taxonomy [OC]</em> with the <em>NOT</em> logical operator. The latter filter was applied since these poorly annotated sequences are typically derived from short environmental samples and are not taxonomically informative. The advanced search fields should look identical to the figure below.</p>
<div class="inline-figure"><img src="images/pmoA_advanced_search.png" style="width:100.0%"></div>
<p>This more specific search should yield &gt;3800 entries with the vast majority being predicted pmoA sequences.</p>
</div>
<div id="download-sequences" class="section level4">
<h4>
<span class="header-section-number">3.4.2.2</span> Download sequences<a class="anchor" aria-label="anchor" href="#download-sequences"><i class="fas fa-link"></i></a>
</h4>
<p>Downloading sequences in UniProt is quick and easy. Click the <img src="images/uniprot_dl_icon.png" width="50"> button (atop the table), ensure the format is FASTA, and click “Go” to specify where the file should be saved on your computer. Change the file name to <code>UniProt_PmoA.fasta</code>.</p>
</div>
<div id="download-taxonomic-lineages" class="section level4">
<h4>
<span class="header-section-number">3.4.2.3</span> Download taxonomic lineages<a class="anchor" aria-label="anchor" href="#download-taxonomic-lineages"><i class="fas fa-link"></i></a>
</h4>
<p>UniProt sequence names are not searchable within the NCBI’s database, so software (including TreeSAPP) is unable to pair them with their corresponding taxonomic lineages like with EggNOG. Therefore, a separate table must be downloaded from UniProt and provided to TreeSAPP during reference package creation.</p>
<div class="inline-figure">To create a properly formatted table, we must first change the columns in the UniProt table of search results. Begin by clicking the <img src="images/uniprot_cols_icon.png" width="50"> icon to the right of the “Download” button.</div>
<div class="inline-figure"><img src="images/uniprot_columns.png" style="width:100.0%"></div>
<p>Once there, only check the boxes for organism, superkingdom, phylum, class, order, family, genus and species as shown above.
Click “Save” near the top-right of the page to save your selection.</p>
<p>You can export this table similar to before.
Click the <img src="images/uniprot_dl_icon.png" width="50"> button to select all sequences, change the format to “Tab-separated”, then click “Go” to save the table to your computer.
Change the file name to <code>UniProt_PmoA.tab</code>.</p>
<p>To gather all sequences in UniProt for the XmoA reference package, repeat the previous three steps with the string “Ammonia monooxygenase alpha subunit”, <em>Gene name [GN]</em> “amoA”, and the same taxonomic filter.
There should be &gt;3,000 entries returned.
Save the sequences and taxonomic lineages in <code>UniProt_AmoA.fasta</code> and <code>UniProt_AmoA.tab</code>, respectively.</p>
</div>
<div id="downloading-reviewed-sequences-from-swissprot" class="section level4">
<h4>
<span class="header-section-number">3.4.2.4</span> Downloading reviewed sequences from SwissProt<a class="anchor" aria-label="anchor" href="#downloading-reviewed-sequences-from-swissprot"><i class="fas fa-link"></i></a>
</h4>
<p>As you may have noticed in the previous steps, there was a single manually-reviewed sequence for PmoA and AmoA each.
In this case, there are too few sequences to build a seed reference package that could be scaffolded on to eventually construct phylogenetically comprehensive reference package.
Still, once there are more than ten or so phylogenetically diverse manually-reviewed sequences, the UniProt/SwissProt database should be sufficient to construct a seed reference package.
By filtering by reviewed sequenced (gold star) and following the previous two steps you can retrieve sequences that have been manually curated, indicating a high-degree of confidence in the functional annotation of those sequences.
Keep this functionality in mind for future reference packages.</p>
</div>
</div>
<div id="copying-data-onto-the-server" class="section level3">
<h3>
<span class="header-section-number">3.4.3</span> Copying data onto the server<a class="anchor" aria-label="anchor" href="#copying-data-onto-the-server"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>
<p><strong>On your computer:</strong> Create a new directory called <code>ts_tutorial_local</code> inside of your home directory by using a file browser, i.e. File Explorer (Windows 10) or Finder (macOS).
Place any reference sequences in <code>ts_tutorial_local</code> that you identified and downloaded from the databases.</p>
<p>Next, connect to your personal server using the terminal app, i.e. Windows Terminal (Windows 10) or Terminal (macOS).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="acquiring-reference-sequence-data.html#cb1-1"></a><span class="fu">ssh</span> root@<span class="op">&lt;</span>server address<span class="op">&gt;</span></span></code></pre></div>
</li>
<li>
<p><strong>On the server:</strong> Move to your directory <code>/data</code> (if it is not your working directory already) and create a new directory called <code>&lt;user&gt;/tutorial</code> where <code>&lt;user&gt;/</code> is unique within your project group.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="acquiring-reference-sequence-data.html#cb2-1"></a><span class="bu">cd</span> /data/</span>
<span id="cb2-2"><a href="acquiring-reference-sequence-data.html#cb2-2"></a><span class="fu">mkdir</span> <span class="op">&lt;</span>user<span class="op">&gt;</span></span>
<span id="cb2-3"><a href="acquiring-reference-sequence-data.html#cb2-3"></a><span class="bu">cd</span> <span class="op">&lt;</span>user<span class="op">&gt;</span></span>
<span id="cb2-4"><a href="acquiring-reference-sequence-data.html#cb2-4"></a><span class="fu">mkdir</span> tutorial</span></code></pre></div>
</li>
<li>
<p><strong>On your computer:</strong> Open a second terminal window that is connected to your local computer, then copy the reference sequences from your computer to your server.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="acquiring-reference-sequence-data.html#cb3-1"></a><span class="fu">scp</span> <span class="op">&lt;</span>path to ts_tutorial_local<span class="op">&gt;</span>/* root@<span class="op">&lt;</span>server address<span class="op">&gt;</span>:/data/<span class="op">&lt;</span>user<span class="op">&gt;</span>/tutorial/</span></code></pre></div>
</li>
</ul>
</div>
</div>
<div id="alternative-sources" class="section level2">
<h2>
<span class="header-section-number">3.5</span> Alternative sources<a class="anchor" aria-label="anchor" href="#alternative-sources"><i class="fas fa-link"></i></a>
</h2>
<p>EggNOG and UniProt are often the most useful sources, though this leaves out the majority of biological sequence databases for no reason other than simplicity. IMG, KEGG, PFam, and other similar databases are all great but their utility is blunted because they either lack an API to retrieve lineage information, requiring a separate table to link this information, or the sequence accession (how database sequences are uniquely named) do not allow for easy referencing in the NCBI taxonomic hierarchy. Details on this table’s format and how to provide it can be found under the <a href="https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP"><code>treesapp create</code> wiki page</a>.</p>
</div>
<div id="data-access-hazards" class="section level2">
<h2>
<span class="header-section-number">3.6</span> Data access hazards<a class="anchor" aria-label="anchor" href="#data-access-hazards"><i class="fas fa-link"></i></a>
</h2>
<p>Accessing the fraction of well-curated data from multiple databases is not trivial. Perhaps the most obvious reason for this are unrelated genes with shared names. For example, the gene product of mcrB in <em>E. coli</em> widely refers to 5-methylcytosine-specific restriction enzyme B, but in the Archaea, it would refer to the beta subunit of methyl-coenzyme M reductase—a completely unrelated protein. This is an inevitability in a field of science as broad as biology, where scientists that are responsible for naming genes cannot possibly be aware of all gene names in circulation. Another issue can be genes or proteins with synonyms, making the search for these data more complex. In these cases, you may need to search multiple databases with several queries in order to access all the sequences belonging to the same protein family.</p>
</div>
<div id="validation" class="section level2">
<h2>
<span class="header-section-number">3.7</span> Validation<a class="anchor" aria-label="anchor" href="#validation"><i class="fas fa-link"></i></a>
</h2>
<p>There are a number of quality control measures that can be taken to ensure the functional characteristics are as expected.</p>
<div id="basic" class="section level3">
<h3>
<span class="header-section-number">3.7.1</span> Basic<a class="anchor" aria-label="anchor" href="#basic"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li><p><strong>BLAST</strong>: The first method for validating a set of candidate reference sequences could be to use the web-based <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">Basic Local Alignment Search Tool</a> available through the NCBI. Submit a FASTA file containing your sequences to Protein BLAST (blastp) with an appropriate database. To further reduce the processing time, you can restrict the reference sequences to specific taxa or use different algorithms. One thing to note is the query size limit - blastp will accept only FASTA files containing fewer than 100,000 characters. So, it is a good idea to take a representative sample or cluster the sequences before submitting the job.</p></li>
<li><p><strong>EggNOG-mapper</strong>: If you didn’t download your query sequences from the EggNOG database, a good resource to check your sequences is <a href="http://eggnog-mapper.embl.de/">EggNOG mapper</a>. It will annotate your query sequences against the EggNOG database and serve the result in several formats.</p></li>
</ul>
</div>
<div id="advanced" class="section level3">
<h3>
<span class="header-section-number">3.7.2</span> Advanced<a class="anchor" aria-label="anchor" href="#advanced"><i class="fas fa-link"></i></a>
</h3>
<ul>
<li>
<strong>Phylogenetic inference</strong>: Try building a phylogeny from the candidate reference sequences and see whether there are any anomolously long branches between clades.</li>
<li>
<strong>hmmsearch across PFam database</strong>: Got some time to kill? Try aligning your candidate reference sequences to all profile HMMs in the PFam database! This will need to be done on the command-line and you will need to download <code>Pfam-A.hmm.gz</code> from the <a href="http://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/">PFam ftp site</a>.</li>
</ul>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></div>
<div class="next"><a href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#acquiring-reference-sequence-data"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="nav-link" href="#introduction-and-goals-1"><span class="header-section-number">3.1</span> Introduction and goals</a></li>
<li><a class="nav-link" href="#the-golden-standard"><span class="header-section-number">3.2</span> The golden standard</a></li>
<li><a class="nav-link" href="#recommended-biological-sequence-databases"><span class="header-section-number">3.3</span> Recommended biological sequence databases</a></li>
<li>
<a class="nav-link" href="#xmoa-example"><span class="header-section-number">3.4</span> XmoA example</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#eggnog"><span class="header-section-number">3.4.1</span> EggNOG</a></li>
<li><a class="nav-link" href="#uniprot"><span class="header-section-number">3.4.2</span> UniProt</a></li>
<li><a class="nav-link" href="#copying-data-onto-the-server"><span class="header-section-number">3.4.3</span> Copying data onto the server</a></li>
</ul>
</li>
<li><a class="nav-link" href="#alternative-sources"><span class="header-section-number">3.5</span> Alternative sources</a></li>
<li><a class="nav-link" href="#data-access-hazards"><span class="header-section-number">3.6</span> Data access hazards</a></li>
<li>
<a class="nav-link" href="#validation"><span class="header-section-number">3.7</span> Validation</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#basic"><span class="header-section-number">3.7.1</span> Basic</a></li>
<li><a class="nav-link" href="#advanced"><span class="header-section-number">3.7.2</span> Advanced</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/golden_data.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/golden_data.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on March 27, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
