<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>7 Analysing the TreeSAPP classifications in R | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="7.1 Introduction and goals These instructions will help you to complete the script template treesapp_analysis.R to complete the initial analysis of your TreeSAPP classifications. The following...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="7 Analysing the TreeSAPP classifications in R | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="7.1 Introduction and goals These instructions will help you to complete the script template treesapp_analysis.R to complete the initial analysis of your TreeSAPP classifications. The following...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="7 Analysing the TreeSAPP classifications in R | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="7.1 Introduction and goals These instructions will help you to complete the script template treesapp_analysis.R to complete the initial analysis of your TreeSAPP classifications. The following...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="active" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="frequently-asked-questions.html"><span class="header-section-number">D</span> Frequently Asked Questions</a></li>
<li><a class="" href="references.html"><span class="header-section-number">E</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="analysing-the-treesapp-classifications-in-r" class="section level1" number="7">
<h1>
<span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R<a class="anchor" aria-label="anchor" href="#analysing-the-treesapp-classifications-in-r"><i class="fas fa-link"></i></a>
</h1>
<div id="introduction-and-goals-5" class="section level2" number="7.1">
<h2>
<span class="header-section-number">7.1</span> Introduction and goals<a class="anchor" aria-label="anchor" href="#introduction-and-goals-5"><i class="fas fa-link"></i></a>
</h2>
<p>These instructions will help you to complete the script template <code>treesapp_analysis.R</code> to complete the initial analysis of your TreeSAPP classifications.
The following code chunks will show you how to load, aggregate, subset and transform continuous and discrete variables in R using the <code>tidyverse</code>.
You will also be provided with examples for plotting TreeSAPP classification data using the <code>ggplot2</code> library.</p>
</div>
<div id="resources-for-analysis-of-treesapp-output-in-r" class="section level2" number="7.2">
<h2>
<span class="header-section-number">7.2</span> Resources for analysis of TreeSAPP output in R<a class="anchor" aria-label="anchor" href="#resources-for-analysis-of-treesapp-output-in-r"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>The TreeSAPP output file <code>layered_classifications.tsv</code> with the metagenome and metatranscriptome abundances and gene functions you generated with <a href="#running-treesapp%20on%20a%20server">TreeSAPP</a>. If you are unable to retrieve this file, it can be downloaded from <a href="https://raw.githubusercontent.com/EDUCE-UBC/MICB425/main/data/layered_classifications.tsv">here</a>.</li>
<li>An R script template called <code>treesapp_analysis.R</code> on Canvas.</li>
<li>
<code>Saanich_TimeSeries_Chemical_DATA.csv</code> file containing geochemical measurements on Canvas.</li>
</ul>
</div>
<div id="checklist-to-write-and-run-r-script" class="section level2" number="7.3">
<h2>
<span class="header-section-number">7.3</span> Checklist to write and run R script<a class="anchor" aria-label="anchor" href="#checklist-to-write-and-run-r-script"><i class="fas fa-link"></i></a>
</h2>
<ul class="task-list">
<li>
<input type="checkbox" disabled>
Place <code>treesapp_analysis.R</code>, <code>layered_classifications.tsv</code> and <code>Saanich_TimeSeries_Chemical_DATA.csv</code> files into a single folder and create a new RStudio project in that folder on your local computer.</li>
<li>
<input type="checkbox" disabled>
Edit the <code>treesapp_analysis.R</code> script following <a href="analysing-the-treesapp-classifications-in-r.html#writing-the-r-script">the instructions below</a>.</li>
</ul>
</div>
<div id="writing-the-r-script" class="section level2" number="7.4">
<h2>
<span class="header-section-number">7.4</span> Writing the R script<a class="anchor" aria-label="anchor" href="#writing-the-r-script"><i class="fas fa-link"></i></a>
</h2>
<p>In the <code>treesapp_analysis.R</code> script you will load and subset your TreeSAPP data to the variables and marker genes of interest. You will then break the one taxonomic information column into multiple taxonomic ranks. You will also load the <code>Saanich_TimeSeries_Chemical_DATA.csv</code> file for geochemical measurements to learn more about the conditions at your assigned depth.
Finally, you will visualize the taxonomic, functional and chemical data using the <code>ggplot2</code> library.</p>
<p>To complete this, replace all instances of test between angle brackets, like <code>&lt;SOME TEXT&gt;</code>, with the code in each section.</p>
</div>
<div id="load-any-required-packages" class="section level2" number="7.5">
<h2>
<span class="header-section-number">7.5</span> Load any required packages<a class="anchor" aria-label="anchor" href="#load-any-required-packages"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span></code></pre></div>
</div>
<div id="load-metagenomic-and-metatranscriptomic-treesapp-data" class="section level2" number="7.6">
<h2>
<span class="header-section-number">7.6</span> Load metagenomic and metatranscriptomic TreeSAPP data<a class="anchor" aria-label="anchor" href="#load-metagenomic-and-metatranscriptomic-treesapp-data"><i class="fas fa-link"></i></a>
</h2>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span>file<span class="op">=</span><span class="st">"layered_classifications.tsv"</span>,
                     header<span class="op">=</span><span class="cn">TRUE</span>,
                     sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span></code></pre></div>
<div id="classification-table" class="section level3" number="7.6.1">
<h3>
<span class="header-section-number">7.6.1</span> Classification table<a class="anchor" aria-label="anchor" href="#classification-table"><i class="fas fa-link"></i></a>
</h3>
<p>Let’s look at TreeSAPP’s classification table for these FunGene XmoA sequences. In RStudio, you are able to click on a variable name under the “Environment” tab in the top-right panel. Optionally, you can use the <code>View</code> function in the R-console to view an object. For example, you could run <code>View(ts_dat)</code>.</p>
<p>A description of each of the fields can be found under the <a href="https://github.com/hallamlab/TreeSAPP/wiki/Classifying-sequences-with-treesapp-assign#classification-table"><code>treesapp assign</code> documentation page</a>.</p>
</div>
</div>
<div id="subset-your-data-to-the-variables-and-marker-genes-of-interest" class="section level2" number="7.7">
<h2>
<span class="header-section-number">7.7</span> Subset your data to the variables and marker genes of interest<a class="anchor" aria-label="anchor" href="#subset-your-data-to-the-variables-and-marker-genes-of-interest"><i class="fas fa-link"></i></a>
</h2>
<p>Determine which variables and marker genes you will need and subset your data to them. The only “Marker” in these data is XmoA, so we will filter to XmoA.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_df</span> <span class="op">&lt;-</span> <span class="va">ts_dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Marker</span> <span class="op">==</span> <span class="st">"XmoA"</span><span class="op">)</span></code></pre></div>
<p>When working with real data you may be analyzing multiple reference package’s worth of data. If you want to subset those data to a couple of reference packages you will find the ‘%in%’ operator handy. In the following example we are keeping data where the Marker variable is equal to any of “XmoA”, “PmoA” or “AmoA”. In reality, they will only match “XmoA”.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_df</span> <span class="op">&lt;-</span> <span class="va">ts_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Marker</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"XmoA"</span>, <span class="st">"PmoA"</span>, <span class="st">"AmoA"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="reformat-variables" class="section level2" number="7.8">
<h2>
<span class="header-section-number">7.8</span> Reformat variables<a class="anchor" aria-label="anchor" href="#reformat-variables"><i class="fas fa-link"></i></a>
</h2>
<div id="processing-taxonomic-information" class="section level3" number="7.8.1">
<h3>
<span class="header-section-number">7.8.1</span> Processing taxonomic information<a class="anchor" aria-label="anchor" href="#processing-taxonomic-information"><i class="fas fa-link"></i></a>
</h3>
<p>Your taxonomic information contains information from the highest to lowest available rank. Look at a few rows containing taxonomic information and determine the separator between each rank. We can use a function in the tidyverse called <code>separate</code> that allows you split a character object at each of these separators. As a result, your initial single taxonomic variable will be split into multiple new variables, one for each rank. Use the R object <code>taxa_ranks</code> to assign the names for those new variables.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">taxa_ranks</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Root"</span>, <span class="st">"Domain"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>,
                <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span><span class="op">)</span>
<span class="va">ts_df</span> <span class="op">&lt;-</span> <span class="va">ts_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">Taxonomy</span>,
           into <span class="op">=</span> <span class="va">taxa_ranks</span>,
           sep <span class="op">=</span> <span class="st">"; "</span>, fill <span class="op">=</span> <span class="st">"right"</span>, remove <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></code></pre></div>
</div>
<div id="distinguishing-metagenomes-from-metatranscriptomes" class="section level3" number="7.8.2">
<h3>
<span class="header-section-number">7.8.2</span> Distinguishing metagenomes from metatranscriptomes<a class="anchor" aria-label="anchor" href="#distinguishing-metagenomes-from-metatranscriptomes"><i class="fas fa-link"></i></a>
</h3>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_df</span> <span class="op">&lt;-</span> <span class="va">ts_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_pe"</span>, <span class="st">"_MetaG"</span>, <span class="va">Sample</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span>col <span class="op">=</span> <span class="va">Sample</span>,
           into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Cruise"</span>, <span class="st">"Depth"</span>, <span class="st">"SeqType"</span><span class="op">)</span>,
           extra <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></code></pre></div>
</div>
<div id="processing-depth-data" class="section level3" number="7.8.3">
<h3>
<span class="header-section-number">7.8.3</span> Processing depth data<a class="anchor" aria-label="anchor" href="#processing-depth-data"><i class="fas fa-link"></i></a>
</h3>
<p>Modify the data frame to split the sample name into “Cruise” and “Depth” variables. Then, remove the ‘m’ character from the “Depth” variable to create a numeric variable called “Depth.m”.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_df</span> <span class="op">&lt;-</span> <span class="va">ts_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Depth.m <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'m'</span>, <span class="st">''</span>, <span class="va">Depth</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Cruise <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">'SI0'</span>, <span class="st">''</span>, <span class="va">Cruise</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="load-the-geochemical-data" class="section level2" number="7.9">
<h2>
<span class="header-section-number">7.9</span> Load the geochemical data<a class="anchor" aria-label="anchor" href="#load-the-geochemical-data"><i class="fas fa-link"></i></a>
</h2>
<p>Geochemical data for the Saanich Inlet time series is in another file called <code>Saanich_TimeSeries_Chemical_DATA.csv</code>. We will first load it into its own data frame called <code>geochem_df</code>, and subset it to just the samples we are interested in.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">geochem_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"Saanich_TimeSeries_Chemical_DATA.csv"</span>,
                          header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">','</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">Cruise</span> <span class="op">==</span> <span class="fl">72</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">Cruise</span>, <span class="va">Depth</span>,
         <span class="va">CTD_O2</span>, <span class="va">NO3</span>,
         <span class="va">Mean_NH4</span>, <span class="va">Mean_NO2</span>, <span class="va">Mean_H2S</span>, <span class="va">Mean_CH4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Mean_CH4 <span class="op">=</span> <span class="va">Mean_CH4</span><span class="op">*</span><span class="fl">1E-3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/across.html">across</a></span><span class="op">(</span>.fns<span class="op">=</span><span class="va">as.numeric</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We are going to use the function <code>left_join</code> from the <code>dplyr</code> library to merge the geochemical data for cruise 72 with the TreeSAPP classifications for XmoA by Sample. The combined data will be saved to a new data frame called <code>ts_geo_df</code>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_geo_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">ts_df</span>,
                       <span class="va">geochem_df</span>,
                       by<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Cruise"</span>,
                            <span class="st">"Depth.m"</span> <span class="op">=</span> <span class="st">"Depth"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="visualize-the-taxonomic-functional-and-chemical-data-from-saanich-inlet" class="section level2" number="7.10">
<h2>
<span class="header-section-number">7.10</span> Visualize the taxonomic, functional and chemical data from Saanich Inlet<a class="anchor" aria-label="anchor" href="#visualize-the-taxonomic-functional-and-chemical-data-from-saanich-inlet"><i class="fas fa-link"></i></a>
</h2>
<div id="analyze-treesapps-taxonomic-classifications" class="section level3" number="7.10.1">
<h3>
<span class="header-section-number">7.10.1</span> Analyze TreeSAPP’s taxonomic classifications<a class="anchor" aria-label="anchor" href="#analyze-treesapps-taxonomic-classifications"><i class="fas fa-link"></i></a>
</h3>
<p>We’re going to begin with making a line plot to show the distribution of abundances of taxonomic orders at different depths. The only taxa that will be shown are those with an XmoA gene and these are <em>not</em> the only organisms in Saanich Inlet!
We will create a new variable called <code>Sum</code> that sums all the <code>Abundance</code> (TPM values) for each taxonomic order at each depth. Without the <code><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by()</a></code> function, it would return a single value but since we’re grouping by Depth and Order, <code>sum</code> yields a value for each taxon at each depth.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_geo_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">SeqType</span> <span class="op">==</span> <span class="st">"MetaG"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Depth.m</span>, <span class="va">Order</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Sum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">Abundance</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Depth.m</span>, y<span class="op">=</span><span class="va">Sum</span>, colour<span class="op">=</span><span class="va">Order</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_reverse</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Depth (m)"</span>,
       y<span class="op">=</span><span class="st">"Relative abundance (TPM)"</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"taxonomic_order_lineplot.png"</span><span class="op">)</span></code></pre></div>
<p>The above plot gives us an idea of the absolute TPM values, but what is each taxon’s share of the total abundance at each depth? To answer this we will create a stacked barplot to show the relative change in proportions at different depths.
Instead of creating a variable to hold each taxon’s total TPM we will create a new variable to store the proportion of TPM called <code>Proportion</code>. This could easily be changed to a percentage by multiplying it by 100.</p>
<p>In the <code>ts_geo_df</code> dataframe there are two variables storing the sample depth: one is a numeric type and the other is a character type. We will be plotting the character type depth (<code>Depth</code>) on the x-axis but by default R’s alphabetical ordering makes ‘100m’ precede ‘10m’. The <code>reorder</code> function is used here to ensure that the depth is sorted numerically by ordering <code>Depth</code> (character type) by <code>Depth.m</code> (numeric type).</p>
<p>Additionally, we are faceting this plot by the sequence type: ‘MetaG’ (metagenome) and ‘MetaT’ (metatranscriptome).</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_geo_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Depth.m</span>, <span class="va">SeqType</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Proportion <span class="op">=</span> <span class="va">Abundance</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">Abundance</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Depth.m</span>, <span class="va">Order</span>, <span class="va">SeqType</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>sum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">Proportion</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Depth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Depth.m</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Depth</span>, y<span class="op">=</span><span class="va">sum</span>, fill<span class="op">=</span><span class="va">Order</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">SeqType</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>x<span class="op">=</span><span class="st">"Depth"</span>,
       y<span class="op">=</span><span class="st">"Relative abundance"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"taxonomic_order_barplot.png"</span><span class="op">)</span></code></pre></div>
</div>
<div id="analyze-the-functional-data" class="section level3" number="7.10.2">
<h3>
<span class="header-section-number">7.10.2</span> Analyze the functional data<a class="anchor" aria-label="anchor" href="#analyze-the-functional-data"><i class="fas fa-link"></i></a>
</h3>
<p>In the following figure we are instead going to visualize the abundance of each classified XmoA’s function (indicated in the <code>CuMMO</code> field) across the water column.
We are going to use a bubble-plot for this figure where the bubbles at each depth-order intersection are scaled by the TPM value (<code>Abundance</code>).
We are colouring the bubbles according to their assigned <code>CuMMO</code>.
We are again faceting this plot by the sequence type.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_geo_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Depth.m</span>, <span class="va">SeqType</span>, <span class="va">Order</span>, <span class="va">CuMMO</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>Sum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">Abundance</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">ungroup</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Depth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html">desc</a></span><span class="op">(</span><span class="va">Depth.m</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Depth</span>, y<span class="op">=</span><span class="va">Order</span>, colour<span class="op">=</span><span class="va">CuMMO</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>size<span class="op">=</span><span class="va">Sum</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">SeqType</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"functional_bubbleplot.png"</span><span class="op">)</span></code></pre></div>
<p>Can you see any discrepancies between the metagenome and metatranscriptome abundances?</p>
</div>
<div id="analyze-the-chemical-measurements" class="section level3" number="7.10.3">
<h3>
<span class="header-section-number">7.10.3</span> Analyze the chemical measurements<a class="anchor" aria-label="anchor" href="#analyze-the-chemical-measurements"><i class="fas fa-link"></i></a>
</h3>
<p>The following plot is for visualizing the chemical profiles for each depth. This is not exhaustive but does include most of the molecules we’re interested in.</p>
<p>To generate this plot, we’ve had to perform a common data manipulation technique called pivoting. Essentially, instead of having a separate variable for each of the chemical measurements, we have combined them into two columns: one for the name of the molecule (which was previously the variable name) and another for their value.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_geo_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html">starts_with</a></span><span class="op">(</span><span class="st">"Mean"</span><span class="op">)</span>, <span class="va">NO3</span>, <span class="va">CTD_O2</span><span class="op">)</span>,
               values_to <span class="op">=</span> <span class="st">"Value.uM"</span>,
               names_to <span class="op">=</span> <span class="st">"Molecule"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Depth.m</span>, y<span class="op">=</span><span class="va">Value.uM</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_path.html">geom_line</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Molecule</span>, scales <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_flip.html">coord_flip</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html">scale_x_reverse</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"geochemical_lineplot.png"</span><span class="op">)</span></code></pre></div>
<p>Notice that the x-axis scales all differ. This was enabled with <code>scales = "free_x"</code> in the function <code>facet_wrap</code> and was essential to observe the trends across the water column. Without it many of the values would be squished against the y-axis as their values are close to zero and the oxygen concentration gets up to ~200 micromolar.</p>
</div>
<div id="bringing-them-all-together" class="section level3" number="7.10.4">
<h3>
<span class="header-section-number">7.10.4</span> Bringing them all together<a class="anchor" aria-label="anchor" href="#bringing-them-all-together"><i class="fas fa-link"></i></a>
</h3>
<p>In this final figure we are going to combine the methane concentration data with the TPM values for the two orders, <em>Methylococcales</em> and <em>Nitrosomonadales</em>.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ts_geo_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">SeqType</span><span class="op">==</span><span class="st">"MetaT"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="va">Order</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>Depth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Depth.m</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Depth</span>, y<span class="op">=</span><span class="va">Abundance</span>, colour<span class="op">=</span><span class="va">Mean_CH4</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">Order</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"taxonomic_order_boxplot.png"</span><span class="op">)</span></code></pre></div>
<p>How does the abundance of these two groups change with methane concentration?</p>
</div>
</div>
<div id="analyzing-classifications-for-your-groups-reference-package" class="section level2" number="7.11">
<h2>
<span class="header-section-number">7.11</span> Analyzing classifications for your group’s reference package<a class="anchor" aria-label="anchor" href="#analyzing-classifications-for-your-groups-reference-package"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li><p>How does abundance of your gene vary with depth? Are trends similar for both RNA and DNA?</p></li>
<li><p>How does microbial diversity vary with depth? Are trends similar for both RNA and DNA?
You will need to make a decision at what taxonomic rank you will calculate diversity. Why might you not be able to use the lowest possible? Why should your taxonomic rank not be too high?</p></li>
<li><p>How does the abundance of your gene relate to water column geochemistry in Saanich Inlet (use the chemical measurement data in <code>Saanich_TimeSeries_Chemical_DATA.csv</code>)?</p></li>
<li><p>Determine the geochemical conditions at your given depths. Which is the best available terminal electron acceptor?</p></li>
</ul>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></div>
<div class="next"><a href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#analysing-the-treesapp-classifications-in-r"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="nav-link" href="#introduction-and-goals-5"><span class="header-section-number">7.1</span> Introduction and goals</a></li>
<li><a class="nav-link" href="#resources-for-analysis-of-treesapp-output-in-r"><span class="header-section-number">7.2</span> Resources for analysis of TreeSAPP output in R</a></li>
<li><a class="nav-link" href="#checklist-to-write-and-run-r-script"><span class="header-section-number">7.3</span> Checklist to write and run R script</a></li>
<li><a class="nav-link" href="#writing-the-r-script"><span class="header-section-number">7.4</span> Writing the R script</a></li>
<li><a class="nav-link" href="#load-any-required-packages"><span class="header-section-number">7.5</span> Load any required packages</a></li>
<li>
<a class="nav-link" href="#load-metagenomic-and-metatranscriptomic-treesapp-data"><span class="header-section-number">7.6</span> Load metagenomic and metatranscriptomic TreeSAPP data</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#classification-table"><span class="header-section-number">7.6.1</span> Classification table</a></li></ul>
</li>
<li><a class="nav-link" href="#subset-your-data-to-the-variables-and-marker-genes-of-interest"><span class="header-section-number">7.7</span> Subset your data to the variables and marker genes of interest</a></li>
<li>
<a class="nav-link" href="#reformat-variables"><span class="header-section-number">7.8</span> Reformat variables</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#processing-taxonomic-information"><span class="header-section-number">7.8.1</span> Processing taxonomic information</a></li>
<li><a class="nav-link" href="#distinguishing-metagenomes-from-metatranscriptomes"><span class="header-section-number">7.8.2</span> Distinguishing metagenomes from metatranscriptomes</a></li>
<li><a class="nav-link" href="#processing-depth-data"><span class="header-section-number">7.8.3</span> Processing depth data</a></li>
</ul>
</li>
<li><a class="nav-link" href="#load-the-geochemical-data"><span class="header-section-number">7.9</span> Load the geochemical data</a></li>
<li>
<a class="nav-link" href="#visualize-the-taxonomic-functional-and-chemical-data-from-saanich-inlet"><span class="header-section-number">7.10</span> Visualize the taxonomic, functional and chemical data from Saanich Inlet</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#analyze-treesapps-taxonomic-classifications"><span class="header-section-number">7.10.1</span> Analyze TreeSAPP’s taxonomic classifications</a></li>
<li><a class="nav-link" href="#analyze-the-functional-data"><span class="header-section-number">7.10.2</span> Analyze the functional data</a></li>
<li><a class="nav-link" href="#analyze-the-chemical-measurements"><span class="header-section-number">7.10.3</span> Analyze the chemical measurements</a></li>
<li><a class="nav-link" href="#bringing-them-all-together"><span class="header-section-number">7.10.4</span> Bringing them all together</a></li>
</ul>
</li>
<li><a class="nav-link" href="#analyzing-classifications-for-your-groups-reference-package"><span class="header-section-number">7.11</span> Analyzing classifications for your group’s reference package</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/treesapp_R.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/treesapp_R.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on April 13, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
