<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>10 Assessment | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="Reports will account for 30% of the course marks distributed as follows.  10.1 Peer evaluation (10%) Total: 40 pts Peer assessment will occur through CATME wherein everyone will assess themselves...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="10 Assessment | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="Reports will account for 30% of the course marks distributed as follows.  10.1 Peer evaluation (10%) Total: 40 pts Peer assessment will occur through CATME wherein everyone will assess themselves...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="10 Assessment | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="Reports will account for 30% of the course marks distributed as follows.  10.1 Peer evaluation (10%) Total: 40 pts Peer assessment will occur through CATME wherein everyone will assess themselves...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="active" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="references.html"><span class="header-section-number">D</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="assessment" class="section level1">
<h1>
<span class="header-section-number">10</span> Assessment<a class="anchor" aria-label="anchor" href="#assessment"><i class="fas fa-link"></i></a>
</h1>
<p>Reports will account for 30% of the course marks distributed as follows.</p>
<div id="peer-evaluation-10" class="section level2">
<h2>
<span class="header-section-number">10.1</span> Peer evaluation (10%)<a class="anchor" aria-label="anchor" href="#peer-evaluation-10"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Total: 40 pts</strong></p>
<p>Peer assessment will occur through <a href="https://www.catme.org/">CATME</a> wherein everyone will assess themselves and their group members. Marks will be scaled to your evaluation score with 90%+ (or 0.9+ out of 1 within CATME) counting as full marks.</p>
<p>Individuals who do not adequately contribute to their group will forfeit the 5% contributions mark and may be asked to complete the project on their own.</p>
<p>Peer assessment includes:</p>
<ul>
<li>Contributing to the team’s work</li>
<li>Interacting with teammates</li>
<li>Expecting quality</li>
<li>Having related knowledge, skills, and abilities</li>
</ul>
<p>Individuals score themselves and all team members on a scale of 1 (unsatisfactory) – 3 (satisfactory) – 5 (excellent) in these categories. Scores are then averaged and scaled within team such that 1 corresponds to all team members contributing equally to all categories of assessment:</p>
<ul>
<li>1+ = Mostly 5s</li>
<li>0.95 = Mostly 4s</li>
<li>0.8 = Mostly 3s</li>
<li>0.65 = Mostly 2s</li>
<li>0.5 = Mostly 1s</li>
</ul>
<p>Marks will be scaled to your CATME score based on the following:</p>
<ul>
<li>0.9 - 1+ = 100% = 40</li>
<li>0.8 - 0.89 = 90% = 36</li>
<li>0.7 - 0.79 = 80% = 32</li>
<li>0.6 - 0.69 = 70% = 28</li>
<li>&lt;0.6 = Assessed on a case-by-case basis</li>
</ul>
<p><em>Unwarranted blanketed low scoring of your team or high scoring of yourself will result in the removal of your scores from the average so that they do not unfairly impact you or your team members.</em></p>
<p>If issues occur within your group, please do not hesitate to contact Dr. Hallam (<a href="mailto:shallam@mail.ubc.ca" class="email">shallam@mail.ubc.ca</a>), or Dr. Koenig (<a href="mailto:stephan.koenig@ubc.ca" class="email">stephan.koenig@ubc.ca</a>) <em>at any point</em>.</p>
</div>
<div id="report-20" class="section level2">
<h2>
<span class="header-section-number">10.2</span> Report (20%)<a class="anchor" aria-label="anchor" href="#report-20"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Total: 80 pts</strong></p>
<p>Each group will submit an electronic copy of their group’s final report (due <strong>April 24</strong>). Reports will be assessed on:</p>
<ul>
<li>Completion of relevant analyses toward answering biological questions</li>
<li>Logic and completeness of conclusions made from these analyses</li>
<li>Writing clarity, grammar, and style</li>
<li>Figure clarity, effectiveness, and relevance</li>
</ul>
<!-- Rubric for capstone project, edit with RStudio visual editor --><div class="inline-table"><table class="table table-sm">
<caption>Rubric for capstone project</caption>
<colgroup>
<col width="3%">
<col width="19%">
<col width="23%">
<col width="23%">
<col width="29%">
</colgroup>
<thead><tr class="header">
<th></th>
<th><strong>Poor (0–1 pts)</strong></th>
<th><strong>Below average (2–4 pts)</strong></th>
<th><strong>Good (5–7 pts)</strong></th>
<th><strong>Excellent (8–10 pts)</strong></th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Abstract</strong></td>
<td>Not provided</td>
<td>Summarizes only part of the report or contains numerous inaccuracies</td>
<td>Summarizes most of the report including relevant results; Some inaccuracies or missing pieces</td>
<td>Accurately summarizes the report including background, relevant results, and conclusions; Few to no inaccuracies</td>
</tr>
<tr class="even">
<td><strong>Introduction</strong></td>
<td>Background information not provided or provided but not relevant to the research question(s); Hypotheses / questions are not stated nor supported by relevant sources</td>
<td>Background is relevant but not sufficient to frame the research question(s); Hypotheses / questions are unclear and/or unsupported</td>
<td>Background is relevant but does not fully frame the research question(s); Hypotheses / questions are unclear or unsupported</td>
<td>Background is relevant and fully frames the research question(s); Hypotheses / questions are clearly stated and supported by scientific sources</td>
</tr>
<tr class="odd">
<td><strong>Methods</strong></td>
<td>Incomplete and missing numerous methods used; Unclear or confusing</td>
<td>Missing several methods used; Numerous errors or unclear statements</td>
<td>Missing one or more methods used; Results could not be replicated due to issues with clarity or accuracy</td>
<td>Sufficient to allow replication of the results including sampling, sequencing, processing, and analysis; Succinct and not overly wordy</td>
</tr>
<tr class="even">
<td><strong>Code (R and TreeSAPP)</strong></td>
<td>Code missing significant portions or is not provided</td>
<td>Code is incomplete or erroneous such that results cannot be replicated; No explanation of code is given</td>
<td>Code replicates results but is verbose or inefficient; Minimal commenting is provided</td>
<td>Code replicates results and is efficient; Detailed explanation of code is provided in comments using # notation</td>
</tr>
<tr class="odd">
<td><strong>Results</strong></td>
<td>Most of the research questions are not addressed or the results are consistently incomplete or irrelevant</td>
<td>Several research questions are not addressed; Numerous results are incomplete, erroneous, or irrelevant</td>
<td>1 or more research questions are not fully addressed; Some results are incomplete or do not to pertain to the questions</td>
<td>All research questions are investigated with relevant analyses and figures; Results are clearly stated and not incorrectly or over-interpreted</td>
</tr>
<tr class="even">
<td><strong>Figures &amp; captions</strong></td>
<td>2 or fewer figures are given; Captions are incomplete or missing</td>
<td>3+ relevant figures are provided but lack proper formatting or completeness; Captions lack numerous details or descriptions</td>
<td>4+ relevant figures are provided with mostly proper formatting; Captions lack some details or descriptions</td>
<td>5+ relevant figures are provided with proper formatting; Captions are included for all figures and contain a title as well as description of axes, other aesthetics, and overall data trends</td>
</tr>
<tr class="odd">
<td><strong>Discussion</strong></td>
<td>Synthesis of results is missing or does not pertain to the data; Statements are not supported by the data</td>
<td>Conclusions are incomplete or not relevant to the data; Multiple research questions are not addressed or results are incorrectly interpreted</td>
<td>Conclusions restate results but without summarization or synthesis across depths; 1 or more questions are not addressed; Some results are incorrectly interpreted</td>
<td>Results are summarized within the context of the original questions and more broadly across depths; Results are correctly interpreted and discussed with scientific language; Conclusions are supported by the data and 1 or more future directions are proposed</td>
</tr>
<tr class="even">
<td><strong>Writing, grammar, &amp; format</strong></td>
<td>Grammatical errors, spelling mistakes, and/or language cause significant issues in understanding of the content throughout</td>
<td>Numerous grammatical, spelling, or language errors that negatively impact understanding; Some language is overly verbose or informal; One or more sections significantly differ from the recommended length</td>
<td>Some grammatical, spelling, or language errors that negatively impact understanding occur; Some language is overly verbose or informal; One section may significantly differ from the recommended length</td>
<td>Minimal to no errors; Formal scientific language used throughout; Text flows and is easy to read; Sections adhere to word limits; Report provided in J Bacteriology format</td>
</tr>
</tbody>
</table></div>
</div>
<div id="ujemi-submission" class="section level2">
<h2>
<span class="header-section-number">10.3</span> UJEMI submission<a class="anchor" aria-label="anchor" href="#ujemi-submission"><i class="fas fa-link"></i></a>
</h2>
<p>Outstanding reports will be invited to submit to The Undergraduate Journal of Experimental Microbiology and Immunology (<a href="https://jemi.microbiology.ubc.ca/">UJEMI</a>). More information on this will be provided near the end of the Finals Period.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="project-description.html"><span class="header-section-number">9</span> Project description</a></div>
<div class="next"><a href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#assessment"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="nav-link" href="#peer-evaluation-10"><span class="header-section-number">10.1</span> Peer evaluation (10%)</a></li>
<li><a class="nav-link" href="#report-20"><span class="header-section-number">10.2</span> Report (20%)</a></li>
<li><a class="nav-link" href="#ujemi-submission"><span class="header-section-number">10.3</span> UJEMI submission</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/assessment.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/assessment.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on March 13, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
