<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>6 Calculating relative abundance of meta’omic data | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="6.1 Introduction and goals Genome assemblies contain less information than the sequence reads and recovering it is not always possible. The shotgun DNA sequencing technique samples subsequences...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="6 Calculating relative abundance of meta’omic data | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="6.1 Introduction and goals Genome assemblies contain less information than the sequence reads and recovering it is not always possible. The shotgun DNA sequencing technique samples subsequences...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="6 Calculating relative abundance of meta’omic data | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="6.1 Introduction and goals Genome assemblies contain less information than the sequence reads and recovering it is not always possible. The shotgun DNA sequencing technique samples subsequences...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="active" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="references.html"><span class="header-section-number">D</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="calculating-relative-abundance-of-metaomic-data" class="section level1">
<h1>
<span class="header-section-number">6</span> Calculating relative abundance of meta’omic data<a class="anchor" aria-label="anchor" href="#calculating-relative-abundance-of-metaomic-data"><i class="fas fa-link"></i></a>
</h1>
<div id="introduction-and-goals-4" class="section level2">
<h2>
<span class="header-section-number">6.1</span> Introduction and goals<a class="anchor" aria-label="anchor" href="#introduction-and-goals-4"><i class="fas fa-link"></i></a>
</h2>
<p>Genome assemblies contain less information than the sequence reads and recovering it is not always possible.
The shotgun DNA sequencing technique samples subsequences from the genomes that comprise a microbial community. Sequencing bias, while present, is subtle enough that we can assume that each subsequence’s frequency in the reads approximates its abundance in the underlying community. Whenever a (meta)genome is assembled from reads, however, sequences that are tricky to assemble (due to repetitive motifs or low coverage) are removed and the abundance information is effectively lost.
One way to recover these read-level abundances is by aligning the reads back to the contigs and calculating the number of reads that mapped to each. One can go a step further still and use relative abundance normalization measures popular in transcriptomics, such as fragments per kilobase per million mappable reads (FPKM) or transcripts per million (TPM) <span class="citation">(<a href="references.html#ref-Wagner.2012" role="doc-biblioref">12</a>)</span>.</p>
<p>TreeSAPP’s subcommand <code>abundance</code> is capable of calculating FPKM or TPM for all classified query sequences. These can be derived from the reads in the FASTQ files of multiple different samples. This is only possible when the input fasta are nucleotide sequences from which ORFs were predicted within the pipeline (i.e. this thisn’t possible with amino acid sequences).</p>
<p>The goal of this tutorial is to show you how to use <code>treesapp abundance</code> to calculate TPM for metagenome-derived ORF sequences.</p>
</div>
<div id="workflow-description" class="section level2">
<h2>
<span class="header-section-number">6.2</span> Workflow description<a class="anchor" aria-label="anchor" href="#workflow-description"><i class="fas fa-link"></i></a>
</h2>
<p>The workflow employed by TreeSAPP is common to many bioinformatics pipelines:</p>
<ol style="list-style-type: decimal">
<li>Collect nucleotide sequences for the classified ORFs.</li>
<li>BWA <span class="citation">(<a href="references.html#ref-Li.2013" role="doc-biblioref">13</a>)</span> creates an index from these nucleotide sequences - this will be the reference database.</li>
<li>Align the short reads to the reference database with BWA MEM.</li>
<li>Calculate the relative abundance metrics. TreeSAPP uses <a href="https://github.com/hallamlab/samsum">samsum</a>.</li>
</ol>
<p>There are potentially two common use cases for <code>treesapp abundance</code>.</p>
<ol style="list-style-type: decimal">
<li>Through <code>treesapp assign</code> where a single sample’s read data are mapped to its corresponding assembly (i.e. contigs or scaffolds) and the relative abundances of these sequences are obtained.</li>
<li>A combined assembly was generated from multiple samples, classified with <code>treesapp assign</code>, then the reads of each sample were mapped individually to generate sample-specific relative abundance values with <code>treesapp abundance</code>.</li>
</ol>
</div>
<div id="usage" class="section level2">
<h2>
<span class="header-section-number">6.3</span> Usage<a class="anchor" aria-label="anchor" href="#usage"><i class="fas fa-link"></i></a>
</h2>
<p>There are two required arguments for <code>treesapp abundance</code>:</p>
<ol style="list-style-type: decimal">
<li>
<strong>treesapp_output</strong>: Path to the output directory from <code>treesapp assign</code>.</li>
<li>
<strong>reads</strong>: Path to a FASTQ file containing single-end reads, or foward mates or interleaved reads from a paired-end sequencing library.</li>
</ol>
<p>Below are descriptions for the most important options:</p>
<ol style="list-style-type: decimal">
<li>
<strong>reverse</strong>: Path to a FASTQ file containing reverse mates from a paired-end sequencing library.</li>
<li>
<strong>report</strong>: This argument controls what <code>treesapp abundance</code> does with the resulting abundance values. The ‘<em>nothing</em>’ option is most applicable to the Python API, and is used by <code>treesapp assign</code> to simply return the abundance object instances. ‘<em>update</em>’ can be used when the abundance values in the classification table should be replaced with these newly calculated values. ‘<em>append</em>’ should be used when reads of new samples are being used to calculate abundance (e.g. reads from a time-course experiment).</li>
<li>
<strong>metric</strong>: Either Fragments per Kilobase per Mllion mappable reads (FPKM) or Transcripts per Million (TPM).</li>
</ol>
</div>
<div id="outputs" class="section level2">
<h2>
<span class="header-section-number">6.4</span> Outputs<a class="anchor" aria-label="anchor" href="#outputs"><i class="fas fa-link"></i></a>
</h2>
<p>The sole output of <code>treesapp abundance</code> is an edited classification table, previously generated by <code>treesapp assign</code>. No new files are generated, beyond intermediate files (e.g. SAM files). The values are in the ‘Abundance’ field of <code>final_outputs/classifications.tsv</code>.</p>
</div>
<div id="tutorial-steps" class="section level2">
<h2>
<span class="header-section-number">6.5</span> Tutorial steps<a class="anchor" aria-label="anchor" href="#tutorial-steps"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>Connect to your group’s server, then move to the directory <code>/data/ts_tutorial</code>.</li>
</ul>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb38-1"><a href="calculating-relative-abundance-of-metaomic-data.html#cb38-1"></a><span class="bu">cd</span> /data/ts_tutorial/</span></code></pre></div>
<p>This tutorial requires many FASTQ and FASTA-formatted files that are subsetted versions of the originals.
They were created by searching metagenome assemblies and reads for XmoA then pulling out all sequences that matched into new files.
They are all available in the <code>SI072_sequence_data</code> directory downloaded during the last tutorial from <a href="https://doi.org/10.5281/zenodo.6323402"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.6323402.svg" alt="DOI"></a>.</p>
<p>We will begin by classifying XmoA sequences found in the metagenome contigs. Since these are nucleotide sequences representing random stretches of DNA from the microbial community, <code>treesapp assign</code> will begin by predicting open reading frames (ORFs) from these. TreeSAPP will then classify the amino acid sequences for each of the predicted ORFs. Once the XmoA sequences are classified from these ORFs we can begin to generate the relative abundance values for the XmoA sequences.</p>
<ul>
<li>Assign taxonomic labels to Saanich Inlet metagenomic contigs using the updated XmoA reference package with <code>treesapp assign</code>.</li>
</ul>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb39-1"><a href="calculating-relative-abundance-of-metaomic-data.html#cb39-1"></a><span class="ex">treesapp</span> assign \</span>
<span id="cb39-2"><a href="calculating-relative-abundance-of-metaomic-data.html#cb39-2"></a>  -n 4 \</span>
<span id="cb39-3"><a href="calculating-relative-abundance-of-metaomic-data.html#cb39-3"></a>  --trim_align \</span>
<span id="cb39-4"><a href="calculating-relative-abundance-of-metaomic-data.html#cb39-4"></a>  --refpkg_dir XmoA_MAG_update/final_outputs/ \</span>
<span id="cb39-5"><a href="calculating-relative-abundance-of-metaomic-data.html#cb39-5"></a>  --fastx_input SI072_sequence_data/SI072_MetaG_contigs.fasta \</span>
<span id="cb39-6"><a href="calculating-relative-abundance-of-metaomic-data.html#cb39-6"></a>  --output SI072_MetaG_contigs_XmoA_assign/</span></code></pre></div>
<p>Now we’ll need to access the processed data generated by <code>treesapp assign</code> in <code>SI072_MetaG_contigs_XmoA_assign/</code>. <code>treesapp abundance</code> will access the ORFs in <code>SI072_MetaG_contigs_XmoA_assign/intermediates/orf-call/SI072_MetaG_contigs_ORFs.fna</code> and pass these to <code>bwa mem</code> for sequence alignment. Due to this, the flag ‘--delete’ should not be used with <code>treesapp assign</code> if you want to eventually run <code>treesapp abundance</code> as the <code>intermediates</code> directory would be removed.</p>
<p>We will begin by calculating TPM for the metagenome reads.
The reads we are using to calculate these values are <em>paired-end</em> meaning that the there are two reads to represent each sequenced DNA fragment. These two reads, in the case of the metagenomic data, are split across two files: one for the reads sequenced in the forward direction and another for the reads sequenced in the reverse. Their extensions are ‘.1.fq.gz’ and ‘.2.fq.gz’, respectively. Therefore, we must use the parameters ‘--reads’ and ‘--reverse’ to point TreeSAPP towards these files. You’ll notice that we didn’t list out seven files for each, either; we used a <em>wildcard</em> character (the asterisk). For the forward reads, the shell will interpret this as all files within the <code>SI072_sequence_data</code> directory that begin with ‘SI072_’ and end with ‘m_pe.1.fq.gz’. You can test this expansion using the <code>ls</code> command.</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb40-1"><a href="calculating-relative-abundance-of-metaomic-data.html#cb40-1"></a><span class="fu">ls</span> SI072_sequence_data/SI072_*m_pe.1.fq.gz</span></code></pre></div>
<p>We would like to replace the current abundance values from <code>treesapp assign</code> (set to 1) with the new TPM values, so we will set the ‘--report’ parameter to ‘update’.
Finally, we can specify that we want TPM values with <code>--metric tpm</code>. Although, ‘tpm’ is the default so this isn’t necessary.</p>
<ul>
<li>Run <code>treesapp abundance</code> to overwrite the previous abundance values with the TPM values calculated from each of the seven SI072 datasets.</li>
</ul>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb41-1"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-1"></a><span class="ex">treesapp</span> abundance \</span>
<span id="cb41-2"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-2"></a>  -n 4 \</span>
<span id="cb41-3"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-3"></a>  --treesapp_output SI072_MetaG_contigs_XmoA_assign/ \</span>
<span id="cb41-4"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-4"></a>  --reads SI072_sequence_data/SI072_*m_pe.1.fq.gz \</span>
<span id="cb41-5"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-5"></a>  --reverse SI072_sequence_data/SI072_*m_pe.2.fq.gz \</span>
<span id="cb41-6"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-6"></a>  --report update \</span>
<span id="cb41-7"><a href="calculating-relative-abundance-of-metaomic-data.html#cb41-7"></a>  --metric tpm</span></code></pre></div>
<p>Now we’re going to generate TPM values for the seven matching metatranscriptome datasets. You will notice the format of these FASTQ files is different from the metagenomes. All of the reads for each sample are contained in a single file, and this is referred to as <em>interleaved</em> FASTQ format.
Since there is no file for the reverse reads we can just use the ‘--reads’ parameter.
Another key difference is we’re setting the ‘--report’ parameter to ‘append’. This is because we don’t want to overwrite the TPM values we just generated for the metagenomes. Instead, we want the new metatranscriptome values to be added to <code>SI072_MetaG_contigs_XmoA_assign/final_outputs/classifications.tsv</code>, so in the end the classification table will hold TPM values for both the metagenomes and metatranscriptomes.</p>
<ul>
<li>Use <code>treesapp abundance</code> to calculate TPM values from the seven metatranscriptomes.</li>
</ul>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb42-1"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-1"></a><span class="ex">treesapp</span> abundance \</span>
<span id="cb42-2"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-2"></a> -n 4 \</span>
<span id="cb42-3"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-3"></a> --treesapp_output SI072_MetaG_contigs_XmoA_assign/ \</span>
<span id="cb42-4"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-4"></a> --reads SI072_sequence_data/SI072_*m_MetaT_QC_Filtered.fq.gz \</span>
<span id="cb42-5"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-5"></a> --pairing pe \</span>
<span id="cb42-6"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-6"></a> --metric tpm \</span>
<span id="cb42-7"><a href="calculating-relative-abundance-of-metaomic-data.html#cb42-7"></a> --report append</span></code></pre></div>
<p>The next tutorial will cover visualizing the TreeSAPP classifications. For this we also want the functional annotations that we generated in the last tutorial.</p>
<ul>
<li>Run <code>treesapp layer</code> to annotate the XmoA query sequences with their respective paralog.</li>
</ul>
<div class="sourceCode" id="cb43"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb43-1"><a href="calculating-relative-abundance-of-metaomic-data.html#cb43-1"></a><span class="ex">treesapp</span> layer \</span>
<span id="cb43-2"><a href="calculating-relative-abundance-of-metaomic-data.html#cb43-2"></a> -o SI072_MetaG_contigs_XmoA_assign/ \</span>
<span id="cb43-3"><a href="calculating-relative-abundance-of-metaomic-data.html#cb43-3"></a> --refpkg_dir XmoA_MAG_update/final_outputs/</span></code></pre></div>
<p>That’s it! You’re now a seasoned veteran of TreeSAPP.</p>
</div>
<div id="use-your-groups-reference-package-to-calculate-abundances" class="section level2">
<h2>
<span class="header-section-number">6.6</span> Use your group’s reference package to calculate abundances<a class="anchor" aria-label="anchor" href="#use-your-groups-reference-package-to-calculate-abundances"><i class="fas fa-link"></i></a>
</h2>
<p>Continue to work with your group’s reference package in the directory <code>/data/&lt;gene name&gt;</code>. Visualize the appropriate outputs with iTOL.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></div>
<div class="next"><a href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#calculating-relative-abundance-of-metaomic-data"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="nav-link" href="#introduction-and-goals-4"><span class="header-section-number">6.1</span> Introduction and goals</a></li>
<li><a class="nav-link" href="#workflow-description"><span class="header-section-number">6.2</span> Workflow description</a></li>
<li><a class="nav-link" href="#usage"><span class="header-section-number">6.3</span> Usage</a></li>
<li><a class="nav-link" href="#outputs"><span class="header-section-number">6.4</span> Outputs</a></li>
<li><a class="nav-link" href="#tutorial-steps"><span class="header-section-number">6.5</span> Tutorial steps</a></li>
<li><a class="nav-link" href="#use-your-groups-reference-package-to-calculate-abundances"><span class="header-section-number">6.6</span> Use your group’s reference package to calculate abundances</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/treesapp_abundance.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/treesapp_abundance.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on March 18, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
