# The Saanich Inlet data set

## Description 

We will work with real-world data collected as part of an ongoing oceanographic time series program in Saanich Inlet, a seasonally anoxic fjord on the East coast of Vancouver Island, British Columbia:

```{r saanich-inlet, echo = FALSE, fig.cap = "The Saanich Inlet"}
knitr::include_graphics("images/saanich_inlet_map.png")
```

Figure¬†\@ref(fig:saanich-inlet) shows a map of Saanich Inlet indicating conventional sample collection stations (S1-S9). The data used in this tutorial (sourced from S3, Cruise 72) include various geochemical measurements and the genomic and transcriptomic data of microbial samples at depths 10, 100, 120, 135, 150, 165 and 200 m.

For more details about these data, see @Hallam.2017, and for more detailed information on the environmental context and time series data, see @Torres-Beltr√°n.2017.

## Data on the server

The data is located on the MICB425 server at `/mnt/datasets/saanich/`.

```{r echo = FALSE, comment = NA}
cat(paste(readLines("data/directory_structure.txt"),
          collapse = "\n"),
    fill = TRUE)
```

## Description of input data

List of directories and their contents

-  `MAGs`

    Within the `Concatonated` subdirectory, there is a single `All_SI072_Metawrap_MAGs.fa` fasta file which you will use with both `treesapp assign` and `treesapp update`. After the MAGs were generated with the MetaWrap pipeline @Uritskiy.2018 (see [Methods]), sample IDs were added to the contig headers of each MAG produced from each SI072 sample depth. The individual MAGs were then concatenated into this file.

-  `MetaG_Assemblies`

    This directory contains the seven assembled metagenomic contigs for each SI072 sample depth. These files will be used for the `treesapp assign` and `abundance` tutorials. These files were generated by assembling the trimmed and quality controlled reads that were generated by the Illumina Sequencer.

-  `MetaG_Trim_QC_Reads`

    This directory contains fourteen fastq files which contain the paired-end forward and reverse metagenomic reads for each SI072 sample depth. These trimmed and QC'd fastqs were generated by processing the raw reads through the Megahit workflow described below. You will use these files for treesapp abundance to get the metagenomic abundance of your assigned genes.

-  `MetaT_Raw_Reads`

    This directory contains seven filtered and QC'd metatranscriptome fastq files for each SI072 Sample depth. These files are paired end and interleaved. These files we provided by the the JGI and generated based on the methodology below. You will use these files in `treesapp abundance` to calculate the metatransciptomic abundances of each gene of interest.

-  `SAGs`

    Within the `Concatenated_Med_Plus` subdirectory, there is a single `Saanich_Med_Plus_SAGs.fasta` file which you will use with both `treesapp assign` and `treesapp update`. After the MAGs were generated with the SAG Assembly and Decontamination Workflow [@Bolger.2014; @Prjibelski.2020; @Parks.2015; @Tennessen.2016] (see [Methods]), sample IDs were added to the contig headers of each MAG produced from each SI072 sample depth. The individual MAGs were then concatenated into this file.

-  `seq2lineage_Tables`

    This directory contains two files that will be used for treesapp update. One file contains the lineage information from the MAGs, and the other contains information for the SAGs. These tables were generated by taking each separate MAG and SAG respectively and passing them through the Genome Taxonomic Database Toolkit v1.4.0 classify workflow (see [Methods]). Then, the first two columns of the archaeal and bacterial summary files were taken and combined into each of these files.

## Methods

### Metagenomes and metatransriptopmes

#### Sample Collection
Water was collected at 10, 100, 120, 135, 150, 165, and 200m at Saanich Inlet on August 1st, 2012 and filtered through a 0.22¬†¬µm Sterivex filter to collect biomass. Total genomic DNA and RNA were extracted from these filters. RNA was reversed transcribed into cDNA and both genomic DNA and cDNA were used in the construction of shotgun Illumina libraries. Sequencing data were generated on the Illumina HiSeq platform with **2x150bp** technology. For more in-depth sampling and sequencing methods, please see @Hawley.2017.


#### Quality filtering of reads

The resulting reads were processed, quality controlled and filtered with Trimmomatic (v.0.35, [documentation](http://www.usadellab.org/cms/?page=trimmomatic)) @Bolger. Trimmomatic command is as follows, where SI072_XXX represents the same command being applied to each of the seven depths.


```
trimmomatic -threads 8 -phred33 \
Path_to/SI072_XXXm/Raw/forward_dir/SI072_XXXm_R1.fastq \
Path_To/SI072_200m/Raw/reverse_dir/SI072_XXXm_R2.fastq \

Path_To/SI072_200m//trim/SI072_XXXm_pe.1.fq \

Path_To/SI072_200m//trim/SI072_XXXm_pe.2.fq \

ILLUMINACLIP:/usr/local/share/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:3:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
```

The resulting trimmed fastq files were also deposited into the MICB425 server in `/mnt/datasets/saanich/MetaG_Trim_QC_Reads`

#### Contig Assembly

MEGAHIT (v.1.1.3, [documentation](https://github.com/voutcn/megahit/wiki/Assembly-Tips)) @Li.2015 was used to assembled the filtered metagenomic reads into contigs.

```
megahit -1 Path_To/SI072_200m//trim/SI072_XXXm_pe.1.fq  -2 Path_To/SI072_200m//trim/SI072_XXXm_pe.2.fq  -m 0.5  -t 12  -o megahit_result --k-min 27 --k-step 10 --min-contig-len 500
```
The resulting assemblies can be found in `/mnt/datasets/saanich/MetaG_Assemblies`.

#### Binning and MAG Genration

MetaWRAP (v1.2.4, [documentation](https://github.com/bxlab/metaWRAP))[ref] was used to generate the MAGs from the metagenome assemblies and filtered reads for this project. MetaWRAP leverages high-quality draft genomes (bins) from metagenomic data by using a variety of software (we chose to use MetaBAT2 v2.12.1 [ref] and MaxBin v2.2.7[ref]). Quality of the resulting bins was calculated with MetaWRAP's implementation of CheckM (v1.0.12)[ref]. The command used to invoke MetaWRAP is as follows:

```
metawrap binning -a SI072_Assembly.fa -o output_directory -t 16 -m 64 -l 1500 --metabat2 --maxbin2 --universal Path_To/SI072_200m//trimSI072_XXXm_pe.1.fastq Path_To/SI072_200m//trimSI072_XXXm_pe.2.fastq
```

#### Taxonomic Assignment of MAGs
The resulting MAGs were then passed through GTDB-TK v1.4.0 classify workflow with the reference data version r95 [documentation](https://ecogenomics.github.io/GTDBTk/commands/classify_wf.html)[ref].

```
gtdbtk classify_wf --genome_dir Path_To_SI072/MAGs/ -x .fa --out_dir Path_To_SI072/MAG_GTDB_Outputs/ --cpus 8 --pplacer_cpus 8

```
The after updating the headers for resulting bins with the sample IDs, the bins were concatenated and deposited into: `/mnt/datasets/saanich/MAGs/Concatonated/` 

### Single-Cell Amplified Genomes

#### Sample Collection
Water was collected at 10, 100, 120, 135, 150, 165, and 200m at Saanich Inlet on August 9th, 2011. 900‚Äâ¬µL of seawater was alequoted into 100‚Äâ¬µL of glycerol-TE buffer, then stored at -80 degrees Celsius. Samples were thawed and had their microbial cells sorted at the Bigelow Laboratory for Ocean Sciences‚Äô Single Cell Genomics Center (scgc.bigelow.org). 


#### Flourecence Activated Cell Sorting
Samples were passed through a sterile 40 ùúám size mesh before microorganisms were sorted by a non-targeted isolation procedure. For non-target isolation, the microbial particles were labeled with a 5‚ÄâùúáM final concentration of the DNA stain SYTO-9 (Thermo Fisher Scientific). Microbial cells were individually sorted using a MoFloTM (Beckman Coulter) or an InFluxTM (BD Biosciences) flow cytometer system equipped with a 488 nm laser for excitation and a 70 Œºm nozzle orifice30. The gates for the untargeted isolation of microbial cells stained with SYTO-9 were defined based on the green fluorescence of particles as a proxy for nucleic acid content, and side scattered light as a proxy for particle size. All microbial single cells were sorted into 384-well plates containing 600 nL of 1X TE buffer per well and then stored at -80 degrees Celsius. The microbial single-cells sorted into TE buffer were lysed as described previously by adding  cold KOH[ref].The microbial nucleic acids were then amplified in each individual wells through Multiple Displacement Amplification (MDA)[ref]. 


#### Amplicon Taxonomic Screening
SAGs generated through the MDA methodology were taxonomically characterized by sequencing a determined phylogenetic marker using an amplification based approach. SAGs were screened for their small subunit ribosomal rRNA gene (SSU rRNA) using primers for bacterial (27-F: 5‚Äô- AGAGTTTGATCMTGGCTCAG -3‚Äô37, 907-R: 5‚Äô- CCGTCAATTCMTTTRAGTTT -3‚Äô38 and archaeal sequences (Arc_344F: 5‚Äô- ACGGGGYGCAGCAGGCGCGA -3‚Äô39, Arch_915R: 5‚Äô- GTGCTCCCCCGCCAATTCCT -3‚Äô40). The real-time PCR and sequencing procedure of the resulting amplicons was performed as previously described [ref16,30]. The partial SSU rRNA gene sequences were then queried against the SILVA database v138.1[41ref] with blastn, from BLAST+ v2.9.0[ref].

```
 blastn -query -db -outfmt "6 qacc sacc stitle staxid pident bitscore" -max_target_seqs 1 -num_threads 4 -out 
```

#### SAG Sequencing and Assembly
MDA products for the SAGs were sequenced as described previously [16,30] on an Illumina HiSeq 2000. The resulting raw reads were then filtered and trimmed to remove adapotrs with Trimmomatic v0.35.

```
trimmomatic -threads 8 -phred33 \
Path_to/SI072_XXXm/Raw/forward_dir/SI072_XXXm_R1.fastq \
Path_To/SI072_200m/Raw/reverse_dir/SI072_XXXm_R2.fastq \

Path_To/SI072_200m//trim/SI072_XXXm_pe.1.fq \
Path_To/SI072_200m//trim/SI072_XXXm_se.1.fq \
Path_To/SI072_200m//trim/SI072_XXXm_pe.2.fq \
Path_To/SI072_200m//trim/SI072_XXXm_se.2.fq \

ILLUMINACLIP:/usr/local/share/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:3:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
```

Trimmed and quality controlled reads were the then assembeld with SPAdes v3.9.0[ref] with standard parameters.

```
spades.py \
--sc \
-o Path_To_assembly_dir/ \
-1 Path_To_trim_dir/SI072_XXXm_pe.1.fq \
-2 Path_To_trim_dir/SI072_XXXm_pe.1.fq \
-s Path_To_trim_dir/$sample\_unpaired.fq \
-k $k_series \
--memory 90 \
--threads $threads \
--careful \
--cov-cutoff auto

```

#### SAG Assembly Quality Control and Decontamination

Quality of the SAGs was assessed using CheckM(v1.0.5) with the following command:

```
checkm lineage_wf --tab_table -x .fna --threads 8 --pplacer_threads 8
```

If a SAG was >5% contamination, then the SAGs were passed through ProDeGe v2.3.0[ref].The resulting decontaminated SAG was re-assesed with CheckM. If the SAG still exceeded 5% contamination after several iterations, then short contigs <2000bp were removed from the assembly and re-assessed with CheckM. 

For this project, Based on the resulting CheckM scores, SAGs that were >50% Complete with <10% contamination were selected.

#### Final Taxonomic Assignment of SAGs

The resulting SAGs were then passed through GTDB-TK v1.4.0 [ref]classify workflow [documentation](https://ecogenomics.github.io/GTDBTk/commands/classify_wf.html)[ref].

```
gtdbtk classify_wf --genome_dir Path_To_SI_SAGs/ -x .fna --out_dir Path_To_SI_SAGs/GTDB_1.4.0/ --cpus 8 --pplacer_cpus 8

```
The After updating the headers for resulting SAGs with the sample IDs, the SAGs were concatenated and deposited into: `/mnt/datasets/saanich/SAGs/Concatonated_Med_Plus/`

