# The Saanich Inlet data set

## Description 

We will work with real-world data collected as part of an ongoing oceanographic time series program in Saanich Inlet, a seasonally anoxic fjord on the East coast of Vancouver Island, British Columbia:

```{r saanich-inlet, echo = FALSE, fig.cap = "The Saanich Inlet"}
knitr::include_graphics("images/saanich_inlet_map.png")
```

Figure \@ref(fig:saanich-inlet) shows a map of Saanich Inlet indicating conventional sample collection stations (S1-S9). The data used in this tutorial (sourced from S3, Cruise 72) include various geochemical measurements and the genomic and transcriptomic data of microbial samples at depths 10, 100, 120, 135, 150, 165 and 200 m.

For more details about these data, see @Hallam.2017, and for more detailed information on the environmental context and time series data, see @Torres-Beltrán.2017.

## Data on the server

The data is located on the MICB425 server at `/mnt/datasets/saanich/`.

```{r echo = FALSE, comment = NA}
cat(paste(readLines("data/directory_structure.txt"),
          collapse = "\n"),
    fill = TRUE)
```

## Description of input data

List of directories and their contents

-  `MAGs`

    Within the `Concatonated` subdirectory, there is a single `All_SI072_Metawrap_MAGs.fa` fasta file which you will use with both `treesapp assign` and `treesapp update`. After the MAGs were generated with the MetaWrap pipeline @Uritskiy.2018 (see [Methods]), sample IDs were added to the contig headers of each MAG produced from each SI072 sample depth. The individual MAGs were then concatenated into this file.

-  `MetaG_Assemblies`

    This directory contains the seven assembled metagenomic contigs for each SI072 sample depth. These files will be used for the `treesapp assign` and `abundance` tutorials. These files were generated by assembling the trimmed and quality controlled reads that were generated by the Illumina Sequencer.

-  `MetaG_Trim_QC_Reads`

    This directory contains fourteen fastq files which contain the paired-end forward and reverse metagenomic reads for each SI072 sample depth. These trimmed and QC'd fastqs were generated by processing the raw reads through the Megahit workflow described below. You will use these files for treesapp abundance to get the metagenomic abundance of your assigned genes.

-  `MetaT_Raw_Reads`

    This directory contains seven filtered and QC'd metatranscriptome fastq files for each SI072 Sample depth. These files are paired end and interleaved. These files we provided by the the JGI and generated based on the methodology below. You will use these files in `treesapp abundance` to calculate the metatransciptomic abundances of each gene of interest.

-  `SAGs`

    Within the `Concatenated_Med_Plus` subdirectory, there is a single `Saanich_Med_Plus_SAGs.fasta` file which you will use with both `treesapp assign` and `treesapp update`. After the MAGs were generated with the SAG Assembly and Decontamination Workflow [@Bolger.2014; @Prjibelski.2020; @Parks.2015; @Tennessen.2016] (see [Methods]), sample IDs were added to the contig headers of each MAG produced from each SI072 sample depth. The individual MAGs were then concatenated into this file.

-  `seq2lineage_Tables`

    This directory contains two files that will be used for treesapp update. One file contains the lineage information from the MAGs, and the other contains information for the SAGs. These tables were generated by taking each separate MAG and SAG respectively and passing them through the Genome Taxonomic Database Toolkit classify workflow (see [Methods]). Then, the first two columns of the archaeal and bacterial summary files were taken and combined into each of these files.

## Methods

Water was collected from various depths at Saanich Inlet and filtered through a 0.22 µm Sterivex filter to collect biomass. Total genomic DNA and RNA were extracted from these filters. RNA was reversed transcribed into cDNA and both genomic DNA and cDNA were used in the construction of shotgun Illumina libraries. Sequencing data were generated on the Illumina HiSeq platform with **2x150bp** technology. For more in-depth sampling and sequencing methods, please see @Hawley.2017.

### Quality filtering of reads

The resulting reads were processed and quality controlled at the Joint Genome Institute using the IMG/M pipeline. 

Quality filtering was performed with Trimmomatic (v.0.35, [documentation](http://www.usadellab.org/cms/?page=trimmomatic)) @Bolger.2014 with default parameters except:

```
LEADING:3 
TRAILING:3 
SLIDINGWINDOW:4:15 
MINLEN:36
```

### Assembly

MEGAHIT (v.1.1.3, [documentation](https://github.com/voutcn/megahit/wiki/Assembly-Tips)) @Li.2015 assembled MAGs using with default parameters except for:

```
--k-min 27 
--k-step 10 
--min-contig-len 500
```
