<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>1 Exploring the microcosmos | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="1.1 Background We live in a world dominated by microbial life. Accounting for bacteria and archaea (prokaryotes) alone, an estimated 1030 cells occupy the biosphere. For most of Earth’s history,...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="1 Exploring the microcosmos | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="1.1 Background We live in a world dominated by microbial life. Accounting for bacteria and archaea (prokaryotes) alone, an estimated 1030 cells occupy the biosphere. For most of Earth’s history,...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="1 Exploring the microcosmos | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="1.1 Background We live in a world dominated by microbial life. Accounting for bacteria and archaea (prokaryotes) alone, an estimated 1030 cells occupy the biosphere. For most of Earth’s history,...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="active" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-golden-data.html"><span class="header-section-number">3</span> Acquiring golden data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">8</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">9</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">10</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="references.html"><span class="header-section-number">D</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="exploring-the-microcosmos" class="section level1">
<h1>
<span class="header-section-number">1</span> Exploring the microcosmos<a class="anchor" aria-label="anchor" href="#exploring-the-microcosmos"><i class="fas fa-link"></i></a>
</h1>
<div id="background" class="section level2">
<h2>
<span class="header-section-number">1.1</span> Background<a class="anchor" aria-label="anchor" href="#background"><i class="fas fa-link"></i></a>
</h2>
<p>We live in a world dominated by microbial life. Accounting for bacteria and archaea (prokaryotes) alone, an estimated 10<sup>30</sup> cells occupy the biosphere. For most of Earth’s history, this microcosmos has been driving matter and energy transformations through distributed networks of metabolite exchange resulting in global biogeochemical cycles and symbiotic interactions that create and sustain the conditions of multicellular existence, including our own.</p>
<p>Recent advances in high-throughput sequencing and mass spectrometry platforms have enabled us to develop more quantitative insights into these networks’ structure and function at the individual, population, and community levels of biological organization. The resulting data sets have revealed both incredible diversity and ecological resilience and provided an opportunity to develop predictive models and sustainable biotechnologies with the potential to transform the way humans interact with the natural world and one another.</p>
<p>Despite the power and the promise of this new perception, we exist dangerously close to upsetting the planetary boundaries that have promoted successful human expansion over the past 20,000 years. Global climate change is one manifestation of the cost we have paid in our pursuit of material and energy resources. Still, there are others, including biodiversity loss through habitat destruction, rampant pollution, and imbalances in integral element cycling of nitrogen and phosphorous.</p>
<p>Given the increasing availability of data sets describing microbial community structure and function in the world around us, we have an opportunity to learn from this information using bioinformatic tools and approaches. For this course, we will focus on one manifestation of climate change-related to dissolved oxygen (DO) loss in the ocean using a model ecosystem as a source of multi-omic (DNA and RNA) and geochemical parameter information over space and time.</p>
<p>Oxygen minimum zones (OMZs) are widespread areas of low DO in subsurface waters. Climate change resulting in increased stratification and reduced oxygen solubility in warming waters leads to OMZ expansion. Consequently, as oxygen levels decline, the microbial communities inhabiting OMZ waters shift their metabolisms to utilize alternative terminal electron acceptors. This shift results in the production of climate-active trace gases such as nitrous oxide (N<sub>2</sub>O) and methane (CH<sub>4</sub>).</p>
<p>Specifically, we will study the effects of changing oxygen levels on communities of bacteria and archaea inhabiting the Saanich Inlet water column. Saanich Inlet, situated on the coast of Vancouver Island, British Columbia, is a seasonally anoxic fjord that provides a biodynamic perspective on OMZ expansion. From a systems ecology perspective, OMZs provide useful environmental contexts in which to study coupled biogeochemical cycling through distributed metabolic networks, especially the integration of the carbon, nitrogen and sulphur cycles.</p>
<p>By combining multi-omics and geochemical parameter information from OMZs, it becomes possible to evaluate microbial communities’ regulatory and response dynamics to changing DO levels. For example, because OMZs are hotspots for nitrogen loss processes reconstructing the nitrogen cycle as a distributed metabolic process can shed new insight into microbial controls on matter and energy transformations integral to ocean health and climate balance. Genes encoding key steps in the nitrogen cycle are well defined (Figure <a href="exploring-the-microcosmos.html#fig:N-cycle-genes">1.1</a>), providing a basis for functional anchor screening to determine their distribution across the Tree of Life.</p>
<div class="figure">
<span style="display:block;" id="fig:N-cycle-genes"></span>
<img src="images/N_cycle_genes.png" alt="Nitrogen cycle" width="100%"><p class="caption">
FIGURE 1.1: Nitrogen cycle
</p>
</div>
</div>
<div id="using-multi-omic-data-to-reconstruct-distributed-networks-of-metabolite-exchange" class="section level2">
<h2>
<span class="header-section-number">1.2</span> Using multi-omic data to reconstruct distributed networks of metabolite exchange<a class="anchor" aria-label="anchor" href="#using-multi-omic-data-to-reconstruct-distributed-networks-of-metabolite-exchange"><i class="fas fa-link"></i></a>
</h2>
<p>The nitrogen cycle is just one aspect of the distributed metabolic network that defines microbial interaction space. Throughout this capstone project, you will have the opportunity to explore this space using a gene-centric approach. This approach is based on identifying and quantifying phylogenetic or functional anchor genes that represent key microbial players and metabolic pathways in the system under study. In a big picture sense, we want to use these data to answer three foundational questions concerning microbial community structure and function: Who is there? What are they doing? How do they respond to change?</p>
<p>Biological information flow can be described as a process of converting genotypic information encoded in the DNA base pairs of the genome transcribed into RNA messages that are translated into proteins at the level of the ribosome. These proteins, in turn, go on to catalyze metabolic transformations and regulatory dynamics that ultimately give rise to phenotypic expression, adaptation and response. This flow of information is formally referred to as the Central Dogma of Biology.</p>
<p>Genomic sequence information provides a basis for determining the types of microorganisms present in a sample and inferring metabolic potential at the individual, population and community levels of biological organization. Because DNA can persist in the environment for longer than RNA or protein, we can also use the abundance of gene X or gene Y to tell us about prior metabolic activity, e.g. gene abundance is a proxy for the process. We can use gene abundance information in numerical models to infer metabolic flux through a given pathway because the abundance of gene X is proportional to genome replication in which it resides. Genomic information is also more robust to sampling noise as we collect and process samples for downstream analysis. For information on gene-centric modelling incorporating DNA, RNA and protein information in Saanich Inlet, see this paper <span class="citation">(<a href="references.html#ref-Louca.2016" role="doc-biblioref">1</a>)</span>.</p>
<p>While genomic information tells us about metabolic potential, the transcriptome guides action or gene expression. To a certain extent, we can use this information to identify who is there, but more often, to indicate which pathways are active at the time of sampling. Transcriptomic data can open a window into response patterns along environmental gradients with the caveat that its short-lived nature is less robust to sampling noise. The presence of transcripts for a given gene X is no guarantee for their conversion into protein. However, because prokaryotes couple transcription to translation, we typically infer a direct relationship. For information on relationships between DNA, RNA and protein in Saanich Inlet, see this paper as well <span class="citation">(<a href="references.html#ref-Hawley.2014" role="doc-biblioref">2</a>)</span>.</p>
</div>
<div id="treesapp" class="section level2">
<h2>
<span class="header-section-number">1.3</span> TreeSAPP<a class="anchor" aria-label="anchor" href="#treesapp"><i class="fas fa-link"></i></a>
</h2>
<p>For this project, you will be implementing a software application called Tree-based Sensitive and Accurate Protein Profiler (TreeSAPP) <span class="citation">(<a href="references.html#ref-Morgan-Lang.2020" role="doc-biblioref">3</a>)</span>. Given the current set of reference packages available, you are in a position to reconstruct key steps in selected biogeochemical cycles, e.g. carbon, nitrogen or sulphur, along defined redox gradients in Saanich Inlet. TreeSAPP takes either metagenomic or metatranscriptomic reads and aligns them to reference packages representing phylogenetic or functional anchor genes of interest. These reference packages are constructed from trusted sources and consist of a phylogenetic tree, multiple sequence alignment, hidden Markov model and taxonomic lineage information. Reference packages can be updated and refined as new sequence information becomes available, including isolate reference genomes, metagenome-assembled genomes (MAGs) and single-cell amplified genomes (SAGs), thereby improving the precision of taxonomic placement. TreeSAPP outputs can be readily visualized using the interactive Tree of Life (iTOL) <span class="citation">(<a href="references.html#ref-Letunic.2019" role="doc-biblioref">4</a>)</span> or plotted in R to make beautiful and informative data visualization products.</p>

</div>
</div>



<div class="chapter-nav">
<div class="prev"><a href="index.html">Overview</a></div>
<div class="next"><a href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#exploring-the-microcosmos"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li><a class="nav-link" href="#background"><span class="header-section-number">1.1</span> Background</a></li>
<li><a class="nav-link" href="#using-multi-omic-data-to-reconstruct-distributed-networks-of-metabolite-exchange"><span class="header-section-number">1.2</span> Using multi-omic data to reconstruct distributed networks of metabolite exchange</a></li>
<li><a class="nav-link" href="#treesapp"><span class="header-section-number">1.3</span> TreeSAPP</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/introduction.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/introduction.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>
</div>
  

  

</div>
 <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on January 10, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
