<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>D Frequently Asked Questions | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="D.1 Why is TreeSAPP reporting an error where an argument is unrecognized? This could be due to several reasons. The most common is a spelling mistake in the argument name, like --fasta_input...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="D Frequently Asked Questions | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="D.1 Why is TreeSAPP reporting an error where an argument is unrecognized? This could be due to several reasons. The most common is a spelling mistake in the argument name, like --fasta_input...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="D Frequently Asked Questions | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="D.1 Why is TreeSAPP reporting an error where an argument is unrecognized? This could be due to several reasons. The most common is a spelling mistake in the argument name, like --fasta_input...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="active" href="frequently-asked-questions.html"><span class="header-section-number">D</span> Frequently Asked Questions</a></li>
<li><a class="" href="references.html"><span class="header-section-number">E</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="frequently-asked-questions" class="section level1" number="15">
<h1>
<span class="header-section-number">D</span> Frequently Asked Questions<a class="anchor" aria-label="anchor" href="#frequently-asked-questions"><i class="fas fa-link"></i></a>
</h1>
<div id="why-is-treesapp-reporting-an-error-where-an-argument-is-unrecognized" class="section level2" number="15.1">
<h2>
<span class="header-section-number">D.1</span> Why is TreeSAPP reporting an error where an argument is unrecognized?<a class="anchor" aria-label="anchor" href="#why-is-treesapp-reporting-an-error-where-an-argument-is-unrecognized"><i class="fas fa-link"></i></a>
</h2>
<p>This could be due to several reasons.
The most common is a spelling mistake in the argument name, like <code>--fasta_input</code> instead of the correct <code>--fastx_input</code>.</p>
<p>If the argument names all look correct you should check the position of backslashes at the end of lines if you are inputting a multi-line command.
There must be a space between a previous argument and a backslash for the command to be interpreted correctly.
For example, the following would fail with TreeSAPP complaining ‘output_directory’ is not recognized:</p>
<div class="sourceCode" id="cb143"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb143-1"><a href="frequently-asked-questions.html#cb143-1" aria-hidden="true" tabindex="-1"></a><span class="ex">treesapp</span> assign <span class="dt">\</span></span>
<span id="cb143-2"><a href="frequently-asked-questions.html#cb143-2" aria-hidden="true" tabindex="-1"></a>-i sample.fasta<span class="dt">\</span></span>
<span id="cb143-3"><a href="frequently-asked-questions.html#cb143-3" aria-hidden="true" tabindex="-1"></a>-o output_directory</span></code></pre></div>
</div>
<div id="do-i-need-to-run-commands-like-treesapp-assign-and-treesapp-abundance-for-each-reference-package-separately" class="section level2" number="15.2">
<h2>
<span class="header-section-number">D.2</span> Do I need to run commands like <code>treesapp assign</code> and <code>treesapp abundance</code> for each reference package separately?<a class="anchor" aria-label="anchor" href="#do-i-need-to-run-commands-like-treesapp-assign-and-treesapp-abundance-for-each-reference-package-separately"><i class="fas fa-link"></i></a>
</h2>
<p>Generally no, all the reference packages should be processed together.
This can be accomplished by copying all the reference packages you’re analyzing in a single directory and pointing the <code>treesapp assign</code> and any other commands to it with the <code>--refpkg_dir</code> option.
Note that commands that have the argument <code>--refpkg_path</code>, such as <code>treesapp update</code> and <code>treesapp colour</code>, DO need to be ran separately for each reference package.</p>
</div>
<div id="why-am-i-suddently-getting-a-no-space-left-on-device-error" class="section level2" number="15.3">
<h2>
<span class="header-section-number">D.3</span> Why am I suddently getting a <code>no space left on device</code> error?<a class="anchor" aria-label="anchor" href="#why-am-i-suddently-getting-a-no-space-left-on-device-error"><i class="fas fa-link"></i></a>
</h2>
<p>You have exhausted the allocated hard-disk space in your home directory.
There are tens of Terabytes of storage capacity under <code>/data</code>, so please move your directories there and proceed.</p>
</div>
<div id="why-is-treesapp-saying-file-already-exists" class="section level2" number="15.4">
<h2>
<span class="header-section-number">D.4</span> Why is TreeSAPP saying file already exists?<a class="anchor" aria-label="anchor" href="#why-is-treesapp-saying-file-already-exists"><i class="fas fa-link"></i></a>
</h2>
<p>This error crops up when the output directory already exists from a previous run.
You should include the <code>--overwrite</code> flag in your TreeSAPP command and try again.</p>
</div>
<div id="can-multiple-treesapp-abundance-processes-be-ran-simultaneously" class="section level2" number="15.5">
<h2>
<span class="header-section-number">D.5</span> Can multiple <code>treesapp abundance</code> processes be ran simultaneously?<a class="anchor" aria-label="anchor" href="#can-multiple-treesapp-abundance-processes-be-ran-simultaneously"><i class="fas fa-link"></i></a>
</h2>
<p>It depends. If they are accessing different <code>treesapp assign</code> output directories (i.e. different <code>--treesapp_output</code> values) then yes. Otherwise, no. This is because the intermediate files may be overwritten by another process causing at least one of the <code>treesapp abundance</code> processes to crash.</p>
</div>
<div id="no-sequences-were-found-in-the-sags-or-mags-when-running-treesapp-assign.-what-do-i-do-now" class="section level2" number="15.6">
<h2>
<span class="header-section-number">D.6</span> No sequences were found in the SAGs or MAGs when running <code>treesapp assign</code>. What do I do now?<a class="anchor" aria-label="anchor" href="#no-sequences-were-found-in-the-sags-or-mags-when-running-treesapp-assign.-what-do-i-do-now"><i class="fas fa-link"></i></a>
</h2>
<p>You can skip updating the reference package(s) with genomes (SAGs or MAGs) where homologs were not found.
There is nothing for <code>treesapp update</code> to add.</p>
<p>The SAGs and MAGs comprise a fraction of the community, and so it isn’t expected that every protein family we have reference packages for will be encoded in the SAGs and MAGs, even if it is known to be encoded by community members and found in a metagenome.
You have done nothing wrong, it’s just the data.</p>
</div>
<div id="what-do-i-need-to-know-about-the-error-clade-exclusion-analysis-could-not-be-performed-for-training-the-reference-package-models-from-treesapp-create" class="section level2" number="15.7">
<h2>
<span class="header-section-number">D.7</span> What do I need to know about the error “<code>Clade exclusion analysis could not be performed for training the reference package models</code>” from <code>treesapp create</code>?<a class="anchor" aria-label="anchor" href="#what-do-i-need-to-know-about-the-error-clade-exclusion-analysis-could-not-be-performed-for-training-the-reference-package-models-from-treesapp-create"><i class="fas fa-link"></i></a>
</h2>
<p>This error can arise when the reference package being created is very small or if the reference sequences lack species-level taxonomic labels. The reference package .pkl file can normally be used but the <code>treesapp assign</code> will be more prone to over-classification where the assigned taxonomic labels may be too resolved (closer to species) than they should be for their evolutionary distance.</p>
</div>
<div id="is-it-okay-if-treesapp-purity-fails-because-no-alignments-were-found" class="section level2" number="15.8">
<h2>
<span class="header-section-number">D.8</span> Is it okay if <code>treesapp purity</code> fails because no alignments were found?<a class="anchor" aria-label="anchor" href="#is-it-okay-if-treesapp-purity-fails-because-no-alignments-were-found"><i class="fas fa-link"></i></a>
</h2>
<p>Yes, this is fine. The TIGRFAM seed reference database used is small and not comprehensive. Therefore, it is entirely expected that reference packages for many protein families will not be able to recruit alignments. Still, this result also indicates that the reference package doesn’t contain misclassified sequences related to orthologs in the database, which is comforting to know.</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></div>
<div class="next"><a href="references.html"><span class="header-section-number">E</span> References</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#frequently-asked-questions"><span class="header-section-number">D</span> Frequently Asked Questions</a></li>
<li><a class="nav-link" href="#why-is-treesapp-reporting-an-error-where-an-argument-is-unrecognized"><span class="header-section-number">D.1</span> Why is TreeSAPP reporting an error where an argument is unrecognized?</a></li>
<li><a class="nav-link" href="#do-i-need-to-run-commands-like-treesapp-assign-and-treesapp-abundance-for-each-reference-package-separately"><span class="header-section-number">D.2</span> Do I need to run commands like treesapp assign and treesapp abundance for each reference package separately?</a></li>
<li><a class="nav-link" href="#why-am-i-suddently-getting-a-no-space-left-on-device-error"><span class="header-section-number">D.3</span> Why am I suddently getting a no space left on device error?</a></li>
<li><a class="nav-link" href="#why-is-treesapp-saying-file-already-exists"><span class="header-section-number">D.4</span> Why is TreeSAPP saying file already exists?</a></li>
<li><a class="nav-link" href="#can-multiple-treesapp-abundance-processes-be-ran-simultaneously"><span class="header-section-number">D.5</span> Can multiple treesapp abundance processes be ran simultaneously?</a></li>
<li><a class="nav-link" href="#no-sequences-were-found-in-the-sags-or-mags-when-running-treesapp-assign.-what-do-i-do-now"><span class="header-section-number">D.6</span> No sequences were found in the SAGs or MAGs when running treesapp assign. What do I do now?</a></li>
<li><a class="nav-link" href="#what-do-i-need-to-know-about-the-error-clade-exclusion-analysis-could-not-be-performed-for-training-the-reference-package-models-from-treesapp-create"><span class="header-section-number">D.7</span> What do I need to know about the error “Clade exclusion analysis could not be performed for training the reference package models” from treesapp create?</a></li>
<li><a class="nav-link" href="#is-it-okay-if-treesapp-purity-fails-because-no-alignments-were-found"><span class="header-section-number">D.8</span> Is it okay if treesapp purity fails because no alignments were found?</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/faq.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/faq.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on April 27, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
