# Project description

In this project, you will work in teams to explore marine microbial communities and geochemical pathways, for example the nitrogen cycle in general or **denitrification** in particular. You will use metagenomic and metatranscriptomic data from Cruise 72 at 7 depths in [Saanich Inlet](#the-saanich-inlet-data-set), a seasonally anoxic fjord that serves as a model ecosystem for studying microbial community responses to changing levels of oxygen. Each group will explore a group of genes with TreeSAPP to assess microbial communities in terms of taxonomic rank, abundance, and expression along the redoxcline in Saanich Inlet.

## Guiding research questions

You will need to decide on the scope and focus of your project:

1.  Select your genes for analysis based on one of the following options:

    a.  Choose a pathway of a geochemical cycle and its associated genes with reference packages already available for TreeSAPP (see table).

    b.  Choose a pathway of a geochemical cycle and its associated genes which do NOT have reference packages available for TreeSAPP. As part of your project, you will create those reference packages.

    c.  Look at all genes available as reference packages for TreeSAPP without a focus on any particular pathway.

2.  Perform a preliminary analysis the Saanich Inlet data for all depths.

3.  Update ref packages based on preliminary analysis.

4.  Look for pattern in abundance of genes across water column.

### Questions for project with a focus on geochemical pathways (repeated in report structure below)

1. How does abundance of those genes differ across the pathway? Are trends similar for both RNA and DNA?

2. How does microbial diversity differ across the pathway? Are trends similar for both RNA and DNA?

3. What specific taxa are responsible for the geochemical cycle under investigation? Are they the same for all steps? For DNA versus RNA?

4. How does the abundance of those genes relate to nitrogen species in Saanich (use the geochemical data in Saanich_Data.csv from our previous data science sessions)?

### Questions for project with a focus on using all reference packages

Add questions.

## Resources

You will be provided with a script template for both the shell and R portion of your analysis (`treesapp_analysis.sh` and `treesapp_analysis.R`) that will guide you as you develop your code.

## Your submission

Your final submission will consist of 3 separate files: the report itself (`docx` or `pdf`), one shell script `treesapp_analysis.sh`, and one R script `treesapp_analysis.R` (both script files must be in plain text format). The report should not contain any code, but should contain versions of software tools used and a high-level description of your workflow (i.e describe *what* was done and NOT *how*).

## Timeline

The following provides an outline as well as some specific milestones within the project.

```{r child = "child_Rmds/timeline.Rmd"}
```

## Reports

Reports should be formatted as per the [Instructions to Authors](https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf) for the [Journal of Bacteriology](https://jb.asm.org/).

Each group will submit **one** report with the sections below.

```{r child = "child_Rmds/report_structure.Rmd"}
```

