# Project description

In this project, you will work in teams to explore marine microbial communities and the nitrogen cycle, particularly **denitrification**. You will use metagenomic and metatranscriptomic data from Cruise 72 at 7 depths in Saanich Inlet, a seasonally anoxic fjord that serves as a model ecosystem for studying microbial community responses to changing levels of oxygen. Each group has been assigned a specific depth (given in Canvas group name) and will assess microbial communities in terms of taxonomic rank, abundance, and expression along the redoxcline in Saanich Inlet.

## Timeline

The following provides an outline as well as some specific milestones within the project.

1.  Mar 27: Introduction and begin running TreeSAPP

    -   Ideally, your GCP analyses should run over the weekend so that you have data to work with in class on Monday.

2.  Mar 30: Introduction to statistics; Group work

3.  Apr 1 and 3: Group work

4.  Apr 6: Project synthesis

    -   Be prepared to discuss the main conclusions that you've reached for your assigned depth.

5.  (April 8: Course recap and discussion)

6.  April 9-24: Report writing.

    -   Groups are expected to meet remotely as needed over the Finals Period in order to complete the report. This report serves as a final for this course and should be treated as such.

7.  April 24: Final reports due with final portfolios

## Reports

Reports should be formatted as per the [Instructions to Authors](https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf) for the [Journal of Bacteriology](https://jb.asm.org/).

Each group will complete **one** report with the following sections.

### Abstract

*200--250 words*

Note that an Importance section is not required.

### Introduction

*500--750 words*

-   Overview of the nitrogen cycle including its global impacts and microbial foundations.
-   Introduce Saanich Inlet as a model ecosystem for studying microbial community responses to ocean deoxygenation *e.g.* seasonal cycles, relevant biogeochemistry, previous studies, etc.

### Methods

*300--500 words*

-   Briefly describe the data (sampling, sequencing, processing, etc.)

-   Briefly describe your analysis methods including

    -   TreeSAPP version and commands used
    -   iTOL version
    -   R version and packages used
    -   Statistics (if applicable)

-   Provide one single shell script and one single R script (i.e "treesapp_analysis.sh" and "treesapp_analysis.R") as individual files (i.e. not as part of your manuscript) containing the complete code to generate your results.

### Results {#research-questions}

*500--750 words*

Your analysis will focus on **denitrification genes at your assigned depth** (which is given in your Canvas group name) and the following questions:

1.  How does abundance of denitrification genes differ across the pathway? Are trends similar for both RNA and DNA?
2.  How does microbial diversity differ across the pathway? Are trends similar for both RNA and DNA?
3.  What specific taxa are responsible for denitrification? Are they the same for all steps? For DNA versus RNA?
4.  How does the abundance of denitrification genes relate to nitrogen species in Saanich (use the geochemical data in `Saanich_Data.csv` from our previous data science sessions)?

You must include â‰¥ 5 figures/panels with titles and full captions. These figures can be combined into multi-panel figures if desired.

### Discussion

*750--1000 words*

-   Frame your depth's results within a broader discussion of Saanich Inlet and the other depths (Apr 6 discussion)
-   Propose evolutionary, environmental, etc. reasoning for distributed metabolism as seen in the denitrification pathway
-   Future directions

### References

*10 or more* formatted in the ASM style such as for the [Journal of Bacteriology](https://jb.asm.org/content/organization-and-format). If you are using a reference manager, this style can be downloaded for [EndNote](https://endnote.com/style_download/american-society-for-microbiology-asm-journals-2/), [Mendeley](https://github.com/citation-style-language/styles/blob/master/american-society-for-microbiology.csl), or [Zotero](https://www.zotero.org/styles?q=microbiology).

Make sure to cite the data source papers!
