<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>8 Project description | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="In this project, you will work in groups to conduct gene-centric mapping of functional and phylogenetic anchors encoded in microbial genomes sourced from the Saanich Inlet water column. Saanich...">
<meta name="generator" content="bookdown 0.23 with bs4_book()">
<meta property="og:title" content="8 Project description | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="In this project, you will work in groups to conduct gene-centric mapping of functional and phylogenetic anchors encoded in microbial genomes sourced from the Saanich Inlet water column. Saanich...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="8 Project description | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="In this project, you will work in groups to conduct gene-centric mapping of functional and phylogenetic anchors encoded in microbial genomes sourced from the Saanich Inlet water column. Saanich...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.5.1/tabs.js"></script><script src="libs/bs3compat-0.2.5.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="libs/kePrint-0.0.1/kePrint.js"></script><link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-golden-data.html"><span class="header-section-number">3</span> Acquiring golden data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="active" href="project-description.html"><span class="header-section-number">8</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">9</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">10</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="references.html"><span class="header-section-number">D</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="project-description" class="section level1">
<h1>
<span class="header-section-number">8</span> Project description<a class="anchor" aria-label="anchor" href="#project-description"><i class="fas fa-link"></i></a>
</h1>
<p>In this project, you will work in groups to conduct gene-centric mapping of functional and phylogenetic anchors encoded in microbial genomes sourced from the Saanich Inlet water column. Saanich Inlet is a seasonally anoxic fjord on the coast of Vancouver Island British Columbia that serves as a model ecosystem for studying microbial community responses to ocean deoxygenation.</p>
<p>Numerous studies from Saanich Inlet over the years have identified key microbial players mediating coupled biogeochemical cycling of carbon, nitrogen and sulfur extensible to expanding marine oxygen minimum zones throughout the global ocean. Here, you will use metagenomic and metatranscriptomic data sets generated from Cruise 72 spanning 7 depths in <a href="the-saanich-inlet-data-set.html#the-saanich-inlet-data-set">Saanich Inlet</a>. In addition to fastq reads, the metagenomic data sets include assembled contigs, metagenome assembled genomes (MAGs) and single-cell amplified genomes (SAGs) spanning the genomic information hierarchy.</p>
<p>The Cruise 72 metagenomic and metatranscriptomic data sets can be evaluated in relation to geochemical parameter information to better resolve both the metabolic potential and gene expression patterns of microbial communities inhabiting the Saanich Inlet water column. Each capstone project group will have the opportunity to use TreeSAPP and iToL to chart the abundance, expression and taxonomic diversity of functional and phylogenetic anchor genes represented in the current reference package collection. Groups can augment this collection with new reference packages produced during the TreeSAPP tutorial or at any time during the project phase depending on their research interests. Reference packages can also be updated using taxonomic information associated with the MAGs and SAGs described above.</p>
<p>The underlying conceptual framework for this project is described in the three lectures associated with course Module 2 along with examples of data visualization approaches useful in developing a coherent scientific narrative. For a given gene encoding a biochemical transformation consider the extent to which this functionality is distributed within the community and how it is represented at different levels of biological information flow e.g. DNA versus RNA.</p>
<div id="guiding-research-questions" class="section level2">
<h2>
<span class="header-section-number">8.1</span> Guiding research questions<a class="anchor" aria-label="anchor" href="#guiding-research-questions"><i class="fas fa-link"></i></a>
</h2>
<p>Several options are provided to help you develop your capstone project, although each group is free to come up with an original plan of action. Please consult with the teaching team if you have questions.</p>
<ol style="list-style-type: decimal">
<li>
<p>Select your reference package(s) for analysis based on one of the following options:</p>
<ol style="list-style-type: lower-alpha">
<li><p>Choose one or more pathways of a geochemical cycle e.g. nitrogen with reference packages already available for TreeSAPP (see table of available <a href="treesapp-reference-packages.html#treesapp-reference-packages">TreeSAPP reference packages</a>). For example, you could select NapA, NirK, NirS, NorB, NorC and NosZ to investigate denitrification in the Saanich Inlet water column.</p></li>
<li><p>Create new reference packages to expand the TreeSAPP collection and analyze the results. You could select a pthway within a biogeocehmical cycle which does NOT have reference packages available covering the complete pathway e.g. sulfur cycle including sulfur oxidation and DMSP conversion, or a completely new pathway with no reference packages avaialble.</p></li>
<li><p>Select all reference packages in the collection including those developed by the class during the treesapp create tutorial and map the results. Focus on evaluating the purity of each reference package and update as needed to reflect diversity of genes endemic to the Saanich Inlet water column.</p></li>
</ol>
</li>
<li><p>Perform a preliminary analysis using the Saanich Inlet data for all depths.</p></li>
<li><p>Update reference packages as needed based on preliminary analysis.</p></li>
<li><p>Look for patterns in gene abundance, expression and diversity as a function of water column geochemical parameter information.</p></li>
</ol>
<div id="questions-for-project-with-a-focus-on-biogeochemical-cycles-repeated-in-report-structure-below" class="section level3">
<h3>
<span class="header-section-number">8.1.1</span> Questions for project with a focus on biogeochemical cycles (repeated in report structure below)<a class="anchor" aria-label="anchor" href="#questions-for-project-with-a-focus-on-biogeochemical-cycles-repeated-in-report-structure-below"><i class="fas fa-link"></i></a>
</h3>
<ol style="list-style-type: decimal">
<li><p>How does gene and transcript abundance vary across depths for each reference package for a given biogeochemical cycle?</p></li>
<li><p>How does microbial diversity differ among and between steps within a pathway e.g. denitrification? Are these trends similar for both genes and transcripts at different depths?</p></li>
<li><p>What is the taxonomic breakdown of genes and transcripts within a given biogeochemcial cycle? Can you discern evidence for horizontal gene transfer of one or more functional anchor genes?</p></li>
<li><p>Can you identify evidence for distributed metabolism within one or more pathways? Are these trends similar for both genes and transcripts at different depths?</p></li>
<li><p>How do answers to questions 2 and 3 vary depending on the taxonomic rank used in the analysis?</p></li>
<li><p>How does the abundance, expression and taxonomy of identified genes relate to water column geochemical parameter information (use the geochemical data in Saanich_Data.csv from our previous data science sessions)?</p></li>
</ol>
</div>
<div id="questions-for-project-with-a-focus-on-using-all-reference-packages" class="section level3">
<h3>
<span class="header-section-number">8.1.2</span> Questions for project with a focus on using all reference packages<a class="anchor" aria-label="anchor" href="#questions-for-project-with-a-focus-on-using-all-reference-packages"><i class="fas fa-link"></i></a>
</h3>
<p>In addition to the questions provided above for biogeochemical cycles consider quality control metrics in your analysis.</p>
<ol style="list-style-type: decimal">
<li><p>How does the purity of each reference package impact your results?</p></li>
<li><p>How much information is discarded due to insufficient taxonomic resolution?</p></li>
<li><p>How much information is retained after updating reference packages with MAGs or SAGs?</p></li>
<li><p>What is the impact on diversity metrics following the use of updated reference packages?</p></li>
</ol>
</div>
</div>
<div id="resources" class="section level2">
<h2>
<span class="header-section-number">8.2</span> Resources<a class="anchor" aria-label="anchor" href="#resources"><i class="fas fa-link"></i></a>
</h2>
<p>You will be provided with a script template for both the shell and R portion of your analysis (<code>treesapp_analysis.sh</code> and <code>treesapp_analysis.R</code>) that will guide you as you develop your code.</p>
<p>As stated in the project description, you will have access to metagenomic and metatranscriptomic data sets generated from Cruise 72 spanning 7 depths in <a href="the-saanich-inlet-data-set.html#the-saanich-inlet-data-set">Saanich Inlet</a>, MAGs and SAGs.</p>
</div>
<div id="your-submission" class="section level2">
<h2>
<span class="header-section-number">8.3</span> Your submission<a class="anchor" aria-label="anchor" href="#your-submission"><i class="fas fa-link"></i></a>
</h2>
<p>Your final submission will consist of 3 separate files: the report itself (<code>docx</code> or <code>pdf</code>), one shell script <code>treesapp_analysis.sh</code>, and one R script <code>treesapp_analysis.R</code> (both script files must be in plain text format). The report should not contain any code, but should contain versions of software tools used and a high-level description of your workflow (i.e describe <em>what</em> was done and NOT <em>how</em>).</p>
</div>
<div id="timeline" class="section level2">
<h2>
<span class="header-section-number">8.4</span> Timeline<a class="anchor" aria-label="anchor" href="#timeline"><i class="fas fa-link"></i></a>
</h2>
<p>The following provides an outline as well as some specific milestones within the project.</p>
<!-- Table for capstone timeline, edit with RStudio visual editor -->
<div class="inline-table"><table class="table table-sm">
<caption>Capstone timeline</caption>
<colgroup>
<col width="7%">
<col width="92%">
</colgroup>
<thead><tr class="header">
<th align="left">Date</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Mar 29</td>
<td align="left">
<p>Introduction and begin running TreeSAPP</p>
<p>Ideally, your analyses on the server should run over night</p>
</td>
</tr>
<tr class="even">
<td align="left">Mar 31</td>
<td align="left">
<p><strong>Start of capstone project</strong></p>
<p>Group work</p>
</td>
</tr>
<tr class="odd">
<td align="left">Apr 7, 9, 12</td>
<td align="left">Group work</td>
</tr>
<tr class="even">
<td align="left">Apr 14</td>
<td align="left">Course recap and discussion</td>
</tr>
<tr class="odd">
<td align="left">April 14–24</td>
<td align="left">
<p><strong>Report writing</strong></p>
<p>Groups are expected to meet remotely as needed over the Finals Period in order to complete the report. This report serves as a final for this course and should be treated as such.</p>
</td>
</tr>
<tr class="even">
<td align="left">April 24</td>
<td align="left">
<p><strong>Final due date</strong></p>
<p>For reports and completed portfolios</p>
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="reports" class="section level2">
<h2>
<span class="header-section-number">8.5</span> Reports<a class="anchor" aria-label="anchor" href="#reports"><i class="fas fa-link"></i></a>
</h2>
<p>Reports should be formatted as per the <a href="https://jb.asm.org/sites/default/files/additional-assets/JB-ITA.pdf">Instructions to Authors</a> for the <a href="https://jb.asm.org/">Journal of Bacteriology</a>.</p>
<p>Each group will submit <strong>one</strong> report with the sections below.</p>
<!-- Report structure for capstone, edit with RStudio visual editor -->
<div class="inline-table"><table class="table table-sm">
<caption>Report structure</caption>
<colgroup>
<col width="4%">
<col width="95%">
</colgroup>
<thead><tr class="header">
<th>Section</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>
<p><strong>Abstract</strong></p>
<p><em>200–250 words</em></p>
</td>
<td>Note that an Importance section is not required.</td>
</tr>
<tr class="even">
<td>
<p><strong>Introduction</strong></p>
<p><em>500–750 words</em></p>
</td>
<td>
<p>Introduce Saanich Inlet as a model ecosystem for studying microbial community responses to ocean deoxygenation e.g. seasonal cycles, relevant biogeochemistry, previous studies, etc.</p>
<p>Overview of a geochemical cycle including its global impacts, microbial foundations and involved genes.</p>
</td>
</tr>
<tr class="odd">
<td>
<p><strong>Methods</strong></p>
<p><em>300–500 words</em></p>
</td>
<td>
<p>Briefly describe the data (sampling, sequencing, processing, etc.)</p>
<p>Briefly describe your analysis methods including</p>
<ul>
<li><p>TreeSAPP version and commands used</p></li>
<li><p>iTOL version</p></li>
<li><p>R version and packages used</p></li>
<li><p>Statistics (if applicable)</p></li>
</ul>
<p>Provide one single shell script and one single R script (i.e <code>treesapp_analysis.sh</code> and <code>treesapp_analysis.R</code>) as individual files (i.e. not as part of your manuscript) containing the complete code to generate your results.</p>
</td>
</tr>
<tr class="even">
<td>
<p><strong>Results</strong></p>
<p><em>500–750 words</em></p>
</td>
<td>
<p>Your analysis should address the <a href="project-description.html#guiding-research-questions">guiding research questions</a> you have developed.</p>
<p>You must include ≥ 5 figures/panels with titles and full captions. These figures can be combined into multi-panel figures if desired.</p>
</td>
</tr>
<tr class="odd">
<td>
<p><strong>Discussion</strong></p>
<p><em>750–1000 words</em></p>
</td>
<td>
<p>Frame results within a broader discussion of Saanich Inlet (Apr 14 discussion).</p>
<p>Propose evolutionary, environmental, etc. reasoning for distributed metabolism as seen in the geochemcial pathway.</p>
<p>Future directions</p>
</td>
</tr>
<tr class="even">
<td>
<p><strong>References</strong></p>
<p><em>10 or more</em></p>
</td>
<td>
<p>Formatted in the ASM style such as for the <a href="https://jb.asm.org/content/organization-and-format">Journal of Bacteriology</a>. If you are using a reference manager, this style can be downloaded for <a href="https://endnote.com/style_download/american-society-for-microbiology-asm-journals-2/">EndNote</a>, <a href="https://github.com/citation-style-language/styles/blob/master/american-society-for-microbiology.csl">Mendeley</a>, or <a href="https://www.zotero.org/styles?q=microbiology">Zotero</a>.</p>
<p>Make sure to include citations for the data source papers and software tools used!</p>
</td>
</tr>
</tbody>
</table></div>
</div>
<div id="addendum" class="section level2">
<h2>
<span class="header-section-number">8.6</span> Addendum<a class="anchor" aria-label="anchor" href="#addendum"><i class="fas fa-link"></i></a>
</h2>
<div id="abundance" class="section level3">
<h3>
<span class="header-section-number">8.6.1</span> Abundance<a class="anchor" aria-label="anchor" href="#abundance"><i class="fas fa-link"></i></a>
</h3>
<p>The workflow introduced in the <a href="#treesapp-abundance">treesapp abundance</a> tutorial was supposed to serve as the guide to generating TPM values for your capstone project. However, it has come to the attention of the teaching team that <code>treesapp abundance</code> fails to generate the simple_bar.txt files for viewing the distribution of TPM values across a reference package’s phylogenetic tree. Therefore, we are providing you another workflow that will generate both the classifications file (marker_contig_map.tsv) with TPM values and all simplebar.txt files for iTOL.</p>
<div id="generate-the-jplace-with-all-classifications" class="section level4">
<h4>
<span class="header-section-number">8.6.1.1</span> Generate the JPlace with all classifications<a class="anchor" aria-label="anchor" href="#generate-the-jplace-with-all-classifications"><i class="fas fa-link"></i></a>
</h4>
<p>With the concatenated metagenome FASTA file for SI072 (<code>MetaG_Assemblies/SI072_All_contigs.fa</code>; see <a href="#the-saanich-inlet-data-set#data-on-the-server">server data locations</a>) run <code>treesapp assign</code> without FASTQ reads and abundance flags to generate a single jplace file containing all reference package homologs across all the depths.</p>
</div>
<div id="generate-tpm-values-for-metagenomes" class="section level4">
<h4>
<span class="header-section-number">8.6.1.2</span> Generate TPM values for metagenomes<a class="anchor" aria-label="anchor" href="#generate-tpm-values-for-metagenomes"><i class="fas fa-link"></i></a>
</h4>
<p>Run <code>treesapp assign</code> seven times with abundance flags (--rel_abund and --metric TPM) and paths to metagenome FASTQ files to generate TPM values for each individual depth and the abundance files needed for iToL visualization.</p>
<p>This gives you a total of 8 <code>treesapp assign</code> runs with associated output files for the metagenome samples.</p>
</div>
<div id="generate-tpm-values-for-metatranscriptomes" class="section level4">
<h4>
<span class="header-section-number">8.6.1.3</span> Generate TPM values for metatranscriptomes<a class="anchor" aria-label="anchor" href="#generate-tpm-values-for-metatranscriptomes"><i class="fas fa-link"></i></a>
</h4>
<p>Run <code>treesapp assign</code> with abundance flags once for each of the seven metatranscriptome samples, including their respective FASTQ file path in the command. This will give you another seven <code>treesapp assign</code> outputs for a total of 15 to complete the process for metagenome and metatranscriptome data sets.</p>
</div>
<div id="generate-the-file-with-all-classifications-and-tpm-values" class="section level4">
<h4>
<span class="header-section-number">8.6.1.4</span> Generate the file with all classifications and TPM values<a class="anchor" aria-label="anchor" href="#generate-the-file-with-all-classifications-and-tpm-values"><i class="fas fa-link"></i></a>
</h4>
<p>Now concatenate the marker_contig_map.tsv files from the seven individual depth runs for analysis in <code>R</code>.</p>
<p>You will first need to write the header from <em>one</em> classification table to the combined table file (<code>SI072_combined_marker_contig_map.tsv</code> below), then write the classifications from all tables without the header. This is to avoid including the table’s header multiple times partway through the table, which would confuse <code>R</code>. We’re using <code>grep</code> to accomplish this by removing any lines that match a word in the header, ‘Query’.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb58-1"><a href="project-description.html#cb58-1"></a><span class="fu">head</span> -n 1 <span class="op">&lt;</span>path to treesapp assign output directory<span class="op">&gt;</span>/final_outputs/marker_contig_map.tsv <span class="op">&gt;</span>SI072_combined_marker_contig_map.tsv</span>
<span id="cb58-2"><a href="project-description.html#cb58-2"></a><span class="fu">cat</span> <span class="op">&lt;</span>path to assign outputs<span class="op">&gt;</span>/*/final_outputs/marker_contig_map.tsv <span class="kw">|</span> <span class="fu">grep</span> -v Query <span class="op">&gt;&gt;</span>SI072_combined_marker_contig_map.tsv</span></code></pre></div>
</div>
<div id="viewing-the-placements-and-tpm-bars-in-itol" class="section level4">
<h4>
<span class="header-section-number">8.6.1.5</span> Viewing the placements and TPM bars in iTOL<a class="anchor" aria-label="anchor" href="#viewing-the-placements-and-tpm-bars-in-itol"><i class="fas fa-link"></i></a>
</h4>
<p>For iToL visualization use the jplace file from <code>treesapp assign</code> with the concatenated metagenome FASTA to visualize the phylogenetic tree with all placements across depths, and use the <gene>_labels.txt file from the concatenated run to name the leaves.
To add the TPM abundance layers for each depth access the simple_bar.txt files from each individual depth run and drag and drop these files into the iToL GUI.</gene></p>

</div>
</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></div>
<div class="next"><a href="assessment.html"><span class="header-section-number">9</span> Assessment</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#project-description"><span class="header-section-number">8</span> Project description</a></li>
<li>
<a class="nav-link" href="#guiding-research-questions"><span class="header-section-number">8.1</span> Guiding research questions</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#questions-for-project-with-a-focus-on-biogeochemical-cycles-repeated-in-report-structure-below"><span class="header-section-number">8.1.1</span> Questions for project with a focus on biogeochemical cycles (repeated in report structure below)</a></li>
<li><a class="nav-link" href="#questions-for-project-with-a-focus-on-using-all-reference-packages"><span class="header-section-number">8.1.2</span> Questions for project with a focus on using all reference packages</a></li>
</ul>
</li>
<li><a class="nav-link" href="#resources"><span class="header-section-number">8.2</span> Resources</a></li>
<li><a class="nav-link" href="#your-submission"><span class="header-section-number">8.3</span> Your submission</a></li>
<li><a class="nav-link" href="#timeline"><span class="header-section-number">8.4</span> Timeline</a></li>
<li><a class="nav-link" href="#reports"><span class="header-section-number">8.5</span> Reports</a></li>
<li>
<a class="nav-link" href="#addendum"><span class="header-section-number">8.6</span> Addendum</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#abundance"><span class="header-section-number">8.6.1</span> Abundance</a></li></ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/instructions.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/instructions.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on December 20, 2021.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>
</body>
</html>
