# Analysing the TreeSAPP output in R

```{r setup-R, echo = FALSE, warning = FALSE, message = FALSE}
knitr::opts_chunk$set(eval = FALSE)
```

These instructions will help you to complete the script template `treesapp_analysis.R` to complete the initial analysis of your TreeSAPP output. To adjust the code examples below and in the script template, replace any angle brackets, e.g. `<SOME TEXT>`, with the missing code.

## Resources for analysis of TreeSAPP output in R

-   Both TreeSAPP output files `marker_contig_map.tsv` (one each for the metagenomic and metatranscriptomic analysis) you generated with [TreeSAPP](#running-treesapp on a server).
-   An R script template called `treesapp_analysis.R` on Canvas.
-   `Saanich_Data.csv` file containing geochemical measurements.

## Checklist to write and run R script

-   [ ] Place `treesapp_analysis.R` and both `marker_contig_map.tsv` files into a single folder and create a new RStudio project in that folder on your local computer.
-   [ ] Edit the `treesapp_analysis.R` script following [the instructions below](#writing-the-r-script).

## Writing the R script

In the `treesapp_analysis.R` script you will load and subset your TreeSAPP data to the variables and marker genes of interest. You will then combine the metagenomic and metatranscriptomic data into a single data frame and then break the taxonomic information into taxonomic ranks. You will also load the `Saanich_Data.csv` file for geochemical measurements to learn more about the conditions at your assigned depth. Further process your data sets to address [the four main research questions](#results).

## Load any required packages

```{r}
<Load packages>
```

## Load metagenomic and metatranscriptomic TreeSAPP data

```{r}
<Load data sets>
```

## Combine the metagenomic and metatranscriptomic data

Look at the documentation of `rbind()` on how to combine two data sets that contain the same variables. Before you do that, you should add a variable that tells you if a particular row comes from the metagenomic or metatranscriptomic source.

```{r}
<Add variable to data frame indicating source>
<Combine data sets>
```


## Plotting abundances in iTOL
Drag and drop `XmoA_abundance_simplebar.txt` in the /final_outputs/iTOL_output folder into iTOL window to add abundance bar plot to ITOL phylogenic trees

### Classification table

Back on the server, let's look at TreeSAPP's classification table for these FunGene XmoA sequences.
 
The classification table contains detailed information corresponding to each classified query sequence. It is a tab-delimited table that can be read by a variety of tools. I tend to analyze most data in R and so the classification table was designed to be compatible with the tidyverse's extensive toolset. Likewise, it should be similarly easy to use Python to analyze these data with its rich suite of visualization libraries, e.g., `plotly` and `matplotlib`.

Here are the first few lines from `p_amoA_FunGene9.5_isolates_assign/final_outputs/marker_contig_map.tsv`:

```
Sample	Query	Marker	Start_pos	End_pos	Taxonomy	Abundance	iNode	E-value	LWR	EvoDist	Distances
p_amoA_FunGene9.5_isolates_C65S300_uclust99	AAL86638  coded_by=1..825,organism=Nitrosomonas sp. JL21,definition=ammonia monooxygenase subunit A	XmoA	4	251	r__Root; d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Nitrosomonadaceae; g__Nitrosomonas; s__Nitrosomonas sp. Is79A3	1.0	13	1.4e-114	0.9234534156	0.0404	0.0227,0.0177,0.0
p_amoA_FunGene9.5_isolates_C65S300_uclust99	AAG60667  coded_by=824..1645,organism=Nitrosomonas cryotolerans ATCC 49181,definition=ammonia monooxygenase subunit A	XmoA	4	251	r__Root; d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Nitrosomonadaceae	1.0	16	3.7e-114	1.0	0.1305	0.0128,0.0205,0.0972
p_amoA_FunGene9.5_isolates_C65S300_uclust99	BAH22839  coded_by=2160..2948,organism=Methylococcaceae bacterium ET-HIRO,definition=methane monooxygenase protein A2	XmoA	2	253	r__Root	1.0	61	7.1e-104	1.0	1.3677	0.083,0.5314,0.7534
```

A description of each of the fields can be found under the [`treesapp assign` documentation page](https://github.com/hallamlab/TreeSAPP/wiki/Classifying-sequences-with-treesapp-assign#classification-table).


## Analyze the classifications

Modify marker_contig_map.tsv to split the sample name into 'cruise' and 'depth'. Two new columns are added 'Length'(End_pos-Start_pos) and 'Depth.m'(Depth in meters). Taxonomy is split into a separate column for each level.

```{r setup, eval=F}
library(ggplot2)
library(dplyr)
library(tidyr)

dat <- read.table(file="data/marker_contig_map.tsv",
                  header=TRUE,
                  sep="\t", quote="") %>% 
  mutate(Length=End_pos-Start_pos) %>% 
  separate(col = Taxonomy, into = c('r', 'd', 'p', 'c', 'o', 'f', 'g', 's'), sep = "; ", fill = "right", remove = T) %>% 
  separate(col = Sample, into = c("Cruise", "Depth"), extra = "drop") %>% 
  mutate(Depth.m = as.numeric(gsub('m', '', Depth)))
```

Distribution of abundances of  XmoA taxonomic orders  at different depths

```{r, eval=F}
dat %>%
  group_by(Depth.m, o) %>%
  summarise(sum = sum(Abundance)) %>%
  ungroup() %>%
  ggplot(aes(x=Depth.m, y=sum, colour=o)) +
  geom_line() +
  labs(x="Depth (m)",
       y="Relative abundance (FPKM)")
```
Distribution of abundances of XmoA taxonomic families  at different depths

```{r, echo=FALSE, eval=F}

dat %>%
  group_by(Depth.m, f) %>%
  summarise(sum = sum(Abundance)) %>%
  ungroup() %>%
  ggplot(aes(x=Depth.m, y=sum, colour=f)) +
  geom_line() +coord_flip() +
  labs(x="Relative abundance (FPKM)",
       y="Depth (m)")
```

Barplot demonstrates relative change in proportions at different depths

```{r, eval=F}
dat %>% 
  group_by(Depth, Depth.m) %>% 
  mutate(Proportion = Abundance/sum(Abundance)) %>% 
  group_by(Depth, Depth.m, o) %>% 
  summarise(sum = sum(Proportion)) %>% 
  ungroup() %>% 
  mutate(Depth = reorder(Depth, Depth.m)) %>% 
  ggplot(aes(x=Depth, y=sum, fill=o)) +
  geom_bar(stat = "identity") +
  labs(x="Depth",
       y="Relative abundance")

```


## Subset your data to the variables and marker genes of interest.

Determine which variables and marker genes you will need and subset your data to them. NarG is an ortholog of NxrA and cannot be distinguished by TreeSAPP. Any sequences that are identified as NxrA by TreeSAPP could be either NarG or NxrA. As a consequence, determine if your data contains NxrA as a marker. If your depth has anoxic conditions (O_2 \< 20 ÂµM, true for all depths \<= 150 m), interpret NxrA assignments by TreeSAPP as NarG instead.

```{r}
<If your depth is <= 150, change "NxrA" entries to "NarG">
<Subset data>
```

## Processing taxonomic information

Your taxonomic information contains information from the highest to lowest available rank. Look at a few rows containing taxonomic information anddetermine the separator between each rank. Google a function that allows you split a character object at each of these separator (Hint: there is a function in the tidyverse with that we did not cover). As a result, your initial single taxonomic variable will be split into multiple new variables, one for each rank. Use the R object `rank` to assign the names for those new variables.

```{r}
rank <- c("root", "kingdom", "phylum", "class",
          "order", "family", "genus", "species")

<split the taxonomic information in the data frame into separate ranks>
```

## Load the geochemical data set into a new data frame.

```{r}
<Load geochemicals>
```

## Additional Hints for each guiding question

### How does abundance of denitrification genes differ across the pathway? Are trends similar for both RNA and DNA?

-   You will need to sum the abundance for each gene. What functions do you know that allow you to summarize data for multiple groups (in this case each marker gene constitutes a group).

### How does microbial diversity differ across the pathway? Are trends similar for both RNA and DNA?

-   You will need to make a decision at what taxonomic rank you will calculate diversity. Why might you not be able to use the lowest possible? Why should your taxonomic rank not be too high?
-   You will need to calculate the total abundance for each taxon (for whichever rank you chose).
-   What R package did you use before to calculate diversity?

### What specific taxa are responsible for denitrification? Are they the same for all steps? For DNA versus RNA?

-   You will need to count how many genes each taxa possess/expresses.

### How does the abundance of denitrification genes relate to nitrogen species in Saanich (use the geochemical data in `Saanich_Data.csv` from our previous data science sessions)?

-   Determine the geochemical conditions at your given depths. Which is the best available terminal electron acceptor?

```{r eval = TRUE, echo = FALSE, warning = FALSE, message = FALSE}
# Reset code chunk settings to `default_chunk_options` (defined in `index.Rmd`)
knitr::opts_chunk$set(default_chunk_options)
```
