<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>E References | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="1. Louca S, Hawley AK, Katsev S, Torres-Beltran M, Bhatia MP, Kheirandish S, Michiels CC, Capelle D, Lavik G, Doebeli M, Crowe SA, Hallam SJ. 2016. Integrating biogeochemistry with multiomic...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="E References | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="1. Louca S, Hawley AK, Katsev S, Torres-Beltran M, Bhatia MP, Kheirandish S, Michiels CC, Capelle D, Lavik G, Doebeli M, Crowe SA, Hallam SJ. 2016. Integrating biogeochemistry with multiomic...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="E References | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="1. Louca S, Hawley AK, Katsev S, Torres-Beltran M, Bhatia MP, Kheirandish S, Michiels CC, Capelle D, Lavik G, Doebeli M, Crowe SA, Hallam SJ. 2016. Integrating biogeochemistry with multiomic...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="frequently-asked-questions.html"><span class="header-section-number">D</span> Frequently Asked Questions</a></li>
<li><a class="active" href="references.html"><span class="header-section-number">E</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="references" class="section level1" number="16">
<h1>
<span class="header-section-number">E</span> References<a class="anchor" aria-label="anchor" href="#references"><i class="fas fa-link"></i></a>
</h1>

<div id="refs" class="references csl-bib-body" entry-spacing="1" line-spacing="2">
<div id="ref-Louca.2016" class="csl-entry">
<div class="csl-left-margin">1. </div>
<div class="csl-right-inline">Louca S, Hawley AK, Katsev S, Torres-Beltran M, Bhatia MP, Kheirandish S, Michiels CC, Capelle D, Lavik G, Doebeli M, Crowe SA, Hallam SJ. 2016. <a href="https://doi.org/10.1073/pnas.1602897113"><span class="nocase">Integrating biogeochemistry with multiomic sequence information in a model oxygen minimum zone</span></a>. Proceedings of the National Academy of Sciences 113:E5925–E5933.</div>
</div>
<div id="ref-Hawley.2014" class="csl-entry">
<div class="csl-left-margin">2. </div>
<div class="csl-right-inline">Hawley AK, Brewer HM, Norbeck AD, Pa a-Toli L, Hallam SJ. 2014. <a href="https://doi.org/10.1073/pnas.1322132111"><span class="nocase">Metaproteomics reveals differential modes of metabolic coupling among ubiquitous oxygen minimum zone microbes</span></a>. Proceedings of the National Academy of Sciences 111:11395–11400.</div>
</div>
<div id="ref-Morgan-Lang.2020" class="csl-entry">
<div class="csl-left-margin">3. </div>
<div class="csl-right-inline">Morgan-Lang C, McLaughlin R, Armstrong Z, Zhang G, Chan K, Hallam SJ. 2020. <a href="https://doi.org/10.1093/bioinformatics/btaa588"><span class="nocase">TreeSAPP: The Tree-based Sensitive and Accurate Phylogenetic Profiler</span></a>. Bioinformatics btaa588–.</div>
</div>
<div id="ref-Letunic.2019" class="csl-entry">
<div class="csl-left-margin">4. </div>
<div class="csl-right-inline">Letunic I, Bork P. 2019. <a href="https://doi.org/10.1093/nar/gkz239"><span class="nocase">Interactive Tree Of Life (iTOL) v4: recent updates and new developments</span></a>. Nucleic Acids Research 47:gkz239–.</div>
</div>
<div id="ref-Khadka.2018" class="csl-entry">
<div class="csl-left-margin">5. </div>
<div class="csl-right-inline">Khadka R, Clothier L, Wang L, Lim CK, Klotz MG, Dunfield PF. 2018. <a href="https://doi.org/10.3389/fmicb.2018.02493"><span class="nocase">Evolutionary History of Copper Membrane Monooxygenases</span></a>. Frontiers in Microbiology 9:2493.</div>
</div>
<div id="ref-Criscuolo.2010" class="csl-entry">
<div class="csl-left-margin">6. </div>
<div class="csl-right-inline">Criscuolo A, Gribaldo S. 2010. <span>BMGE</span> (block mapping and gathering with entropy): A new software for selection of phylogenetic informative regions from multiple sequence alignments. BMC Evol Biol 10.</div>
</div>
<div id="ref-Chaumeil.2019" class="csl-entry">
<div class="csl-left-margin">7. </div>
<div class="csl-right-inline">Chaumeil P-A, Mussig AJ, Hugenholtz P, Parks DH. 2019. <span>GTDB-Tk</span>: A toolkit to classify genomes with the genome taxonomy database. Bioinformatics.</div>
</div>
<div id="ref-Hallam2006" class="csl-entry">
<div class="csl-left-margin">8. </div>
<div class="csl-right-inline">Hallam SJ, Mincer TJ, Schleper C, Preston CM, Roberts K, Richardson PM, DeLong EF. 2006. <a href="https://doi.org/10.1371/journal.pbio.0040095"><span class="nocase">Pathways of carbon assimilation and ammonia oxidation suggested by environmental genomic analyses of marine Crenarchaeota.</span></a> PLoS biology 4:e95.</div>
</div>
<div id="ref-Suzuki2012" class="csl-entry">
<div class="csl-left-margin">9. </div>
<div class="csl-right-inline">Suzuki T, Nakamura T, Fuse H. 2012. <a href="https://doi.org/10.1264/jsme2.ME11256"><span class="nocase">Isolation of two novel marine ethylene-assimilating bacteria, Haliea species ETY-M and ETY-NAG, containing particulate methane monooxygenase-like genes</span></a>. Microbes and Environments 27:54–60.</div>
</div>
<div id="ref-Oswald2017" class="csl-entry">
<div class="csl-left-margin">10. </div>
<div class="csl-right-inline">Oswald K, Graf JS, Littmann S, Tienken D, Brand A, Wehrli B, Albertsen M, Daims H, Wagner M, Kuypers MMM, Schubert CJ, Milucka J. 2017. <a href="https://doi.org/10.1038/ismej.2017.77"><span class="nocase">Crenothrix are major methane consumers in stratified lakes</span></a>. ISME Journal 11:2124–2140.</div>
</div>
<div id="ref-Rochman2020" class="csl-entry">
<div class="csl-left-margin">11. </div>
<div class="csl-right-inline">Rochman FF, Kwon M, Khadka R, Tamas I, Lopez-Jauregui AA, Sheremet A, V. Smirnova A, Malmstrom RR, Yoon S, Woyke T, Dunfield PF, Verbeke TJ. 2020. <a href="https://doi.org/10.1038/s41396-019-0561-2"><span class="nocase">Novel copper-containing membrane monooxygenases (CuMMOs) encoded by alkane-utilizing Betaproteobacteria</span></a>. ISME Journal 14:714–726.</div>
</div>
<div id="ref-Wagner.2012" class="csl-entry">
<div class="csl-left-margin">12. </div>
<div class="csl-right-inline">Wagner GP, Kin K, Lynch VJ. 2012. <a href="https://doi.org/10.1007/s12064-012-0162-3"><span class="nocase">Measurement of mRNA abundance using RNA-seq data: RPKM measure is inconsistent among samples</span></a>. Theory in Biosciences 131:281–285.</div>
</div>
<div id="ref-Li.2013" class="csl-entry">
<div class="csl-left-margin">13. </div>
<div class="csl-right-inline">Li H. 2013. <span class="nocase">Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</span>. arXiv.</div>
</div>
<div id="ref-Hallam.2017" class="csl-entry">
<div class="csl-left-margin">14. </div>
<div class="csl-right-inline">Hallam SJ, Torres-Beltrán M, Hawley AK. 2017. <a href="https://doi.org/10.1038/sdata.2017.158"><span class="nocase">Monitoring microbial responses to ocean deoxygenation in a model oxygen minimum zone</span></a>. Scientific Data 4:170158.</div>
</div>
<div id="ref-Torres-Beltrán.2017" class="csl-entry">
<div class="csl-left-margin">15. </div>
<div class="csl-right-inline">Torres-Beltrán M, Hawley AK, Capelle D, Zaikova E, Walsh DA, Mueller A, Scofield M, Payne C, Pakhomova L, Kheirandish S, Finke J, Bhatia M, Shevchuk O, Gies EA, Fairley D, Michiels C, Suttle CA, Whitney F, Crowe SA, Tortell PD, Hallam SJ. 2017. <a href="https://doi.org/10.1038/sdata.2017.159"><span class="nocase">A compendium of geochemical information from the Saanich Inlet water column</span></a>. Scientific Data 4:170159.</div>
</div>
<div id="ref-Uritskiy.2018" class="csl-entry">
<div class="csl-left-margin">16. </div>
<div class="csl-right-inline">Uritskiy GV, DiRuggiero J, Taylor J. 2018. <a href="https://doi.org/10.1186/s40168-018-0541-1"><span class="nocase">MetaWRAP—a flexible pipeline for genome-resolved metagenomic data analysis</span></a>. Microbiome 6:158.</div>
</div>
<div id="ref-Bolger.2014" class="csl-entry">
<div class="csl-left-margin">17. </div>
<div class="csl-right-inline">Bolger AM, Lohse M, Usadel B. 2014. <a href="https://doi.org/10.1093/bioinformatics/btu170"><span class="nocase">Trimmomatic: a flexible trimmer for Illumina sequence data</span></a>. Bioinformatics 30:2114–2120.</div>
</div>
<div id="ref-Prjibelski.2020" class="csl-entry">
<div class="csl-left-margin">18. </div>
<div class="csl-right-inline">Prjibelski A, Antipov D, Meleshko D, Lapidus A, Korobeynikov A. 2020. <a href="https://doi.org/10.1002/cpbi.102"><span>Using SPAdes De Novo Assembler</span></a>. Current Protocols in Bioinformatics 70:e102.</div>
</div>
<div id="ref-Parks.2015" class="csl-entry">
<div class="csl-left-margin">19. </div>
<div class="csl-right-inline">Parks DH, Imelfort M, Skennerton CT, Hugenholtz P, Tyson GW. 2015. <a href="https://doi.org/10.1101/gr.186072.114"><span class="nocase">CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes</span></a>. Genome Research 25:1043–1055.</div>
</div>
<div id="ref-Tennessen.2016" class="csl-entry">
<div class="csl-left-margin">20. </div>
<div class="csl-right-inline">Tennessen K, Andersen E, Clingenpeel S, Rinke C, Lundberg DS, Han J, Dangl JL, Ivanova N, Woyke T, Kyrpides N, Pati A. 2016. <a href="https://doi.org/10.1038/ismej.2015.100"><span class="nocase">ProDeGe: a computational protocol for fully automated decontamination of genomes</span></a>. The ISME Journal 10:269–272.</div>
</div>
<div id="ref-Hawley.2017" class="csl-entry">
<div class="csl-left-margin">21. </div>
<div class="csl-right-inline">Hawley AK, Torres-Beltrán M, Zaikova E, Walsh DA, Mueller A, Scofield M, Kheirandish S, Payne C, Pakhomova L, Bhatia M, Shevchuk O, Gies EA, Fairley D, Malfatti SA, Norbeck AD, Brewer HM, Pasa-Tolic L, Rio TG del, Suttle CA, Tringe S, Hallam SJ. 2017. <a href="https://doi.org/10.1038/sdata.2017.160"><span class="nocase">A compendium of multi-omic sequence information from the Saanich Inlet water column</span></a>. Scientific Data 4:170160.</div>
</div>
<div id="ref-Li.2015" class="csl-entry">
<div class="csl-left-margin">22. </div>
<div class="csl-right-inline">Li D, Liu C-M, Luo R, Sadakane K, Lam T-W. 2015. <a href="https://doi.org/10.1093/bioinformatics/btv033"><span class="nocase">MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph</span></a>. Bioinformatics 31:1674–1676.</div>
</div>
<div id="ref-kang.2019" class="csl-entry">
<div class="csl-left-margin">23. </div>
<div class="csl-right-inline">Kang DD, Li F, Kirton E, Thomas A, Egan R, An H, Wang Z. 2019. MetaBAT 2: An adaptive binning algorithm for robust and efficient genome reconstruction from metagenome assemblies. PeerJ 7:e7359.</div>
</div>
<div id="ref-Wu.2014" class="csl-entry">
<div class="csl-left-margin">24. </div>
<div class="csl-right-inline">Wu Y-W, Tang Y-H, Tringe SG, Simmons BA, Singer SW. 2014. MaxBin: An automated binning method to recover individual genomes from metagenomes using an expectation-maximization algorithm. Microbiome 2:1–18.</div>
</div>
<div id="ref-Wu.2016" class="csl-entry">
<div class="csl-left-margin">25. </div>
<div class="csl-right-inline">Wu Y-W, Simmons BA, Singer SW. 2016. MaxBin 2.0: An automated binning algorithm to recover genomes from multiple metagenomic datasets. Bioinformatics 32:605–607.</div>
</div>
<div id="ref-Raghunathan.2005" class="csl-entry">
<div class="csl-left-margin">26. </div>
<div class="csl-right-inline">Raghunathan A, Ferguson HR, Bornarth CJ, Song W, Driscoll M, Lasken RS. 2005. Genomic DNA amplification from a single bacterium. Applied and environmental microbiology 71:3342–3347.</div>
</div>
<div id="ref-Rinke.2014" class="csl-entry">
<div class="csl-left-margin">27. </div>
<div class="csl-right-inline">Rinke C, Lee J, Nath N, Goudeau D, Thompson B, Poulton N, Dmitrieff E, Malmstrom R, Stepanauskas R, Woyke T. 2014. Obtaining genomes from uncultivated environmental microorganisms using <span class="nocase">FACS-based</span> single-cell genomics. Nat Protoc 9:1038–1048.</div>
</div>
<div id="ref-Tyson.2004" class="csl-entry">
<div class="csl-left-margin">28. </div>
<div class="csl-right-inline">Tyson GW, Chapman J, Hugenholtz P, Allen EE, Ram RJ, Richardson PM, Solovyev VV, Rubin EM, Rokhsar DS, Banfield JF. 2004. Community structure and metabolism through reconstruction of microbial genomes from the environment. Nature 428:37–43.</div>
</div>
<div id="ref-Swan.2011" class="csl-entry">
<div class="csl-left-margin">29. </div>
<div class="csl-right-inline">Swan BK, Martinez-Garcia M, Preston CM, Sczyrba A, Woyke T, Lamy D, Reinthaler T, Poulton NJ, Masland EDP, Gomez ML, Sieracki ME, DeLong EF, Herndl GJ, Stepanauskas R. 2011. Potential for chemolithoautotrophy among ubiquitous bacteria lineages in the dark ocean. Science 333:1296–1300.</div>
</div>
<div id="ref-Quast.2013" class="csl-entry">
<div class="csl-left-margin">30. </div>
<div class="csl-right-inline">Quast C, Pruesse E, Yilmaz P, Gerken J, Schweer T, Yarza P, Peplies J, Glöckner FO. 2013. The <span>SILVA</span> ribosomal <span>RNA</span> gene database project: Improved data processing and web-based tools. Nucleic Acids Res 41:D590–6.</div>
</div>
<div id="ref-Camacho.2009" class="csl-entry">
<div class="csl-left-margin">31. </div>
<div class="csl-right-inline">Camacho C, Coulouris G, Avagyan V, Ma N, Papadopoulos J, Bealer K, Madden TL. 2009. <span>BLAST+</span>: Architecture and applications. BMC Bioinformatics 10:421.</div>
</div>
<div id="ref-Bankevich.2012" class="csl-entry">
<div class="csl-left-margin">32. </div>
<div class="csl-right-inline">Bankevich A, Nurk S, Antipov D, Gurevich AA, Dvorkin M, Kulikov AS, Lesin VM, Nikolenko SI, Pham S, Prjibelski AD, Pyshkin AV, Sirotkin AV, Vyahhi N, Tesler G, Alekseyev MA, Pevzner PA. 2012. <span>SPAdes</span>: A new genome assembly algorithm and its applications to single-cell sequencing. J Comput Biol 19:455–477.</div>
</div>
</div>
</div>






  <div class="chapter-nav">
<div class="prev"><a href="frequently-asked-questions.html"><span class="header-section-number">D</span> Frequently Asked Questions</a></div>
<div class="empty"></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav"><li><a class="nav-link" href="#references"><span class="header-section-number">E</span> References</a></li></ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/references.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/references.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on April 13, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
