<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>11 Running TreeSAPP on a server | Gene-centric mapping of microbial community structure and function</title>
<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam">
<meta name="description" content="These instructions will help you to complete the script template treesapp_analysis.sh to run your TreeSAPP analysis of the Saanich Inlet data of your assigned depth on the server and copy the...">
<meta name="generator" content="bookdown 0.24 with bs4_book()">
<meta property="og:title" content="11 Running TreeSAPP on a server | Gene-centric mapping of microbial community structure and function">
<meta property="og:type" content="book">
<meta property="og:description" content="These instructions will help you to complete the script template treesapp_analysis.sh to run your TreeSAPP analysis of the Saanich Inlet data of your assigned depth on the server and copy the...">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="11 Running TreeSAPP on a server | Gene-centric mapping of microbial community structure and function">
<meta name="twitter:description" content="These instructions will help you to complete the script template treesapp_analysis.sh to run your TreeSAPP analysis of the Saanich Inlet data of your assigned depth on the server and copy the...">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/header-attrs-2.11/header-attrs.js"></script><script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.3.1/transition.js"></script><script src="libs/bs3compat-0.3.1/tabs.js"></script><script src="libs/bs3compat-0.3.1/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- CSS --><link rel="stylesheet" href="bs4_style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="2020.21 MICB425 Mirobial Ecological Genomics Capstone project">Gene-centric mapping of microbial community structure and function</a>:
        <small class="text-muted">2020.21 MICB425 Mirobial Ecological Genomics Capstone project</small>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Overview</a></li>
<li class="book-part">Introduction</li>
<li><a class="" href="exploring-the-microcosmos.html"><span class="header-section-number">1</span> Exploring the microcosmos</a></li>
<li class="book-part">Tutorials</li>
<li><a class="" href="a-tutorial-for-using-treesapp.html"><span class="header-section-number">2</span> A tutorial for using TreeSAPP</a></li>
<li><a class="" href="acquiring-reference-sequence-data.html"><span class="header-section-number">3</span> Acquiring reference sequence data</a></li>
<li><a class="" href="treesapp-create.html"><span class="header-section-number">4</span> Creating a reference package for TreeSAPP</a></li>
<li><a class="" href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><span class="header-section-number">5</span> Classifying unknown sequences with TreeSAPP and updating a reference package</a></li>
<li><a class="" href="calculating-relative-abundance-of-metaomic-data.html"><span class="header-section-number">6</span> Calculating relative abundance of meta’omic data</a></li>
<li><a class="" href="analysing-the-treesapp-classifications-in-r.html"><span class="header-section-number">7</span> Analysing the TreeSAPP classifications in R</a></li>
<li><a class="" href="basic-automation-with-r.html"><span class="header-section-number">8</span> Basic Automation with R</a></li>
<li class="book-part">Capstone project</li>
<li><a class="" href="project-description.html"><span class="header-section-number">9</span> Project description</a></li>
<li><a class="" href="assessment.html"><span class="header-section-number">10</span> Assessment</a></li>
<li><a class="active" href="running-treesapp-on-a-server.html"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></li>
<li><a class="" href="treesapp-reference-packages.html"><span class="header-section-number">B</span> TreeSAPP reference packages</a></li>
<li><a class="" href="shell-cheat-sheet.html"><span class="header-section-number">C</span> Shell cheat sheet</a></li>
<li><a class="" href="frequently-asked-questions.html"><span class="header-section-number">D</span> Frequently Asked Questions</a></li>
<li><a class="" href="references.html"><span class="header-section-number">E</span> References</a></li>
</ul>

        <div class="book-extra">
          <p><a id="book-repo" href="https://github.com/EDUCE-UBC/MICB425">View book source <i class="fab fa-github"></i></a></p>
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="running-treesapp-on-a-server" class="section level1" number="11">
<h1>
<span class="header-section-number">11</span> Running TreeSAPP on a server<a class="anchor" aria-label="anchor" href="#running-treesapp-on-a-server"><i class="fas fa-link"></i></a>
</h1>
<p>These instructions will help you to complete the script template <code>treesapp_analysis.sh</code> to run your TreeSAPP analysis of the Saanich Inlet data of your assigned depth on the server and copy the output files to your local machine. To adjust the code examples below and in the script template, replace any angle brackets, e.g. <code>&lt;SOME TEXT&gt;</code>, with the missing code.</p>
<div id="resources-for-treesapp-analysis" class="section level2" number="11.1">
<h2>
<span class="header-section-number">11.1</span> Resources for TreeSAPP analysis<a class="anchor" aria-label="anchor" href="#resources-for-treesapp-analysis"><i class="fas fa-link"></i></a>
</h2>
<ul>
<li>A shell script template called <code>treesapp_analysis.sh</code> on Canvas.</li>
<li>Access to a server and <code>&lt;server address&gt;</code>.</li>
<li>The Saanich Inlet data located in <code>/mnt/datasets</code>
</li>
</ul>
</div>
<div id="checklist-to-write-and-run-shell-script" class="section level2" number="11.2">
<h2>
<span class="header-section-number">11.2</span> Checklist to write and run shell script<a class="anchor" aria-label="anchor" href="#checklist-to-write-and-run-shell-script"><i class="fas fa-link"></i></a>
</h2>
<ul class="task-list">
<li><p><input type="checkbox" disabled>
Edit the <code>treesapp_analysis.sh</code> script following <a href="running-treesapp-on-a-server.html#shell-writing">the instructions below</a>.</p></li>
<li>
<p><input type="checkbox" disabled>
Using the terminal on your computer, copy the script to your group’s server using:</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb110-1"><a href="running-treesapp-on-a-server.html#cb110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> <span class="op">&lt;</span>path to file on your computer<span class="op">&gt;</span> <span class="op">&lt;</span>user<span class="op">&gt;</span>@<span class="op">&lt;</span>server address<span class="op">&gt;</span>:<span class="op">&lt;</span>destination<span class="op">&gt;</span></span></code></pre></div>
</li>
<li>
<p><input type="checkbox" disabled>
Make the script executable on the server (otherwise you won’t be able to run it). In the CLI and while in the directory containing your script, run the following command:</p>
<div class="sourceCode" id="cb111"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb111-1"><a href="running-treesapp-on-a-server.html#cb111-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chmod</span> u+x treesapp_analysis.sh</span></code></pre></div>
</li>
<li><p><input type="checkbox" disabled>
Still in the CLI, start a new session with <code>screen -S &lt;session name&gt;</code>.</p></li>
<li><p><input type="checkbox" disabled>
In the new session, start the script with <code>./treesapp_analysis.sh</code>.</p></li>
<li><p><input type="checkbox" disabled>
After the script has run (will take at most 8 hours), locate the <code>final_output/marker_contig_map.tsv</code> files in the TreeSAPP output directories and copy each one to your local computer using <code>scp</code>.</p></li>
</ul>
<p>After these steps, you will continue with the analysis of your results in R on your local computer using R and the provided <code>treesapp_analysis.R</code> script template.</p>
</div>
<div id="shell-writing" class="section level2" number="11.3">
<h2>
<span class="header-section-number">11.3</span> Writing the shell script<a class="anchor" aria-label="anchor" href="#shell-writing"><i class="fas fa-link"></i></a>
</h2>
<p>In the <code>treesapp_analysis.sh</code> script, we will define some variables to make our code more readable when running commands, give some feedback to the user as the script runs, and finally run the TreeSAPP workflow. Edit the script either on your local computer or after you have copied it to the server using the text editor of your choice (e.g. RStudio on your local machine or <code>nano</code> on the server).</p>
<div id="defining-the-shell" class="section level3" number="11.3.1">
<h3>
<span class="header-section-number">11.3.1</span> Defining the shell<a class="anchor" aria-label="anchor" href="#defining-the-shell"><i class="fas fa-link"></i></a>
</h3>
<p>The script template starts with:</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb112-1"><a href="running-treesapp-on-a-server.html#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="co">#! bin/bash</span></span></code></pre></div>
<p>and defines the shell used to interpret this script and where the shell is located (i.e. the <code>bash</code> shell in the <code>/bin</code> directory). You do not need to modify anything about that step.</p>
</div>
<div id="setting-a-variable-for-your-depth" class="section level3" number="11.3.2">
<h3>
<span class="header-section-number">11.3.2</span> Setting a variable for your depth<a class="anchor" aria-label="anchor" href="#setting-a-variable-for-your-depth"><i class="fas fa-link"></i></a>
</h3>
<p>Let’s define our first variable with the name “depth”. Replace <code>&lt;your depth&gt;</code> with the depth your group has been assigned to (e.g. <code>100</code>).</p>
<div class="sourceCode" id="cb113"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb113-1"><a href="running-treesapp-on-a-server.html#cb113-1" aria-hidden="true" tabindex="-1"></a><span class="va">depth</span><span class="op">=</span><span class="st">"&lt;your  depth&gt;"</span></span></code></pre></div>
</div>
<div id="providing-user-feedback" class="section level3" number="11.3.3">
<h3>
<span class="header-section-number">11.3.3</span> Providing user feedback<a class="anchor" aria-label="anchor" href="#providing-user-feedback"><i class="fas fa-link"></i></a>
</h3>
<p>It is good practice to provide feedback to the user while a script is running by printing messages to the terminal. Let’s print a welcome message to the terminal using the <code>echo</code> command to informing the user what the script is going to do. This is a great opportunity to call the <code>depth</code> variable that we have just defined.</p>
<p><em>Hint</em>: Remember, to call a variable, you have to prepend its name with a <code>$</code>, e.g. if the variable’s name is <code>some_variable</code> then you call it with <code>$some_variable</code>.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb114-1"><a href="running-treesapp-on-a-server.html#cb114-1" aria-hidden="true" tabindex="-1"></a><span class="bu">echo</span> <span class="st">"TreeSAPP analysis of Saanich Inlet Data at Depth &lt;call depth variable&gt; m"</span></span></code></pre></div>
</div>
<div id="defining-variable-for-input-directory" class="section level3" number="11.3.4">
<h3>
<span class="header-section-number">11.3.4</span> Defining variable for input directory<a class="anchor" aria-label="anchor" href="#defining-variable-for-input-directory"><i class="fas fa-link"></i></a>
</h3>
<p>Define a variable for path to the input data directory. Let’s call the variable <code>input_dir</code>.</p>
<div class="sourceCode" id="cb115"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb115-1"><a href="running-treesapp-on-a-server.html#cb115-1" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>variable_name<span class="op">&gt;</span>=<span class="st">"&lt;path to data directory&gt;"</span></span></code></pre></div>
</div>
<div id="creating-output-directory" class="section level3" number="11.3.5">
<h3>
<span class="header-section-number">11.3.5</span> Creating output directory<a class="anchor" aria-label="anchor" href="#creating-output-directory"><i class="fas fa-link"></i></a>
</h3>
<p>Define a variable for the TreeSAPP output directory and choose a name for it yourself. The directory should be located in your home directory (which is also shared between your host and container). To refer to your home directory, use the environmental variable <code>HOME</code> which is automatically defined by the shell (i.e. you do not need to first define it yourself). Finally, create the output directory.</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb116-1"><a href="running-treesapp-on-a-server.html#cb116-1" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>set <span class="ex">output</span> directory variable<span class="op">&gt;</span></span>
<span id="cb116-2"><a href="running-treesapp-on-a-server.html#cb116-2" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>create <span class="ex">the</span> ouput directory<span class="op">&gt;</span></span></code></pre></div>
</div>
<div id="reporting-variables-to-user" class="section level3" number="11.3.6">
<h3>
<span class="header-section-number">11.3.6</span> Reporting variables to user<a class="anchor" aria-label="anchor" href="#reporting-variables-to-user"><i class="fas fa-link"></i></a>
</h3>
<p>Let’s provide the user some feedback where TreeSAPP is going to save its output and what bind paths have been set.</p>
<div class="sourceCode" id="cb117"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb117-1"><a href="running-treesapp-on-a-server.html#cb117-1" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>print <span class="bu">command</span><span class="op">&gt;</span> <span class="st">"&lt;Some text explaining what variables have been set&gt;"</span></span></code></pre></div>
</div>
<div id="looping-over-multiple-files" class="section level3" number="11.3.7">
<h3>
<span class="header-section-number">11.3.7</span> Looping over multiple files<a class="anchor" aria-label="anchor" href="#looping-over-multiple-files"><i class="fas fa-link"></i></a>
</h3>
<p>For-loops minimize the amount of redundant code and ensure all data is processed similarly.
Below is an example for-loop that could be used to classify metagenome sequences with treesapp assign.</p>
<div class="sourceCode" id="cb118"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb118-1"><a href="running-treesapp-on-a-server.html#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="op">&lt;</span>path/to/assembly_files<span class="op">&gt;</span>/SI072_<span class="pp">*</span>m_contig.fa</span>
<span id="cb118-2"><a href="running-treesapp-on-a-server.html#cb118-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span> <span class="va">sample</span><span class="op">=</span><span class="va">$(</span> <span class="fu">basename</span> <span class="va">$f</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/_contig.fa//g'</span><span class="va">)</span></span>
<span id="cb118-3"><a href="running-treesapp-on-a-server.html#cb118-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">treesapp</span> assign <span class="at">-i</span> <span class="va">$f</span> <span class="dt">\</span></span>
<span id="cb118-4"><a href="running-treesapp-on-a-server.html#cb118-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--refpkg_dir</span> <span class="op">&lt;</span>path_to_refpkgs/<span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb118-5"><a href="running-treesapp-on-a-server.html#cb118-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--output</span> <span class="op">&lt;</span>path/to/outputs<span class="op">&gt;</span>/<span class="va">${sample}</span>_assign <span class="dt">\</span></span>
<span id="cb118-6"><a href="running-treesapp-on-a-server.html#cb118-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--trim_align</span> <span class="at">-n</span> 8</span>
<span id="cb118-7"><a href="running-treesapp-on-a-server.html#cb118-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
<p>This loop iterates over the seven metagenome assembly files that match the pattern ’SI072_*m_contig.fa’, where the asterisk represents any characters one or more times.
Effectively, it replaces seven nearly identical <code>treesapp assign</code> commands.
A variable called ‘sample’ is created by calling <code>basename</code> on the file path to return the filename, and using the <code>sed</code> (stream editor) to remove the assembly-specific suffix ’_contig.fa’.
The name of the output directory is based on this ‘sample’ variable to ensure the directories are unique across metagenomes.</p>
<p>This for-loop can be modified to loop over <code>treesapp abundance</code> commands as well (assuming you’ve followed the above naming convention for the <code>treesapp assign</code> output directory):</p>
<div class="sourceCode" id="cb119"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb119-1"><a href="running-treesapp-on-a-server.html#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> f <span class="kw">in</span> <span class="op">&lt;</span>path/to/outputs<span class="op">&gt;</span>/SI072_<span class="pp">*</span>assign</span>
<span id="cb119-2"><a href="running-treesapp-on-a-server.html#cb119-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">do</span> <span class="va">sample</span><span class="op">=</span><span class="va">$(</span> <span class="fu">basename</span> <span class="va">$f</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/_assign//g'</span><span class="va">)</span></span>
<span id="cb119-3"><a href="running-treesapp-on-a-server.html#cb119-3" aria-hidden="true" tabindex="-1"></a>  <span class="ex">treesapp</span> abundance <span class="dt">\</span></span>
<span id="cb119-4"><a href="running-treesapp-on-a-server.html#cb119-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">--treesapp_output</span> <span class="va">$f</span> <span class="dt">\</span></span>
<span id="cb119-5"><a href="running-treesapp-on-a-server.html#cb119-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">--reads</span> <span class="op">&lt;</span>path/to<span class="op">&gt;</span>/MetaG_Trim_QC_Reads/<span class="va">${sample}</span>_pe.1.fq.gz <span class="dt">\</span></span>
<span id="cb119-6"><a href="running-treesapp-on-a-server.html#cb119-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">--reverse</span> <span class="op">&lt;</span>path/to<span class="op">&gt;</span>/MetaG_Trim_QC_Reads/<span class="va">${sample}</span>_pe.2.fq.gz <span class="dt">\</span></span>
<span id="cb119-7"><a href="running-treesapp-on-a-server.html#cb119-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">-n</span> 8 <span class="dt">\</span></span>
<span id="cb119-8"><a href="running-treesapp-on-a-server.html#cb119-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">--report</span> update</span>
<span id="cb119-9"><a href="running-treesapp-on-a-server.html#cb119-9" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div>
</div>
<div id="listing-treesapp-version" class="section level3" number="11.3.8">
<h3>
<span class="header-section-number">11.3.8</span> Listing TreeSAPP version<a class="anchor" aria-label="anchor" href="#listing-treesapp-version"><i class="fas fa-link"></i></a>
</h3>
<p>Let’s print the information about the configuration of TreeSAPP.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb120-1"><a href="running-treesapp-on-a-server.html#cb120-1" aria-hidden="true" tabindex="-1"></a> <span class="ex">treesapp</span> info</span></code></pre></div>
<p><em>Hint</em>: When we have very long commands in a script, we can introduce line breaks using <code>\</code> which are best used right after a space in the command (see example above after the word “following”).</p>
</div>
<div id="executing-treesapp-analysis" class="section level3" number="11.3.9">
<h3>
<span class="header-section-number">11.3.9</span> Executing TreeSAPP analysis<a class="anchor" aria-label="anchor" href="#executing-treesapp-analysis"><i class="fas fa-link"></i></a>
</h3>
<p>In the terminal (i.e. not as part of this script), check the documentation for <code>treesapp assign</code>, the command you will use for the automated reconstruction of a nutrient cycle (visit its <a href="https://github.com/hallamlab/TreeSAPP/wiki/Classifying-sequences-with-treesapp-assign">wikipage</a> to learn more).</p>
<div class="sourceCode" id="cb121"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb121-1"><a href="running-treesapp-on-a-server.html#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="ex">treesapp</span> assign <span class="at">-h</span></span></code></pre></div>
<p>Looking at the TreeSAPP documentation, determine the flags you need in addition to the ones already included below.</p>
<p>Time to run our analysis! You will have to identify the input file(s) that you need to process. You will run the analysis once for metagenomic (two files, forward reads are in the file that contains <code>.1</code> in its name, and reverse reads in file the with <code>.2</code>) and once for metatranscriptomic reads (a single file that contains both forward and reverse reads) and align them to the assembled contigs. All reads are pair-end.</p>
<p>You will need to provide the following settings using flags:</p>
<ul>
<li>Which assembly file to use.</li>
<li>To use 8 CPUs for the analysis.</li>
<li>To do the analysis for all marker genes provided in TreeSAPP.</li>
<li>To output a verbose runtime log.</li>
<li>Where to save the results.</li>
<li>To calculate the rpkm values for detected sequences.</li>
<li>Which reads to use and what type they are (i.e. pair-end or single-end).</li>
<li>To delete any intermediate files generated by the analysis.</li>
<li>To use position masking of multiple sequence alignment.</li>
</ul>
<p><em>Hints</em>:</p>
<ul>
<li>Will you ouput the data for metagenomic and metatranscriptomic analysis to the same folder? Does the folder already exist?</li>
<li>If you want the default value of a flag, then you do not need to provide it.</li>
<li>Where is the input data located in the singularity container?</li>
<li>Metatrascriptomic reads are <strong>NOT</strong> rRNA.</li>
<li>In scripts you often refer to variables while combined with additional text. Let’s say you have defined the variable <code>file_name="some_file"</code> to refer to the file <code>some_file</code>. Now you want to save a modified version of the file as <code>some_file_modified</code>. If you would write <code>$file_name_modified</code>, then the shell does not know that you just want to call a variable called <code>$file_name</code> and instead will assume you are calling a variable called <code>$file_name_modified</code> and will return an empty string. To indicate the beginning and end of a variable name, use curly braces. In the above example, you would use <code>${file_name}_modified</code>.</li>
</ul>
<p>Optional bonus challenge:<br>
Earlier we defined the <code>depth</code> variable. Could you use it in some way when you define which assembly and reads you are using in the analysis?</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb122-1"><a href="running-treesapp-on-a-server.html#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Metagenomic analysis</span></span>
<span id="cb122-2"><a href="running-treesapp-on-a-server.html#cb122-2" aria-hidden="true" tabindex="-1"></a><span class="ex">treesapp</span> assign <span class="dt">\</span></span>
<span id="cb122-3"><a href="running-treesapp-on-a-server.html#cb122-3" aria-hidden="true" tabindex="-1"></a>-i <span class="op">&lt;</span>input data dir<span class="op">&gt;&lt;</span>file path to your assembly<span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb122-4"><a href="running-treesapp-on-a-server.html#cb122-4" aria-hidden="true" tabindex="-1"></a>-m <span class="op">&lt;</span>choose the correct option<span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb122-5"><a href="running-treesapp-on-a-server.html#cb122-5" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>number of CPU threads, set to <span class="dv">8</span><span class="op">&gt;</span></span>
<span id="cb122-6"><a href="running-treesapp-on-a-server.html#cb122-6" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>your <span class="ex">output</span> directory<span class="op">&gt;</span> <span class="dt">\</span></span>
<span id="cb122-7"><a href="running-treesapp-on-a-server.html#cb122-7" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>your remaining settings<span class="op">&gt;</span></span>
<span id="cb122-8"><a href="running-treesapp-on-a-server.html#cb122-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb122-9"><a href="running-treesapp-on-a-server.html#cb122-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Metatranscriptomic analysis</span></span>
<span id="cb122-10"><a href="running-treesapp-on-a-server.html#cb122-10" aria-hidden="true" tabindex="-1"></a><span class="op">&lt;</span>you <span class="ex">are</span> on your own now :<span class="er">)</span><span class="op">&gt;</span></span></code></pre></div>
</div>
<div id="concatenating-classification-tables" class="section level3" number="11.3.10">
<h3>
<span class="header-section-number">11.3.10</span> Concatenating classification tables<a class="anchor" aria-label="anchor" href="#concatenating-classification-tables"><i class="fas fa-link"></i></a>
</h3>
<p>Once the <code>treesapp assign</code>, <code>treesapp abundance</code> and <code>treesapp layer</code> commands have completed successfully it is convenient to store classification data from all samples in a single table.
This one table can then be downloaded from the server and loaded into R.</p>
<p>You can combine all tables with two lines of code in bash, using <code>cat</code>, <code>head</code>, and <code>tail</code>.
The name of the combined table for this example is ‘SI072_classifications.tsv’.
This first line writes <em>just</em> the header from one of the classification tables to the combined table while the second writes the rows (i.e. not including the column names) to the combined table.
This ensures the final table is properly formatted, i.e., one row containing the column names and the remaining rows containing data.</p>
<div class="sourceCode" id="cb123"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb123-1"><a href="running-treesapp-on-a-server.html#cb123-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> /data/SI072_<span class="pp">*</span>m_assign/final_outputs/classifications.tsv <span class="kw">|</span> <span class="fu">head</span> <span class="at">-n</span> 1 <span class="op">&gt;</span>SI072_classifications.tsv</span>
<span id="cb123-2"><a href="running-treesapp-on-a-server.html#cb123-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span> <span class="at">-q</span> <span class="at">-n</span> +2 /data/SI072_<span class="pp">*</span>m_assign/final_outputs/classifications.tsv <span class="op">&gt;&gt;</span>SI072_classifications.tsv</span></code></pre></div>
<p>The above command will work for both the classifications.tsv file created by <code>treesapp assign</code> or the layered_classifications.tsv file created by <code>treesapp layer</code> if the file name is changed from ‘classifications.tsv’ to ‘layered_classifications.tsv’ in the commands above.</p>

</div>
</div>
</div>



<div class="chapter-nav">
<div class="prev"><a href="assessment.html"><span class="header-section-number">10</span> Assessment</a></div>
<div class="next"><a href="the-saanich-inlet-data-set.html"><span class="header-section-number">A</span> The Saanich Inlet data set</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#running-treesapp-on-a-server"><span class="header-section-number">11</span> Running TreeSAPP on a server</a></li>
<li><a class="nav-link" href="#resources-for-treesapp-analysis"><span class="header-section-number">11.1</span> Resources for TreeSAPP analysis</a></li>
<li><a class="nav-link" href="#checklist-to-write-and-run-shell-script"><span class="header-section-number">11.2</span> Checklist to write and run shell script</a></li>
<li>
<a class="nav-link" href="#shell-writing"><span class="header-section-number">11.3</span> Writing the shell script</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#defining-the-shell"><span class="header-section-number">11.3.1</span> Defining the shell</a></li>
<li><a class="nav-link" href="#setting-a-variable-for-your-depth"><span class="header-section-number">11.3.2</span> Setting a variable for your depth</a></li>
<li><a class="nav-link" href="#providing-user-feedback"><span class="header-section-number">11.3.3</span> Providing user feedback</a></li>
<li><a class="nav-link" href="#defining-variable-for-input-directory"><span class="header-section-number">11.3.4</span> Defining variable for input directory</a></li>
<li><a class="nav-link" href="#creating-output-directory"><span class="header-section-number">11.3.5</span> Creating output directory</a></li>
<li><a class="nav-link" href="#reporting-variables-to-user"><span class="header-section-number">11.3.6</span> Reporting variables to user</a></li>
<li><a class="nav-link" href="#looping-over-multiple-files"><span class="header-section-number">11.3.7</span> Looping over multiple files</a></li>
<li><a class="nav-link" href="#listing-treesapp-version"><span class="header-section-number">11.3.8</span> Listing TreeSAPP version</a></li>
<li><a class="nav-link" href="#executing-treesapp-analysis"><span class="header-section-number">11.3.9</span> Executing TreeSAPP analysis</a></li>
<li><a class="nav-link" href="#concatenating-classification-tables"><span class="header-section-number">11.3.10</span> Concatenating classification tables</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
<li><a id="book-source" href="https://github.com/EDUCE-UBC/MICB425/blob/main/shell.Rmd">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="https://github.com/EDUCE-UBC/MICB425/edit/main/shell.Rmd">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
</div>
    </nav>
</div>
</div>
  

  

</div>
 <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>Gene-centric mapping of microbial community structure and function</strong>: 2020.21 MICB425 Mirobial Ecological Genomics Capstone project" was written by Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam. It was last built on June 01, 2022.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script><script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>
</html>
