# Denitrification summary {#summary}

```{r setup-summary, echo = FALSE, warning = FALSE, message = FALSE}
library(tidyverse)
library(vegan)
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, message = FALSE, fig.width = 8)
```

```{r data}
# Helper function to load TreeSAPP maker contig maps
load_treesapp_map <- function(depth) {
  depth_path <- paste0(depth, "m")
  metaG_path <- file.path("clean_data", depth_path, "MetaG",
                          "final_outputs", "marker_contig_map.tsv")
  metaT_path <- file.path("clean_data", depth_path, "MetaT",
                          "final_outputs", "marker_contig_map.tsv")
  metaG <- read_tsv(metaG_path) %>% 
    mutate("depth" = depth,
           sample  = "DNA")
  metaT <- read_tsv(metaT_path) %>% 
    mutate("depth" = depth,
           sample  = "RNA")
  rbind(metaG, metaT)
}



# Capitalize gene
capitalize_gene <- function(string)  {
  paste0(str_to_upper(str_sub(string, 1, 1)), str_sub(string, 2))
}



# Call NarG based on NxrA presence and oxic conditions (visually inspected from
# geochemicals conditions)
detect_NarG <- function(gene, depth, max_oxic_depth) {
 ifelse(gene == "NxrA" & depth > max_oxic_depth,
        "NarG", gene) 
}



# Function to plot Phyla bubbble plots
plot_phyla_abundances <- function(df, source, depths) {
  df %>%
  # Select variables
  select(Marker, depth, sample, Abundance, phylum, kingdom) %>%
  # Use just DNA and depths with oxygen
  filter(sample == source & depth %in% depths) %>%
  # Fill in phyla NAs with kingdom data, if available
  mutate(phylum = ifelse(is.na(phylum),
                         ifelse(
                           is.na(kingdom),
                           "Unclassified",
                           paste("Unclassified", kingdom)
                         ),
                         phylum)) %>%
  # Calculate total abundance for DNA or RNA (sample) for each gene at each
  # depth in each phylum
  group_by(Marker, depth, sample, phylum, kingdom) %>%
  summarize(abund_tot = sum(Abundance)) %>%
  
  # Plot phyla abundance by gene and depth
  ggplot(aes(y = phylum, x = Marker, color = phylum)) +
  # Size to abundance. Convert 0 to NA
  geom_point(aes(size = ifelse(abund_tot == 0, NA, abund_tot))) +
  facet_grid(~ depth) +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 16) +
  ## Rename size legend. Set breaks to match next plot of RNA
  scale_size_continuous(
    name = "Abundance",
    breaks = c(500, 1000, 2000, 3000, 4000),
    limits = c(0, 4300)
  ) +
  ## Remove axes labels and color legend
  labs(y = "", x = "") +
  guides(color = FALSE) +
  ## Rotate x axis labels, move size legend to bottom
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "bottom")
}



plot_proteobacteria <- function(df, source) {
  df %>%
  # Select variables
  select(Marker, depth, sample, Abundance, phylum, class) %>%
  # Filter to 1 phyla and DNA data
  filter(phylum == "Proteobacteria" & sample == source) %>%
  # Fill in class NAs with kingdom data, if available
  mutate(class = ifelse(is.na(class),
                        paste0("Unclassified\n", phylum),
                        class)) %>%
  
  # Calculate total abundance for DNA or RNA (sample) for each gene at each depth
  group_by(Marker, depth, sample, class) %>%
  summarize(abund_tot = sum(Abundance)) %>%
  separate(class, into = c("class", NA), sep = "proteo") %>%
  
  # Plot total DNA and RNA abundance by depth and gene
  ggplot(aes(y = depth, x = Marker, color = class)) +
  # Plot abundance by size. Change 0s to NAs so no dot appears
  geom_point(aes(size = ifelse(abund_tot == 0, NA, abund_tot)),
             # Put class side by side
             position = position_dodge(width = 0.8)) +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Reverse y and set axis ticks
  scale_y_reverse(
    labels = c(10, 100, 120, 135, 150, 165, 200),
    breaks = c(10, 100, 120, 135, 150, 165, 200)
  ) +
  ## Remove color legend title
  scale_color_discrete(name = "") +
  ## Rename size legend and axes
  scale_size_continuous(
    name = "Abundance",
    breaks = c(500, 1000, 2000, 3000, 4000),
    limits = c(0, 4300)
  ) +
  labs(y = "Depth (m)", x = "") +
  theme(legend.position = "bottom", legend.box = "vertical")
}



# Define  genes of interest
denitrification_genes  <- c("NarG", "NapA", "NirK",
                            "NirS", "NorB", "NosZ")

# Define different depths levels for later plotting
lower_depths  <- c(10, 100, 120, 135)
higher_depths <- c(150, 165, 200)
all_depths <- c(lower_depths, higher_depths)
ranks <- c("root", "kingdom", "phylum", "class",
           "order", "family", "genus", "species")



mark <- map_df(all_depths, load_treesapp_map) %>% 
  mutate(Marker = capitalize_gene(Marker),
         Marker = detect_NarG(Marker, depth, 135)) %>% 
  filter(Marker %in% denitrification_genes) %>% 
  # Put Marker genes in specific sequence
  mutate(Marker = fct_relevel(Marker, denitrification_genes),
         # Remove non-taxonomic levels
         Taxonomy = str_replace(Taxonomy, "Cellular organisms; ", ""),
         Confident_Taxonomy = str_replace(Confident_Taxonomy,
                                          "Cellular organisms; ", ""),
         Taxonomy = str_replace(Taxonomy, "; environmental samples", ""),
         Confident_Taxonomy = str_replace(Confident_Taxonomy,
                                          "; environmental samples", ""),
         # Remove any phyla or lower level for "unclassified Bacteria"
         Taxonomy = str_replace(Taxonomy, "; unclassified Bacteria.*", ""),
         Confident_Taxonomy = str_replace(Confident_Taxonomy,
                                          "; unclassified Bacteria.*", "")) %>% 
  
  separate(Taxonomy, into = ranks,
           sep="; ") %>% 
  separate(Confident_Taxonomy,
           into = paste0(ranks, 2),
           sep="; ")



alpha_diversity <- mark %>%
  # Use only DNA data
  filter(sample == "DNA") %>%
  # Select variables
  select(sample, Marker, depth, root:species, Abundance) %>%
  group_by(sample, Marker, depth, root, kingdom, phylum, class, order,
           family, genus, species) %>%
  summarise(Abundance = sum(Abundance)) %>% 
  group_by(sample, Marker, depth) %>% 
  summarise(shannon = diversity(Abundance))



# Geochemical
geochem <- read_csv("clean_data/Saanich_Data.csv") %>%
  mutate(Depth_m = Depth * 1000) %>%
  filter(Cruise == 72 &
         Depth_m %in% all_depths) %>%
  select(Cruise,
         Depth_m,
         O2  = WS_O2,
         NO3 = WS_NO3,
         NO2 = Mean_NO2,
         N2O = Mean_N2O,
         NH4 = Mean_NH4,
         H2S = WS_H2S,
         CH4 = Mean_CH4,
         PO4)
```

```{r echo = FALSE, fig.cap = "Nitrogen cycle"}
knitr::include_graphics("N_cycle_genes.png")
```

## Geochemical data

```{r fig.cap = "Cruise 72 by chemical"}
geochem %>%
  gather(key = "geochemical", value = "concentration", O2:PO4) %>%
  mutate(geochemical = fct_relevel(geochemical, "O2", "NO3", "NO2", "N2O",
                                   "NH4", "H2S", "CH4", "PO4")) %>%
  
  ggplot(aes(x = concentration, y = Depth_m)) +
  geom_point() +
  geom_path() +
  facet_wrap( ~ geochemical, scales = "free") +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Reverse y and set axis ticks
  scale_y_reverse(labels = c(10, 100, 150, 200),
                  breaks = c(10, 100, 150, 200)) +
  ## Rename axes
  labs(y = "Depth (m)", x = "Concentration")
```

```{r fig.cap = "Cruise 72 by depth"}
geochem %>%
  gather(key = "geochemical", value = "concentration", O2:PO4) %>%
  mutate(geochemical = fct_relevel(geochemical, "O2", "CH4", "NO3", "NO2",
                                   "N2O", "NH4", "H2S", "PO4")) %>%
  mutate(group = ifelse(geochemical %in% c("O2", "CH4"), "high", "low")) %>%
  
  ggplot(aes(x = geochemical, y = concentration)) +
  geom_col(aes(fill = geochemical)) +
  facet_grid(group ~ Depth_m, scales = "free") +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Rename axes
  labs(x = "", y = "Concentration") +
  theme(axis.text.x = element_blank(),
        legend.position = "bottom")
```

## Overall abundance of genes

```{r fig.cap = "Total overall abundance of genes"}
mark %>%
  # Calculate total abundance for DNA or RNA (sample) for each gene at each depth
  group_by(Marker, depth, sample) %>%
  summarize(abund_tot = sum(Abundance)) %>%
  
  # Plot total DNA and RNA abundance by depth and gene
  ggplot(aes(x = Marker, y = depth, color = sample)) +
  # Plot abundance by size. Change 0s to NAs so no dot appears
  geom_point(aes(size = ifelse(abund_tot == 0, NA, abund_tot)),
             # Put DNA and RNA side by side
             position = position_dodge(width = 0.5)) +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Reverse y and set axis ticks
  scale_y_reverse(
    labels = c(10, 100, 120, 135, 150, 165, 200),
    breaks = c(10, 100, 120, 135, 150, 165, 200)
  ) +
  ## Remove color legend title
  scale_color_discrete(name = "") +
  ## Rename size legend and axes
  scale_size_continuous(name = "Abundance") +
  labs(y = "Depth (m)", x = "")
```

```{r fig.cap = "Relative overall abundance of genes"}
mark %>%
  # Calculate total abundance for DNA or RNA (sample) for each gene at each depth
  group_by(Marker, depth, sample) %>%
  summarize(abund_tot = sum(Abundance)) %>%
  
  # Plot total DNA and RNA abundance by depth and gene
  ggplot(aes(y = abund_tot, x = Marker, fill = sample)) +
  # Plot relative abundance
  geom_bar(position = "fill", stat = "identity") +
  facet_grid( ~ depth) +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 16) +
  ## Remove color legend title
  scale_fill_discrete(name = "") +
  ## Rename and alter labels
  labs(y = "Relative abundance", x = "") +
  theme(axis.text.x = element_text(angle = 45,  hjust = 1, size = 10),
        legend.position = "bottom")
```

## Diversity

```{r fig.cap = "Diversity within depth"}
alpha_diversity %>% 
  filter(sample == "DNA") %>% 

  # Plot diversity by gene and depth
  ggplot(aes(y = shannon, x = Marker, color = Marker)) +
  geom_point(size = 2) +
  facet_wrap( ~ depth) +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Rename axes
  labs(y = "Diversity (Shannon)", x = "") +
  ## Rotate x-axis labels. Remove legend
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")
```

```{r fig.cap = "Diversity across depth"}
alpha_diversity %>%
  filter(sample == "DNA") %>% 

  # Plot diversity by gene and depth
  ggplot(aes(y = depth, x = shannon, color = Marker)) +
  geom_point(size = 2) +
  geom_path(size = 1) +
  facet_wrap(~ Marker) +
  #Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Reverse y and set axis ticks
  scale_y_reverse(labels = c(10, 100, 150, 200),
                  breaks = c(10, 100, 150, 200)) +
  ## Rename axes
  labs(y = "Depth (m)", x = "Diversity (Shannon)") +
  ## Remove legend
  theme(legend.position = "none")
```

## Relative abundance of kingdoms

```{r fig.cap = "Relative abundance of kingdoms"}
mark %>%
  # Select variables
  select(Marker, sample, Abundance, kingdom, phylum) %>%
  # Combine all classified phyla
  mutate(classified = ifelse(is.na(kingdom),
                             "Unclassified",
                             ifelse(
                               is.na(phylum),
                               paste("Unclassified", kingdom),
                               paste("Classified", kingdom)
                             ))) %>%
  # Calculate total abundance for DNA or RNA (sample) for each gene for each
  # kingdom
  group_by(Marker, sample, classified) %>%
  summarize(abund_tot = sum(Abundance)) %>% 
  
  # Plot total DNA and RNA abundance by depth and gene
  ggplot(aes(y = abund_tot, x = Marker, fill = classified)) +
  # Plot relative  abundance
  geom_col(position = "fill") +
  facet_grid(~ sample) +
  # Beautify plot
  ## Enlarge font
  theme_classic(base_size = 18) +
  ## Remove color legend title
  scale_fill_discrete(name = "") +
  ## Rename and labels
  labs(y = "Relative abundance", x = "") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

## Total abundance by phyla

```{r fig.cap = "Metagenomic abundance of phyla (up  to 135 m depth)", fig.height = 6}
plot_phyla_abundances(mark, "DNA", lower_depths)
```

```{r fig.cap = "Metagenomic abundance of phyla (150 m depth and below)", fig.height = 6}
plot_phyla_abundances(mark, "DNA", higher_depths)
```

```{r fig.cap = "Metatranscriptomic abundance of phyla (up  to 135 m depth)", fig.height = 6}
plot_phyla_abundances(mark, "RNA", lower_depths)
```

```{r fig.cap = "Metatranscriptomic abundance of phyla (150 m depth and below)", fig.height = 6}
plot_phyla_abundances(mark, "RNA", higher_depths)
```

## Abundance of Proteobacteria DNA

```{r fig.cap = "Metagenomic abundance of Proteobacteria"}
plot_proteobacteria(mark, "DNA")
```

```{r fig.cap = "Metatranscriptomic abundance of Proteobacteria"}
plot_proteobacteria(mark, "RNA")
```

```{r eval = TRUE, echo = FALSE, warning = FALSE, message = FALSE}
# Reset code chunk settings to `default_chunk_options` (defined in `index.Rmd`)
knitr::opts_chunk$set(default_chunk_options)
```
