<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>A The Saanich Inlet data set | Gene-centric mapping of microbial community structure and function</title>
  <meta name="description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="A The Saanich Inlet data set | Gene-centric mapping of microbial community structure and function" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="A The Saanich Inlet data set | Gene-centric mapping of microbial community structure and function" />
  
  <meta name="twitter:description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  

<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="treesapp-create.html"/>
<link rel="next" href="treesapp-reference-packages.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html"><i class="fa fa-check"></i><b>1</b> Exploring the microcosmos</a><ul>
<li class="chapter" data-level="1.1" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#background"><i class="fa fa-check"></i><b>1.1</b> Background</a></li>
<li class="chapter" data-level="1.2" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#using-multi-omic-data-to-reconstruct-distributed-networks-of-metabolite-exchange"><i class="fa fa-check"></i><b>1.2</b> Using multi-omic data to reconstruct distributed networks of metabolite exchange</a></li>
<li class="chapter" data-level="1.3" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#treesapp"><i class="fa fa-check"></i><b>1.3</b> TreeSAPP</a></li>
</ul></li>
<li class="part"><span><b>II Tutorials</b></span></li>
<li class="chapter" data-level="2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html"><i class="fa fa-check"></i><b>2</b> A tutorial for using TreeSAPP</a><ul>
<li class="chapter" data-level="2.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#a-typical-treesapp-workflow"><i class="fa fa-check"></i><b>2.1</b> A typical TreeSAPP workflow</a></li>
<li class="chapter" data-level="2.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#genes-for-used-for-creating-reference-packages"><i class="fa fa-check"></i><b>2.2</b> Genes for used for creating reference packages</a><ul>
<li class="chapter" data-level="2.2.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#xmoa"><i class="fa fa-check"></i><b>2.2.1</b> XmoA</a></li>
<li class="chapter" data-level="2.2.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#group-assigned-genes"><i class="fa fa-check"></i><b>2.2.2</b> Group-assigned genes</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#tools"><i class="fa fa-check"></i><b>2.3</b> Tools</a><ul>
<li class="chapter" data-level="2.3.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#shell"><i class="fa fa-check"></i><b>2.3.1</b> Shell</a></li>
<li class="chapter" data-level="2.3.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#treesapp-1"><i class="fa fa-check"></i><b>2.3.2</b> TreeSAPP</a></li>
<li class="chapter" data-level="2.3.3" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#itol"><i class="fa fa-check"></i><b>2.3.3</b> iTOL</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#data"><i class="fa fa-check"></i><b>2.4</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html"><i class="fa fa-check"></i><b>3</b> Acquiring golden data</a><ul>
<li class="chapter" data-level="3.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#the-golden-standard"><i class="fa fa-check"></i><b>3.1</b> The golden standard</a></li>
<li class="chapter" data-level="3.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#what-databases-can-i-retrieve-sequences-from"><i class="fa fa-check"></i><b>3.2</b> What databases can I retrieve sequences from?</a></li>
<li class="chapter" data-level="3.3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#xmoa-example"><i class="fa fa-check"></i><b>3.3</b> XmoA example</a></li>
<li class="chapter" data-level="3.4" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#alternative-sources"><i class="fa fa-check"></i><b>3.4</b> Alternative sources</a></li>
<li class="chapter" data-level="3.5" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#data-access-hazards"><i class="fa fa-check"></i><b>3.5</b> Data access hazards</a></li>
<li class="chapter" data-level="3.6" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#validation"><i class="fa fa-check"></i><b>3.6</b> Validation</a><ul>
<li class="chapter" data-level="3.6.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#basic"><i class="fa fa-check"></i><b>3.6.1</b> Basic</a></li>
<li class="chapter" data-level="3.6.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#advanced"><i class="fa fa-check"></i><b>3.6.2</b> Advanced</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="treesapp-create.html"><a href="treesapp-create.html"><i class="fa fa-check"></i><b>4</b> Creating a Reference Package For TreeSAPP</a><ul>
<li class="chapter" data-level="4.1" data-path="treesapp-create.html"><a href="treesapp-create.html#introduction-and-goals"><i class="fa fa-check"></i><b>4.1</b> Introduction and Goals</a></li>
<li class="chapter" data-level="4.2" data-path="treesapp-create.html"><a href="treesapp-create.html#making-the-reference-package"><i class="fa fa-check"></i><b>4.2</b> Making the reference package</a></li>
<li class="chapter" data-level="4.3" data-path="treesapp-create.html"><a href="treesapp-create.html#testing-the-purity-of-the-reference-package"><i class="fa fa-check"></i><b>4.3</b> Testing the purity of the reference package</a></li>
<li class="chapter" data-level="4.4" data-path="treesapp-create.html"><a href="treesapp-create.html#looking-at-tree-of-reference-package-with-itol"><i class="fa fa-check"></i><b>4.4</b> Looking at tree of reference package with iTOL</a></li>
<li class="chapter" data-level="4.5" data-path="treesapp-create.html"><a href="treesapp-create.html#building-your-groups-reference-package"><i class="fa fa-check"></i><b>4.5</b> Building your group’s reference package</a></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html"><i class="fa fa-check"></i><b>A</b> The Saanich Inlet data set</a><ul>
<li class="chapter" data-level="A.1" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#description"><i class="fa fa-check"></i><b>A.1</b> Description</a></li>
<li class="chapter" data-level="A.2" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#data-on-the-server"><i class="fa fa-check"></i><b>A.2</b> Data on the server</a></li>
<li class="chapter" data-level="A.3" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#description-of-input-data"><i class="fa fa-check"></i><b>A.3</b> Description of input data</a></li>
<li class="chapter" data-level="A.4" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#methods"><i class="fa fa-check"></i><b>A.4</b> Methods</a><ul>
<li class="chapter" data-level="A.4.1" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#quality-filtering-of-reads"><i class="fa fa-check"></i><b>A.4.1</b> Quality filtering of reads</a></li>
<li class="chapter" data-level="A.4.2" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#assembly"><i class="fa fa-check"></i><b>A.4.2</b> Assembly</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="B" data-path="treesapp-reference-packages.html"><a href="treesapp-reference-packages.html"><i class="fa fa-check"></i><b>B</b> TreeSAPP reference packages</a></li>
<li class="chapter" data-level="C" data-path="shell-cheat-sheet.html"><a href="shell-cheat-sheet.html"><i class="fa fa-check"></i><b>C</b> Shell cheat sheet</a></li>
<li class="chapter" data-level="D" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>D</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Gene-centric mapping of microbial community structure and function</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="the-saanich-inlet-data-set" class="section level1">
<h1><span class="header-section-number">A</span> The Saanich Inlet data set</h1>
<div id="description" class="section level2">
<h2><span class="header-section-number">A.1</span> Description</h2>
<p>We will work with real-world data collected as part of an ongoing oceanographic time series program in Saanich Inlet, a seasonally anoxic fjord on the East coast of Vancouver Island, British Columbia:</p>
<div class="figure"><span id="fig:saanich-inlet"></span>
<img src="images/saanich_inlet_map.png" alt="The Saanich Inlet"  />
<p class="caption">
FIGURE A.1: The Saanich Inlet
</p>
</div>
<p>Figure <a href="the-saanich-inlet-data-set.html#fig:saanich-inlet">A.1</a> shows a map of Saanich Inlet indicating conventional sample collection stations (S1-S9). The data used in this tutorial (sourced from S3, Cruise 72) include various geochemical measurements and the genomic and transcriptomic data of microbial samples at depths 10, 100, 120, 135, 150, 165 and 200 m.</p>
<p>For more details about these data, see <span class="citation">(<a href="references.html#ref-Hallam.2017" role="doc-biblioref">6</a>)</span>, and for more detailed information on the environmental context and time series data, see <span class="citation">(<a href="references.html#ref-Torres-Beltrán.2017" role="doc-biblioref">7</a>)</span>.</p>
</div>
<div id="data-on-the-server" class="section level2">
<h2><span class="header-section-number">A.2</span> Data on the server</h2>
<p>The data is located on the MICB425 server at <code>/mnt/datasets/saanich/</code>.</p>
<pre><code>/mnt/datasets/saanich/
├── [   0]  directory_structure.txt
├── [492M]  MAGs
│   └── [492M]  Concatenated
│       └── [492M]  All_SI072_Metawrap_MAGs.fa
├── [3.5G]  MetaG_Assemblies
│   ├── [630M]  SI072_100m_contig.fa
│   ├── [525M]  SI072_10m_contig.fa
│   ├── [571M]  SI072_120m_contig.fa
│   ├── [505M]  SI072_135m_contig.fa
│   ├── [396M]  SI072_150m_contig.fa
│   ├── [748M]  SI072_165m_contig.fa
│   └── [219M]  SI072_200m_contig.fa
├── [ 64G]  MetaG_Trim_QC_Reads
│   ├── [4.7G]  SI072_100m_pe.1.fq.gz
│   ├── [4.7G]  SI072_100m_pe.2.fq.gz
│   ├── [4.2G]  SI072_10m_pe.1.fq.gz
│   ├── [4.2G]  SI072_10m_pe.2.fq.gz
│   ├── [4.2G]  SI072_120m_pe.1.fq.gz
│   ├── [4.2G]  SI072_120m_pe.2.fq.gz
│   ├── [4.4G]  SI072_135m_pe.1.fq.gz
│   ├── [4.4G]  SI072_135m_pe.2.fq.gz
│   ├── [4.8G]  SI072_150m_pe.1.fq.gz
│   ├── [4.7G]  SI072_150m_pe.2.fq.gz
│   ├── [6.5G]  SI072_165m_pe.1.fq.gz
│   ├── [6.4G]  SI072_165m_pe.2.fq.gz
│   ├── [3.3G]  SI072_200m_pe.1.fq.gz
│   └── [3.3G]  SI072_200m_pe.2.fq.gz
├── [ 54G]  MetaT_Raw_Reads
│   ├── [8.9G]  SI072_100m_MetaT_QC_Filtered.fastq.gz
│   ├── [6.4G]  SI072_10m_MetaT_QC_Filtered.fastq.gz
│   ├── [6.8G]  SI072_120m_MetaT_QC_Filtered.fastq.gz
│   ├── [7.2G]  SI072_135m_MetaT_QC_Filtered.fastq.gz
│   ├── [10.0G]  SI072_150m_MetaT_QC_Filtered.fastq.gz
│   ├── [8.7G]  SI072_165m_MetaT_QC_Filtered.fastq.gz
│   └── [6.0G]  SI072_200m_MetaT_QC_Filtered.fastq.gz
├── [181M]  SAGs
│   └── [181M]  Concatenated_Med_Plus
│       └── [181M]  Saanich_Med_Plus_SAGs.fasta
└── [ 42K]  seq2lineage_Tables
    ├── [ 17K]  Med_Plus_SAGs_GTDB_Taxonomies.tsv
    └── [ 25K]  SI072_MAGs_All_GTDB_taxonomies.tsv

 122G used in 8 directories, 33 files</code></pre>
</div>
<div id="description-of-input-data" class="section level2">
<h2><span class="header-section-number">A.3</span> Description of input data</h2>
<p>List of directories and their contents</p>
<ul>
<li><code>MAGs</code></li>
<li><code>MetaG_Assemblies</code></li>
<li><code>MetaG_Trim_QC_Reads</code></li>
<li><code>MetaT_Raw_Reads</code></li>
<li><code>SAGs</code></li>
<li><code>seq2lineage_Tables</code></li>
</ul>
</div>
<div id="methods" class="section level2">
<h2><span class="header-section-number">A.4</span> Methods</h2>
<p>Water was collected from various depths at Saanich Inlet and filtered through a 0.22 µm Sterivex filter to collect biomass. Total genomic DNA and RNA were extracted from these filters. RNA was reversed transcribed into cDNA and both genomic DNA and cDNA were used in the construction of shotgun Illumina libraries. Sequencing data were generated on the Illumina HiSeq platform with <strong>2x150bp</strong> technology. For more in-depth sampling and sequencing methods, please see <span class="citation">(<a href="references.html#ref-Hawley.2017" role="doc-biblioref">8</a>)</span>.</p>
<div id="quality-filtering-of-reads" class="section level3">
<h3><span class="header-section-number">A.4.1</span> Quality filtering of reads</h3>
<p>The resulting reads were processed and quality controlled at the Joint Genome Institute using the IMG/M pipeline.</p>
<p>Quality filtering was performed with Trimmomatic (v.0.35, <a href="http://www.usadellab.org/cms/?page=trimmomatic">documentation</a>) <span class="citation">(<a href="references.html#ref-Bolger.2014" role="doc-biblioref">9</a>)</span> with default parameters except:</p>
<pre><code>LEADING:3 
TRAILING:3 
SLIDINGWINDOW:4:15 
MINLEN:36</code></pre>
</div>
<div id="assembly" class="section level3">
<h3><span class="header-section-number">A.4.2</span> Assembly</h3>
<p>MEGAHIT (v.1.1.3, <a href="https://github.com/voutcn/megahit/wiki/Assembly-Tips">documentation</a>) <span class="citation">(<a href="references.html#ref-Li.2015" role="doc-biblioref">10</a>)</span> assembled MAGs using with default parameters except for:</p>
<pre><code>--k-min 27 
--k-step 10 
--min-contig-len 500</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="treesapp-create.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="treesapp-reference-packages.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
