# Creating a Reference Package For TreeSAPP

## Introduction and Goals

This tutorial is meant for users who would like to analyze a gene family lacking an associated TreeSAPP reference package (refpkg). It is also useful to those who would rather start over than use `treesapp update` to improve an existing refpkg. This tutorial will cover the following subcommand of `treesapp`: `create`, `package`, `purity` and `assign`. TreeSAPP will automatically determine the taxonomic lineage assigned to each sequence if these are NCBI accessions. For ways to provide lineage information for as-of-yet unaccessions sequences please refer to our Wiki page [Building reference packages with TreeSAPP](https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#alternative-sources).

## Making the reference package

When building reference packages if you can source manually curated sequences---such as those hosted by SwissProt---always use those. There are plenty of sources for curated PmoA & AmoA sequences. More details on choosing databases for retrieving sequences can be found [here](https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#step-1-reference-sequences).

For this tutorial curated PmoA & AmoA sequences were downloaded from [TIGRFAM](http://tigrfams.jcvi.org/cgi-bin/index.cgi), [EggNOG v5.0](http://eggnog5.embl.de/#/app/home), and [FunGene](http://fungene.cme.msu.edu/). For now we'll just use the TIGRFAM seed sequences and EggNOG sequences as these are best curated. Let's call this our seed reference package. The steps involved are:

-   Connect to the server. We will move to the folder `/root/data` (if it is not your working directory already). Then we will create a new folder called `ts_tutorial` and move into it.

```{bash eval = FALSE}
cd /root/data/
mkdir ts_tutorial
cd ts_tutorial
```

-   Download the fasta files we'll be using to create, update and evaluate our XmoA reference package from [xmoa_file_list.txt](https://github.com/hallamlab/TreeSAPP/blob/master/docs/xmoa_file_list.txt) with wget then run wget again to retrieve each file within this list.

```{bash eval = FALSE}
wget https://raw.githubusercontent.com/hallamlab/TreeSAPP/master/docs/xmoa_file_list.txt`
wget -i xmoa_file_list.txt
```

There are 72 well curated PmoA and AmoA fasta sequences in these files

-   Confirm the number of sequences in the fasta file

```{bash eval = FALSE}
grep -c "^>" *faa
```

-   Create the input FASTA file

```{bash eval = FALSE}
cat ENOG5028JPK_EggNOGv5.faa TIGR03080.faa arCOG08676_EggNOGv5.faa > XmoA_seed.faa
```

-   Use submodule `treesapp create` to build the XmoA refpkg with argument. Visualize the subcommand's workflow [here](https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#ingredients).

To speed things along we'll be using FastTree to infer the phylogeny and will skip bootstrapping with `--fast`. To remove potential redundant sequences, we'll cluster the candidate reference sequences at 97% similarity using the argument `-p`. When clustering the candidate sequences TreeSAPP would normally ask which sequence to use for the representative of the cluster. This can be handy in cases when some sequences are better annotated and/or are especially important. To speed things up even more the flag `--headless` will prevent these requests. This command will take \~2 minutes to complete.

```{bash eval = FALSE}
treesapp create \
  --fast \
  --headless \
  --overwrite \
  --cluster \
  --trim_align \
  -n 4 \
  -m prot \
  -p 0.97 \
  --fastx_input XmoA_seed.faa \
  -c XmoA \
  --output XmoA_seed
```

The final reference package file is located in `XmoA_seed/final_outputs/XmoA_build.pkl`. This file contains all the individual components of a reference package (multiple sequence alignment, profile HMM, phylogenetic tree, taxonomic lineages) as well as some other data. These files were bundled up using the joblib Python library. They can be accessed individually using the submodule `treesapp package`.

-   Replace the current refpkg_code `Z1111` with unique identifier `N0102` (if you plan on just using this one reference package then you don't really need to worry about changing it).

```{bash eval = FALSE}
treesapp package edit \
  refpkg_code N0102 \
  --overwrite \
  --refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl
```

-   We can also modify the reference package's description while we're here.

```{bash eval = FALSE}
treesapp package edit \
  description "Alpha subunits of copper membrane monooxygenase enzymes" \
  --overwrite \
  --refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl
```

## Testing the purity of the reference package

We will determine whether there were any mis-annotated sequences that were included in our reference package with treesapp purity. To do this, we take a well-curated database and attempt to classify sequences in it using `treesapp assign`. The results are then analysed and displayed for the user to evaluate.The TIGRFAM database is fairly comprehensive, representing 4488 different groups in version 15, and by using just the TIGRFAM seed sequences (`TIGRFAM_seed_named.faa`) we can be fairly sure we won't be evaluating our classifications with mis-annotated sequences.

```{bash eval = FALSE}
treesapp purity \
  --trim_align \
  -m prot \
  -n 4 \
  -r XmoA_seed/final_outputs/XmoA_build.pkl \
  --extra_info TIGRFAM_info.tsv \
  -i TIGRFAM_seed_named.faa \
  --output XmoA_purity
```

The important bit of the output should look like this

    Summarizing assignments for reference package XmoA
    Ortholog        Hits    Leaves  Tree-coverage   Description
    --------------------------------------------------------------------------------
    TIGR03080       3       4       11.1    methane monooxygenase/ammonia monooxygenase, subunit A

From this summary it appears that the reference package classified three homologous sequences that were placed at leaf nodes in the tree (i.e. they're closely related) and the sequences were all from the family "TIGR03080", also known as "methane monooxygenase/ammonia monooxygenase, subunit A". In all, we're probably good to proceed!

## Looking at tree of reference package with iTOL

To do: Assign reference sequences to reference package to generate tree with labels

First off, we can quickly and easily view where the sequences were placed on the phylogeny using the interactive Tree of Life.

-   Transfer the directory containing the XmoA iTOL outputs `p_amoA_FunGene9.5_isolates_assign/iTOL_output/XmoA` to your desktop

```{bash eval = FALSE}
scp -r root@<server address>:data/ts_tutorial/p_amoA_FunGene9.5_isolates_assign/iTOL_output/XmoA  <path to your local directory>
```

-   Visit the [iTOL](https://itol.embl.de/) website.

-   Upload the file `XmoA_complete_profile.jplace` from the XmoA directory. This `jplace` file contains both the phylogeny and the location of the placements that will be visualized.

-   Navigate to the page displaying the phylogeny and click-and-drag the file XmoA_labels.txt into the iTOL window. This should convert the TreeSAPP identifiers (e.g. 1_XmoA) to more useful descriptions with the organism name and accession for each leaf node.

-   Finally, turn on the "Phylogenetic Placements" dataset (right of the screen) and the figure should look identical to FigureÂ \@ref(fig:xmoa).

![(\#fig:xmoa) Phylogenetic tree of XmoA <br> The XmoA phylogeny with FunGene sequences placed onto it. The red bubbles represent sequence placements with the size of the bubble indicating the number of placements on a branch. There were 74 sequences classified, 34 of which were placed on the clade representing members of the Gammaproteobacteria order Methylococcales. Another 15 sequences were placed on the clade of Betaproteobacteria, represented by the genera Nitrosospira and Nitrosomonas](images/iTOL_XmoA2.jpg)
