# Analysing the TreeSAPP classifications in R

# Load any required packages

<Load required packages>

# Load metagenomic and metatranscriptomic TreeSAPP data

<Load data sets>

# Subset your data to the variables and marker genes of interest

<Filter to just XmoA>
  
# Reformat variables
  
<Split the taxonomic information in the data frame into separate ranks>

<Split the sample name into cruise ID and distinguish metatranscriptomes from metagenomes>

<Modify the water column depth data>

# Load the geochemical data set into a new data frame.

<Load geochemicals>

# Combine the TreeSAPP data with the geochemical measurements

<Use left_join to combine the two data frames>

# Visualize the taxonomic, functional and chemical data from Saanich Inlet

<Distribution of abundances of XmoA taxonomic orders at different depths>

<Barplot demonstrates relative change in proportions at different depths>

<Analyze the functional data>

<Analyze the chemical measurements>
  
<Combine the classifications and chemical data in one plot>

# Analyzing classifications for your group's reference package

<You code follows>
