<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>B TreeSAPP reference packages | Gene-centric mapping of microbial community structure and function</title>
  <meta name="description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="B TreeSAPP reference packages | Gene-centric mapping of microbial community structure and function" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="B TreeSAPP reference packages | Gene-centric mapping of microbial community structure and function" />
  
  <meta name="twitter:description" content="A collection of tutorials and the capstone project information for MICB425 Module 04." />
  

<meta name="author" content="Stephan Koenig, Kim Dill-McFarland, Connor Morgan-Lang, Ryan McLaughlin, Julia Anstett, Resmi Radhamony, Sean Crowe and Steven Hallam" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="the-saanich-inlet-data-set.html"/>
<link rel="next" href="shell-cheat-sheet.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="1" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html"><i class="fa fa-check"></i><b>1</b> Exploring the microcosmos</a><ul>
<li class="chapter" data-level="1.1" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#background"><i class="fa fa-check"></i><b>1.1</b> Background</a></li>
<li class="chapter" data-level="1.2" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#using-multi-omic-data-to-reconstruct-distributed-networks-of-metabolite-exchange"><i class="fa fa-check"></i><b>1.2</b> Using multi-omic data to reconstruct distributed networks of metabolite exchange</a></li>
<li class="chapter" data-level="1.3" data-path="exploring-the-microcosmos.html"><a href="exploring-the-microcosmos.html#treesapp"><i class="fa fa-check"></i><b>1.3</b> TreeSAPP</a></li>
</ul></li>
<li class="part"><span><b>II Tutorials</b></span></li>
<li class="chapter" data-level="2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html"><i class="fa fa-check"></i><b>2</b> A tutorial for using TreeSAPP</a><ul>
<li class="chapter" data-level="2.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#introduction-and-goals"><i class="fa fa-check"></i><b>2.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="2.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#a-typical-treesapp-workflow"><i class="fa fa-check"></i><b>2.2</b> A typical TreeSAPP workflow</a></li>
<li class="chapter" data-level="2.3" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#genes-for-creating-reference-packages"><i class="fa fa-check"></i><b>2.3</b> Genes for creating reference packages</a><ul>
<li class="chapter" data-level="2.3.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#xmoa"><i class="fa fa-check"></i><b>2.3.1</b> XmoA</a></li>
<li class="chapter" data-level="2.3.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#group-assigned-genes"><i class="fa fa-check"></i><b>2.3.2</b> Group-assigned genes</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#tools"><i class="fa fa-check"></i><b>2.4</b> Tools</a><ul>
<li class="chapter" data-level="2.4.1" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#shell"><i class="fa fa-check"></i><b>2.4.1</b> Shell</a></li>
<li class="chapter" data-level="2.4.2" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#treesapp-1"><i class="fa fa-check"></i><b>2.4.2</b> TreeSAPP</a></li>
<li class="chapter" data-level="2.4.3" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#itol"><i class="fa fa-check"></i><b>2.4.3</b> iTOL</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="a-tutorial-for-using-treesapp.html"><a href="a-tutorial-for-using-treesapp.html#data"><i class="fa fa-check"></i><b>2.5</b> Data</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html"><i class="fa fa-check"></i><b>3</b> Acquiring golden data</a><ul>
<li class="chapter" data-level="3.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#the-golden-standard"><i class="fa fa-check"></i><b>3.1</b> The golden standard</a></li>
<li class="chapter" data-level="3.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#what-databases-can-i-retrieve-sequences-from"><i class="fa fa-check"></i><b>3.2</b> What databases can I retrieve sequences from?</a></li>
<li class="chapter" data-level="3.3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#xmoa-example"><i class="fa fa-check"></i><b>3.3</b> XmoA example</a><ul>
<li class="chapter" data-level="3.3.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#fungene"><i class="fa fa-check"></i><b>3.3.1</b> FunGene</a></li>
<li class="chapter" data-level="3.3.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#eggnog"><i class="fa fa-check"></i><b>3.3.2</b> EggNOG</a></li>
<li class="chapter" data-level="3.3.3" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#copying-data-onto-the-server"><i class="fa fa-check"></i><b>3.3.3</b> Copying data onto the server</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#alternative-sources"><i class="fa fa-check"></i><b>3.4</b> Alternative sources</a></li>
<li class="chapter" data-level="3.5" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#data-access-hazards"><i class="fa fa-check"></i><b>3.5</b> Data access hazards</a></li>
<li class="chapter" data-level="3.6" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#validation"><i class="fa fa-check"></i><b>3.6</b> Validation</a><ul>
<li class="chapter" data-level="3.6.1" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#basic"><i class="fa fa-check"></i><b>3.6.1</b> Basic</a></li>
<li class="chapter" data-level="3.6.2" data-path="acquiring-golden-data.html"><a href="acquiring-golden-data.html#advanced"><i class="fa fa-check"></i><b>3.6.2</b> Advanced</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="treesapp-create.html"><a href="treesapp-create.html"><i class="fa fa-check"></i><b>4</b> Creating a reference package for TreeSAPP</a><ul>
<li class="chapter" data-level="4.1" data-path="treesapp-create.html"><a href="treesapp-create.html#introduction-and-goals-1"><i class="fa fa-check"></i><b>4.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="4.2" data-path="treesapp-create.html"><a href="treesapp-create.html#making-the-reference-package"><i class="fa fa-check"></i><b>4.2</b> Making the reference package</a></li>
<li class="chapter" data-level="4.3" data-path="treesapp-create.html"><a href="treesapp-create.html#testing-the-purity-of-the-reference-package"><i class="fa fa-check"></i><b>4.3</b> Testing the purity of the reference package</a></li>
<li class="chapter" data-level="4.4" data-path="treesapp-create.html"><a href="treesapp-create.html#viewing-the-reference-package-tree-with-itol"><i class="fa fa-check"></i><b>4.4</b> Viewing the reference package tree with iTOL</a></li>
<li class="chapter" data-level="4.5" data-path="treesapp-create.html"><a href="treesapp-create.html#building-your-groups-reference-package"><i class="fa fa-check"></i><b>4.5</b> Building your group’s reference package</a><ul>
<li class="chapter" data-level="4.5.1" data-path="treesapp-create.html"><a href="treesapp-create.html#booster-shot"><i class="fa fa-check"></i><b>4.5.1</b> Booster shot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><i class="fa fa-check"></i><b>5</b> Classifying unknown sequences with TreeSAPP and updating a reference package</a><ul>
<li class="chapter" data-level="5.1" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#introduction-and-goals-2"><i class="fa fa-check"></i><b>5.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="5.2" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#change-notes"><i class="fa fa-check"></i><b>5.2</b> Change notes</a></li>
<li class="chapter" data-level="5.3" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#classify-new-sequences"><i class="fa fa-check"></i><b>5.3</b> Classify new sequences</a><ul>
<li class="chapter" data-level="5.3.1" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#classifying-amino-acid-sequences"><i class="fa fa-check"></i><b>5.3.1</b> Classifying amino acid sequences</a></li>
<li class="chapter" data-level="5.3.2" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#classifying-orfs-predicted-from-genomes"><i class="fa fa-check"></i><b>5.3.2</b> Classifying ORFs predicted from genomes</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#update-the-reference-package-with-new-sequences"><i class="fa fa-check"></i><b>5.4</b> Update the reference package with new sequences</a><ul>
<li class="chapter" data-level="5.4.1" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#updating-with-publicly-available-sequences"><i class="fa fa-check"></i><b>5.4.1</b> Updating with publicly available sequences</a></li>
<li class="chapter" data-level="5.4.2" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#updating-with-mag-sequences"><i class="fa fa-check"></i><b>5.4.2</b> Updating with MAG sequences</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#annotate-clades"><i class="fa fa-check"></i><b>5.5</b> Annotate clades</a></li>
<li class="chapter" data-level="5.6" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#reclassify-and-layer-annotations-onto-classification-table"><i class="fa fa-check"></i><b>5.6</b> Reclassify and layer annotations onto classification table</a></li>
<li class="chapter" data-level="5.7" data-path="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html"><a href="classifying-unknown-sequences-with-treesapp-and-updating-a-reference-package.html#use-your-groups-reference-package-to-classify-unkown-sequences-then-update-your-package"><i class="fa fa-check"></i><b>5.7</b> Use your group’s reference package to classify unkown sequences then update your package</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html"><i class="fa fa-check"></i><b>6</b> Calculating relative abundance of meta’omic data</a><ul>
<li class="chapter" data-level="6.1" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#introduction-and-goals-3"><i class="fa fa-check"></i><b>6.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="6.2" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#workflow-description"><i class="fa fa-check"></i><b>6.2</b> Workflow description</a></li>
<li class="chapter" data-level="6.3" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#usage"><i class="fa fa-check"></i><b>6.3</b> Usage</a></li>
<li class="chapter" data-level="6.4" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#outputs"><i class="fa fa-check"></i><b>6.4</b> Outputs</a></li>
<li class="chapter" data-level="6.5" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#tutorial-steps"><i class="fa fa-check"></i><b>6.5</b> Tutorial steps</a></li>
<li class="chapter" data-level="6.6" data-path="calculating-relative-abundance-of-metaomic-data.html"><a href="calculating-relative-abundance-of-metaomic-data.html#use-your-groups-reference-package-to-calculate-abundances"><i class="fa fa-check"></i><b>6.6</b> Use your group’s reference package to calculate abundances</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html"><i class="fa fa-check"></i><b>7</b> Analysing the TreeSAPP classifications in R</a><ul>
<li class="chapter" data-level="7.1" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#introduction-and-goals-4"><i class="fa fa-check"></i><b>7.1</b> Introduction and goals</a></li>
<li class="chapter" data-level="7.2" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#resources-for-analysis-of-treesapp-output-in-r"><i class="fa fa-check"></i><b>7.2</b> Resources for analysis of TreeSAPP output in R</a></li>
<li class="chapter" data-level="7.3" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#checklist-to-write-and-run-r-script"><i class="fa fa-check"></i><b>7.3</b> Checklist to write and run R script</a></li>
<li class="chapter" data-level="7.4" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#writing-the-r-script"><i class="fa fa-check"></i><b>7.4</b> Writing the R script</a></li>
<li class="chapter" data-level="7.5" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#load-any-required-packages"><i class="fa fa-check"></i><b>7.5</b> Load any required packages</a></li>
<li class="chapter" data-level="7.6" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#load-metagenomic-and-metatranscriptomic-treesapp-data"><i class="fa fa-check"></i><b>7.6</b> Load metagenomic and metatranscriptomic TreeSAPP data</a><ul>
<li class="chapter" data-level="7.6.1" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#classification-table"><i class="fa fa-check"></i><b>7.6.1</b> Classification table</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#subset-your-data-to-the-variables-and-marker-genes-of-interest"><i class="fa fa-check"></i><b>7.7</b> Subset your data to the variables and marker genes of interest</a></li>
<li class="chapter" data-level="7.8" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#reformat-variables"><i class="fa fa-check"></i><b>7.8</b> Reformat variables</a><ul>
<li class="chapter" data-level="7.8.1" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#processing-taxonomic-information"><i class="fa fa-check"></i><b>7.8.1</b> Processing taxonomic information</a></li>
<li class="chapter" data-level="7.8.2" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#distinguishing-metagenomes-from-metatranscriptomes"><i class="fa fa-check"></i><b>7.8.2</b> Distinguishing metagenomes from metatranscriptomes</a></li>
<li class="chapter" data-level="7.8.3" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#processing-depth-data"><i class="fa fa-check"></i><b>7.8.3</b> Processing depth data</a></li>
</ul></li>
<li class="chapter" data-level="7.9" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#load-the-geochemical-data"><i class="fa fa-check"></i><b>7.9</b> Load the geochemical data</a></li>
<li class="chapter" data-level="7.10" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#visualize-the-taxonomic-functional-and-chemical-data-from-saanich-inlet"><i class="fa fa-check"></i><b>7.10</b> Visualize the taxonomic, functional and chemical data from Saanich Inlet</a><ul>
<li class="chapter" data-level="7.10.1" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#analyze-treesapps-taxonomic-classifications"><i class="fa fa-check"></i><b>7.10.1</b> Analyze TreeSAPP’s taxonomic classifications</a></li>
<li class="chapter" data-level="7.10.2" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#analyze-the-functional-data"><i class="fa fa-check"></i><b>7.10.2</b> Analyze the functional data</a></li>
<li class="chapter" data-level="7.10.3" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#analyze-the-chemical-measurements"><i class="fa fa-check"></i><b>7.10.3</b> Analyze the chemical measurements</a></li>
<li class="chapter" data-level="7.10.4" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#bringing-them-all-together"><i class="fa fa-check"></i><b>7.10.4</b> Bringing them all together</a></li>
</ul></li>
<li class="chapter" data-level="7.11" data-path="analysing-the-treesapp-classifications-in-r.html"><a href="analysing-the-treesapp-classifications-in-r.html#analyzing-classifications-for-your-groups-reference-package"><i class="fa fa-check"></i><b>7.11</b> Analyzing classifications for your group’s reference package</a></li>
</ul></li>
<li class="part"><span><b>III Capstone project</b></span></li>
<li class="chapter" data-level="8" data-path="project-description.html"><a href="project-description.html"><i class="fa fa-check"></i><b>8</b> Project description</a><ul>
<li class="chapter" data-level="8.1" data-path="project-description.html"><a href="project-description.html#guiding-research-questions"><i class="fa fa-check"></i><b>8.1</b> Guiding research questions</a><ul>
<li class="chapter" data-level="8.1.1" data-path="project-description.html"><a href="project-description.html#questions-for-project-with-a-focus-on-biogeochemical-cycles-repeated-in-report-structure-below"><i class="fa fa-check"></i><b>8.1.1</b> Questions for project with a focus on biogeochemical cycles (repeated in report structure below)</a></li>
<li class="chapter" data-level="8.1.2" data-path="project-description.html"><a href="project-description.html#questions-for-project-with-a-focus-on-using-all-reference-packages"><i class="fa fa-check"></i><b>8.1.2</b> Questions for project with a focus on using all reference packages</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="project-description.html"><a href="project-description.html#resources"><i class="fa fa-check"></i><b>8.2</b> Resources</a></li>
<li class="chapter" data-level="8.3" data-path="project-description.html"><a href="project-description.html#your-submission"><i class="fa fa-check"></i><b>8.3</b> Your submission</a></li>
<li class="chapter" data-level="8.4" data-path="project-description.html"><a href="project-description.html#timeline"><i class="fa fa-check"></i><b>8.4</b> Timeline</a></li>
<li class="chapter" data-level="8.5" data-path="project-description.html"><a href="project-description.html#reports"><i class="fa fa-check"></i><b>8.5</b> Reports</a></li>
<li class="chapter" data-level="8.6" data-path="project-description.html"><a href="project-description.html#addendum"><i class="fa fa-check"></i><b>8.6</b> Addendum</a><ul>
<li class="chapter" data-level="8.6.1" data-path="project-description.html"><a href="project-description.html#abundance"><i class="fa fa-check"></i><b>8.6.1</b> Abundance</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="assessment.html"><a href="assessment.html"><i class="fa fa-check"></i><b>9</b> Assessment</a><ul>
<li class="chapter" data-level="9.1" data-path="assessment.html"><a href="assessment.html#peer-evaluation-10"><i class="fa fa-check"></i><b>9.1</b> Peer evaluation (10%)</a></li>
<li class="chapter" data-level="9.2" data-path="assessment.html"><a href="assessment.html#report-20"><i class="fa fa-check"></i><b>9.2</b> Report (20%)</a></li>
<li class="chapter" data-level="9.3" data-path="assessment.html"><a href="assessment.html#ujemi-submission"><i class="fa fa-check"></i><b>9.3</b> UJEMI submission</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html"><i class="fa fa-check"></i><b>10</b> Running TreeSAPP on a server</a><ul>
<li class="chapter" data-level="10.1" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#resources-for-treesapp-analysis"><i class="fa fa-check"></i><b>10.1</b> Resources for TreeSAPP analysis</a></li>
<li class="chapter" data-level="10.2" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#checklist-to-write-and-run-shell-script"><i class="fa fa-check"></i><b>10.2</b> Checklist to write and run shell script</a></li>
<li class="chapter" data-level="10.3" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#shell-writing"><i class="fa fa-check"></i><b>10.3</b> Writing the shell script</a><ul>
<li class="chapter" data-level="10.3.1" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#defining-the-shell"><i class="fa fa-check"></i><b>10.3.1</b> Defining the shell</a></li>
<li class="chapter" data-level="10.3.2" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#setting-a-variable-for-your-depth"><i class="fa fa-check"></i><b>10.3.2</b> Setting a variable for your depth</a></li>
<li class="chapter" data-level="10.3.3" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#providing-user-feedback"><i class="fa fa-check"></i><b>10.3.3</b> Providing user feedback</a></li>
<li class="chapter" data-level="10.3.4" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#defining-variable-for-input-directory"><i class="fa fa-check"></i><b>10.3.4</b> Defining variable for input directory</a></li>
<li class="chapter" data-level="10.3.5" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#creating-output-directory"><i class="fa fa-check"></i><b>10.3.5</b> Creating output directory</a></li>
<li class="chapter" data-level="10.3.6" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#reporting-variables-to-user"><i class="fa fa-check"></i><b>10.3.6</b> Reporting variables to user</a></li>
<li class="chapter" data-level="10.3.7" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#listing-treesapp-version"><i class="fa fa-check"></i><b>10.3.7</b> Listing TreeSAPP version</a></li>
<li class="chapter" data-level="10.3.8" data-path="running-treesapp-on-a-server.html"><a href="running-treesapp-on-a-server.html#executing-treesapp-analysis"><i class="fa fa-check"></i><b>10.3.8</b> Executing TreeSAPP analysis</a></li>
</ul></li>
</ul></li>
<li class="appendix"><span><b>Appendix</b></span></li>
<li class="chapter" data-level="A" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html"><i class="fa fa-check"></i><b>A</b> The Saanich Inlet data set</a><ul>
<li class="chapter" data-level="A.1" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#description"><i class="fa fa-check"></i><b>A.1</b> Description</a></li>
<li class="chapter" data-level="A.2" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#data-on-the-server"><i class="fa fa-check"></i><b>A.2</b> Data on the server</a></li>
<li class="chapter" data-level="A.3" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#description-of-input-data"><i class="fa fa-check"></i><b>A.3</b> Description of input data</a></li>
<li class="chapter" data-level="A.4" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#methods"><i class="fa fa-check"></i><b>A.4</b> Methods</a><ul>
<li class="chapter" data-level="A.4.1" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#metagenomes-and-metatransriptopmes"><i class="fa fa-check"></i><b>A.4.1</b> Metagenomes and metatransriptopmes</a></li>
<li class="chapter" data-level="A.4.2" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#single-cell-amplified-genomes"><i class="fa fa-check"></i><b>A.4.2</b> Single-Cell Amplified Genomes</a></li>
</ul></li>
<li class="chapter" data-level="A.5" data-path="the-saanich-inlet-data-set.html"><a href="the-saanich-inlet-data-set.html#installing-and-running-gtdb-tk"><i class="fa fa-check"></i><b>A.5</b> Installing and running GTDB-Tk</a></li>
</ul></li>
<li class="chapter" data-level="B" data-path="treesapp-reference-packages.html"><a href="treesapp-reference-packages.html"><i class="fa fa-check"></i><b>B</b> TreeSAPP reference packages</a></li>
<li class="chapter" data-level="C" data-path="shell-cheat-sheet.html"><a href="shell-cheat-sheet.html"><i class="fa fa-check"></i><b>C</b> Shell cheat sheet</a></li>
<li class="chapter" data-level="D" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>D</b> References</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Gene-centric mapping of microbial community structure and function</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="treesapp-reference-packages" class="section level1">
<h1><span class="header-section-number">B</span> TreeSAPP reference packages</h1>
<!-- Current list of TreeSAPP reference packages, get newest version from https://github.com/hallamlab/RefPkgs/wiki/refpkgs -->
<table>
<caption><span id="tab:reference-packages">TABLE B.1: </span> Available TreeSAPP reference packages</caption>
<colgroup>
<col width="44%" />
<col width="31%" />
<col width="4%" />
<col width="16%" />
<col width="2%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Name</th>
<th align="left">Description</th>
<th align="left">Leaf nodes</th>
<th align="left">Pathway</th>
<th align="left">Cycle</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/AA10/seed_refpkg/final_outputs">AA10</a></td>
<td align="left">Auxiliary Activity family 10</td>
<td align="left">699</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH101/seed_refpkg/final_outputs">GH101</a></td>
<td align="left">Glycoside hydrolase family 101</td>
<td align="left">113</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH102/seed_refpkg/final_outputs">GH102</a></td>
<td align="left">Glycoside hydrolase family 102</td>
<td align="left">727</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH103/seed_refpkg/final_outputs">GH103</a></td>
<td align="left">Glycoside hydrolase family 103</td>
<td align="left">1145</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH109/seed_refpkg/final_outputs">GH109</a></td>
<td align="left">Glycoside hydrolase family 109</td>
<td align="left">376</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH115/seed_refpkg/final_outputs">GH115</a></td>
<td align="left">Glycoside hydrolase family 115</td>
<td align="left">267</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH126/seed_refpkg/final_outputs">GH126</a></td>
<td align="left">Glycoside hydrolase family 126</td>
<td align="left">112</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH163/seed_refpkg/final_outputs">GH163</a></td>
<td align="left">Glycoside hydrolase family 163</td>
<td align="left">102</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH67/seed_refpkg/final_outputs">GH67</a></td>
<td align="left">Glycoside hydrolase family 67</td>
<td align="left">372</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/GH94/seed_refpkg/final_outputs">GH94</a></td>
<td align="left">Glycoside hydrolase family 94</td>
<td align="left">695</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Methanogenesis/McrA/MAG_update_RaxML-NG/final_outputs">McrA</a></td>
<td align="left">Methyl coenzyme M reductase alpha subunit</td>
<td align="left">424</td>
<td align="left">Methanogenesis</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Methanogenesis/McrA/seed_refpkg/final_outputs">McrA</a></td>
<td align="left">Methyl coenzyme M reductase alpha subunit</td>
<td align="left">249</td>
<td align="left">Methanogenesis</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Methanogenesis/McrB/seed_refpkg/final_outputs">McrB</a></td>
<td align="left">Methyl coenzyme M reductase beta subunit</td>
<td align="left">110</td>
<td align="left">Methanogenesis</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Methanogenesis/McrG/seed_refpkg/final_outputs">McrG</a></td>
<td align="left">Methyl coenzyme M reductase gamma subunit</td>
<td align="left">123</td>
<td align="left">Methanogenesis</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Hydrocarbons/PHA_degradation/PHA_depolymerases/final_outputs">PHAdeg</a></td>
<td align="left">Poly(hydroxyalkanoate) degradation enzymes</td>
<td align="left">10</td>
<td align="left">Plastics degradation</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/CAZymes/PL11/seed_refpkg/final_outputs">PL11</a></td>
<td align="left">Polysaccharide lyase family 11</td>
<td align="left">320</td>
<td align="left">Carbohydrate metabolism</td>
<td align="left">C</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Photosynthesis/PuhA/seed_refpkg/final_outputs">PuhA</a></td>
<td align="left">photosynthetic reaction center H subunit</td>
<td align="left">47</td>
<td align="left">Photosynthesis</td>
<td align="left">C</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Direct_Interspecies_Electron_Transfer/pillin/seed_refpkg/final_outputs">PilA</a></td>
<td align="left">Type IV pilin N-terminus, including PilA</td>
<td align="left">70</td>
<td align="left">Direct interspecies electron transfer</td>
<td align="left">E</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Hydrogen/HydA/seed_refpkg/final_outputs">HydA</a></td>
<td align="left">[FeFe] hydrogenase, group A, B1, B3</td>
<td align="left">93</td>
<td align="left">Hydrogenase</td>
<td align="left">H</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Mercury/HgcA/seed_refpkg/final_outputs">HgcA</a></td>
<td align="left">Methylmercury active enzyme, subunit A</td>
<td align="left">108</td>
<td align="left">NA</td>
<td align="left">Hg</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Mercury/HgcB/seed_refpkg/final_outputs">HgcB</a></td>
<td align="left">Methylmercury active enzyme, subunit B</td>
<td align="left">122</td>
<td align="left">NA</td>
<td align="left">Hg</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NapA/seed_refpkg/final_outputs">NapA</a></td>
<td align="left">Periplasmic nitrate reductase, large subunit</td>
<td align="left">765</td>
<td align="left">Denitrification</td>
<td align="left">N</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Fixation/NifD/seed_refpkg/final_outputs">NifD</a></td>
<td align="left">nitrogenase molybdenum-iron protein alpha chain</td>
<td align="left">926</td>
<td align="left">Nitrogen fixation</td>
<td align="left">N</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Fixation/NifH/seed_refpkg/final_outputs">NifH</a></td>
<td align="left">Nitrogenase iron protein [TIGR01287,EC:1.18.6.1]</td>
<td align="left">660</td>
<td align="left">Nitrogen fixation</td>
<td align="left">N</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NirK/seed_refpkg/final_outputs">NirK</a></td>
<td align="left">nitrite reductase (NO-forming)</td>
<td align="left">325</td>
<td align="left">Denitrification,Annamox</td>
<td align="left">N</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NirS/seed_refpkg/final_outputs">NirS</a></td>
<td align="left">nitrite reductase (NO-forming) / hydroxylamine reductase</td>
<td align="left">511</td>
<td align="left">Denitrification,Annamox</td>
<td align="left">N</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NorB/seed_refpkg/final_outputs">NorB</a></td>
<td align="left">nitric oxide reductase subunit B</td>
<td align="left">460</td>
<td align="left">Denitrification</td>
<td align="left">N</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NorC/seed_refpkg/final_outputs">NorC</a></td>
<td align="left">nitric oxide reductase subunit C</td>
<td align="left">173</td>
<td align="left">Denitrification</td>
<td align="left">N</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NosZ/seed_refpkg/final_outputs">NosZ</a></td>
<td align="left">Nitrous-oxide reductase (TAT-dependent|Sec-dependent)</td>
<td align="left">26</td>
<td align="left">Denitrification</td>
<td align="left">N</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Denitrification/NxrA/seed_refpkg/final_outputs">NxrA</a></td>
<td align="left">nitrate reductase / nitrite oxidoreductase, alpha subunit</td>
<td align="left">658</td>
<td align="left">Nitrification</td>
<td align="left">N</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Nitrification/NxrB/seed_refpkg/final_outputs">NxrB</a></td>
<td align="left">nitrate reductase / nitrite oxidoreductase, beta subunit</td>
<td align="left">648</td>
<td align="left">Nitrification</td>
<td align="left">N</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Nitrogen_metabolism/Nitrification/XmoA/XmoA_seed/final_outputs">XmoA</a></td>
<td align="left">Alpha subunits of copper membrane monooxygenase enzymes</td>
<td align="left">37</td>
<td align="left">Nitrification</td>
<td align="left">N</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00380/seed_refpkg_arc/final_outputs">PF00380a</a></td>
<td align="left">Ribosomal protein S9/S16</td>
<td align="left">82</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00380/seed_refpkg_bac/final_outputs">PF00380b</a></td>
<td align="left">Ribosomal protein S9/S16</td>
<td align="left">1065</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00410/seed_refpkg_arc/final_outputs">PF00410a</a></td>
<td align="left">Ribosomal protein S8</td>
<td align="left">155</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00410/seed_refpkg_bac/final_outputs">PF00410b</a></td>
<td align="left">Ribosomal protein S8</td>
<td align="left">1216</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00687/seed_refpkg_arc/final_outputs">PF00687a</a></td>
<td align="left">Ribosomal protein L1p/L10e family</td>
<td align="left">192</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00687/seed_refpkg_bac/final_outputs">PF00687b</a></td>
<td align="left">Ribosomal protein L1p/L10e family</td>
<td align="left">864</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00750/seed_refpkg_arc/final_outputs">PF00750a</a></td>
<td align="left">tRNA synthetases class I (R)</td>
<td align="left">193</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00750/seed_refpkg_bac/final_outputs">PF00750b</a></td>
<td align="left">tRNA synthetases class I (R)</td>
<td align="left">1317</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF00900/seed_refpkg/final_outputs">PF00900</a></td>
<td align="left">Ribosomal family S4e</td>
<td align="left">328</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01015/seed_refpkg/final_outputs">PF01015</a></td>
<td align="left">Ribosomal S3Ae family</td>
<td align="left">101</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01092/seed_refpkg/final_outputs">PF01092</a></td>
<td align="left">Ribosomal protein S6e</td>
<td align="left">169</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01157/seed_refpkg/final_outputs">PF01157</a></td>
<td align="left">Ribosomal protein L21e</td>
<td align="left">74</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01200/seed_refpkg/final_outputs">PF01200</a></td>
<td align="left">Ribosomal protein S28e</td>
<td align="left">129</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01280/seed_refpkg/final_outputs">PF01280</a></td>
<td align="left">Ribosomal protein L19e</td>
<td align="left">281</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01409/seed_refpkg_arc/final_outputs">PF01409a</a></td>
<td align="left">tRNA synthetases class II core domain (F)</td>
<td align="left">243</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01409/seed_refpkg_bac/final_outputs">PF01409b</a></td>
<td align="left">tRNA synthetases class II core domain (F)</td>
<td align="left">891</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01655/seed_refpkg/final_outputs">PF01655</a></td>
<td align="left">Ribosomal protein L32</td>
<td align="left">220</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/PF01866/seed_refpkg/final_outputs">PF01866</a></td>
<td align="left">Putative diphthamide synthesis protein</td>
<td align="left">229</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Translation/RpoB/seed_refpkg/final_outputs">RpoB</a></td>
<td align="left">RNA polymerase beta subunit</td>
<td align="left">697</td>
<td align="left">Translation</td>
<td align="left">NA</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/CysD/seed_refpkg/final_outputs">CysD</a></td>
<td align="left">sulfate adenylyltransferase (SAT), small subunit</td>
<td align="left">93</td>
<td align="left">Assimilatory sulfate reduction</td>
<td align="left">S</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/CysH/seed_refpkg/final_outputs">CysH</a></td>
<td align="left">phosophoadenylyl-sulfate reductase [PF01507, EC:1.8.4.8|1.8.4.10]</td>
<td align="left">710</td>
<td align="left">Assimilatory sulfate reduction</td>
<td align="left">S</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/CysI_Sir/seed_refpkg/final_outputs">CysI_Sir</a></td>
<td align="left">(assimilatory) sulfite reductase (NADPH) hemoprotein, beta subunit|ferredoxin dependent</td>
<td align="left">1055</td>
<td align="left">Assimilatory sulfate reduction</td>
<td align="left">S</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/CysJ/seed_refpkg/final_outputs">CysJ</a></td>
<td align="left">sulfite reductase [NADPH] flavoprotein, alpha-component [COG0369, EC:1.8.1.2]</td>
<td align="left">474</td>
<td align="left">Assimilatory sulfate reduction</td>
<td align="left">S</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/CysN/seed_refpkg/final_outputs">CysN</a></td>
<td align="left">Sulfate adenylyltransferase (SAT), large subunit</td>
<td align="left">1221</td>
<td align="left">Assimilatory sulfate reduction</td>
<td align="left">S</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/DsrAB/seed_refpkg/final_outputs">DsrAB</a></td>
<td align="left">Dissimilatory sulfite reductase, alpha and beta subunits</td>
<td align="left">752</td>
<td align="left">Dissimilatory sulfate reduction and oxidation</td>
<td align="left">S</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/DsrC/seed_refpkg/final_outputs">DsrC</a></td>
<td align="left"></td>
<td align="left">341</td>
<td align="left">Dissimilatory sulfate reduction and oxidation</td>
<td align="left">S</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/Sat/seed_refpkg/final_outputs">Sat</a></td>
<td align="left">Sulfate adenylyltransferase [PF01747, EC:2.7.7.4]</td>
<td align="left">254</td>
<td align="left">Dissimilatory sulfate reduction and oxidation</td>
<td align="left">S</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/SoxB/seed_refpkg/final_outputs">SoxB</a></td>
<td align="left">S-sulfosulfanyl-L-cysteine sulfohydrolase [TIGR04486]</td>
<td align="left">1012</td>
<td align="left">SOX system</td>
<td align="left">S</td>
</tr>
<tr class="odd">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/SoxY/seed_refpkg/final_outputs">SoxY</a></td>
<td align="left">Thiosulfate oxidation carrier protein [TIGR04488,ENOG501RH4J]</td>
<td align="left">364</td>
<td align="left">SOX system</td>
<td align="left">S</td>
</tr>
<tr class="even">
<td align="left"><a href="https://github.com/hallamlab/RefPkgs/blob/master/Sulfur_metabolism/SoxZ/seed_refpkg/final_outputs">SoxZ</a></td>
<td align="left">Thiosulfate oxidation carrier complex protein [TIGR04490,ENOG501N097]</td>
<td align="left">247</td>
<td align="left">SOX system</td>
<td align="left">S</td>
</tr>
</tbody>
</table>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="the-saanich-inlet-data-set.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="shell-cheat-sheet.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section",
"scroll_highlight": true
}
});
});
</script>

</body>

</html>
