# Classifying unknown sequences with TreeSAPP and updating a reference package

## Introduction and Goals

We will be using the new particulate methane monooxygenase (pMMO) and ammonia monooxygenase (AMO) alpha subunit, or XmoA, reference package and update it with XmoA amino acid sequences sourced from FunGene. You already created the reference package ([Creating a Reference Package For TreeSAPP]) and downloaded the sequences ([Acquiring golden data]).

There are three goals of this tutorial:

1) Learn how to update a TreeSAPP reference package with `treesapp update`.
1) Learn how to use `treesapp colour` to colour a phylogeny in iTOL.
1) Learn how to annotate features of phylogenetic clades using `treesapp layer`.

The first step involves using `treesapp assign` to classify the FunGene XmoA sequences. The second step is to use `treesapp update` to add the new sequences to the original XmoA reference package.

## Classify new sequences

Before we can begin using TreeSAPP, we must log onto the server and make sure we're in the same directory as all our data.

```{bash eval = FALSE}
cd /data/ts_tutorial/
```

According to `treesapp purity` in the [last chapter](#treesapp-create), and all other sanity checks, the XmoA reference package (contained in your working directory, `/data/ts_tutorial`, at `XmoA_seed/final_outputs/XmoA_build.pkl`) looks functional and only contains XmoA sequences. At this point, you're able to use this reference package to classify sequences in any dataset using `treesapp assign`.

The command `treesapp assign` is perhaps the most popular command and is described in detail on the [TreeSAPP wiki](https://github.com/hallamlab/TreeSAPP/wiki/Classifying-sequences-with-treesapp-assign).

For these commands we will use Block Mapping and Gathering with Entropy (BMGE) which filters out regions of questionable homology from the alignment [@Criscuolo.2010]. Please see the [BMGE publication](https://bmcecolevol.biomedcentral.com/articles/10.1186/1471-2148-10-210) for more information. To trim the multiple sequence alignments prior to phylogenetic placement use the `--trim_align` flag.
To use the PmoA and AmoA reference package we've built instead of any other reference packages that come with TreeSAPP we can use the argument `--refpkg_dir` with the path to a directory containing reference packages we want to use.

As usual, if you're unclear as to what any arguments are, you can use `treesapp assign -h` to get the complete usage and descriptions of each argument.

### Classifying amino acid sequences

We're first going to classify PmoA and AmoA sequences sourced from FunGene. The sequences are from genomes of isolated bacteria and archaea and, because of FunGene's quality annotation pipeline, can likely be trusted. Environmental sequences (i.e. those from metagenomes and amplicon studies) have been excluded as they're more difficult to determine what organism they're from, and so the taxonomic labels are less trustworthy.

```{bash eval = FALSE}
treesapp assign \
  -n 4 \
  -m prot \
  --trim_align \
  --refpkg_dir XmoA_seed/final_outputs/ \
  --fastx_input p_amoA_FunGene9.5_isolates_C65S300_uclust99.faa \
  --output p_amoA_FunGene9.5_isolates_assign/
```

The command should take a few seconds to run and if you see a 'TreeSAPP has finished successfully.' at the end then you're good to move on.

### Classifying ORFs predicted from genomes

The second batch of sequences that we will classify XmoA from are metagenome-assembled genomes (MAGs). MAGs are derived from assembled metagenome datasets (i.e. contigs or scaffolds) and represent the genome of a closely related microbial population. You can think of them as representing a single strain present in a microbial community.

We will need to download two files: the FASTA file containing the genomes and a table with the taxonomic label for each genome.

```
wget https://raw.githubusercontent.com/EDUCE-UBC/MICB425/main/data/SI072_MAGs.fa ./
wget https://raw.githubusercontent.com/EDUCE-UBC/MICB425/main/data/SI072_MAGs_gtdbtk.bac120.summary.tsv ./
```

The first file was created by concatenating twelve different MAGs into a single file and prepending each sequence name (i.e. header) with the MAG name using the software [seqkit](https://bioinf.shenwei.me/seqkit/). Concatenating multiple genomes into a single file saves time since you only need to run `treesapp assign` once instead of classifying each MAG individually. Prepending the header with its respective MAG name is necessary to properly map the names of the query sequences back to their original genome names during the update process. Here is the shell command used (do not run):

```{bash eval = FALSE}
for f in *fa
do
  mag_name=$( basename $f | sed 's/.fa//g' )
  cat $f | seqkit replace -p "^" -r "${mag_name}_" >>SI072_MAGs.fa
done
```

The `treesapp assign` command we use is very similar from those we've used in the past. The main difference is we do not need to include `-m prot` as TreeSAPP assumes the input is comprised of nucleotide sequences by default.

```{bash eval = FALSE}
treesapp assign \
  -n 4 \
  --trim_align \
  --refpkg_dir XmoA_seed/final_outputs/ \
  --fastx_input SI072_MAGs.fa \
  --output SI072_MAGs_assign/
```

This command should take less than a minute to complete.

## Update the reference package with new sequences

The initial XmoA_seed reference package was pretty small with just 36 sequences (information was retrieved using `treesapp package view num_seqs -r XmoA_seed/final_outputs/XmoA_build.pkl`). Also, there were only bacterial sequences included in this version and we know there are archaeal ammonia oxidizers too. Let's see if we can expand on this seed reference package using the sequences from FunGene.

### Updating with publicly available sequences

You have already seen many of these flags and arguments before, and they're used again here just to remind TreeSAPP that we want results fast. A couple of new ones are used, though. `treesapp update` won't take just any old FASTA file but relies on the outputs produced by `treesapp assign`. So we'll guide it to the relevant output directory with `--treesapp_output p_amoA_FunGene9.5_isolates_assign/`. From there, it will figure out what sequences were classified and what their assigned lineages were.

By default TreeSAPP will use the lineage that was assigned to each classified query sequence from the `treesapp assign` outputs. Since these sequences are from FunGene (and originally GenBank) they have NCBI accessions, meaning we can use this information to download their true taxonomic lineages. To turn on that behaviour we use the `--skip_assign` flag.\
Lastly, we must give it the path to the reference package to update with `--refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl`.

```{bash eval = FALSE}
treesapp update \
  --fast \
  --headless \
  --overwrite \
  --delete \
  --cluster \
  --trim_align \
  -n 4 \
  -m prot \
  --fastx_input XmoA_seed.faa \
  --output XmoA_FunGene_update/ \
  --skip_assign \
  --treesapp_output p_amoA_FunGene9.5_isolates_assign/ \
  --refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl
```

It looks like 28 new sequences were introduced into the reference package, bringing the total to 63 sequences! And judging from the taxonomic rank summary printed during runtime both Archaea and Bacteria are now included:

```
Number of unique lineages:
	root       1
	domain     2
	phylum     5
	class      6
	order     10
	family    13
	genus     25
	species   38
```

### Updating with MAG sequences

We are now going to update the reference package that we just updated using the MAG sequences to demonstrate how to iteratively update reference packages. A key difference between using sequences accessioned in the NCBI and here is we must provide the taxonomic labels for each MAG. The MAGs were assigned taxonomic labels using the [Genome Taxonomy Database](https://gtdb.ecogenomic.org/) toolkit (GTDB-tk)[@Chaumeil.2019].

We first need to reformat the table containing the taxonomic labels of the MAGs. We will employ the shell functions `awk` and `sed` to pull out the two columns we need and find-and-replace words and characters.

```{bash, eval = FALSE}
awk -F"\t" '{ OFS="\t"; print $1,$2 }' SI072_MAGs_gtdbtk.bac120.summary.tsv | \
sed 's/user_genome/Organism/g' | \
sed 's/classification/Lineage/g' | \
sed 's/;/; /g' >gtdb_classifications.tsv
```

Now that we have everything we need we can proceed to run `treesapp update` to add the MAG sequences to the reference package.

```{bash eval = FALSE}
treesapp update \
  --fast \
  --headless \
  --overwrite \
  --delete \
  --cluster \
  --trim_align \
  -n 4 \
  -m prot \
  --fastx_input XmoA_seed.faa \
  --output XmoA_MAG_update/ \
  --skip_assign \
  --seqs2lineage gtdb_classifications.tsv \
  --treesapp_output SI072_MAGs_assign/ \
  --refpkg_path XmoA_FunGene_update/final_outputs/XmoA_build.pkl
```

One of the XmoA sequences from the MAGs should have been added to the reference package bringing the total to 64 reference sequences.

A summary of the reference package will have been printed to the screen if `treesapp update` completed properly, and it should look like this:

```
Summary of the updated reference package:
ReferencePackage instance of XmoA (N0102):
	Molecule type:                                      'prot'
	TreeSAPP version:                                   '0.10.1'
	Profile HMM length:                                 '247'
	Substitution model used for phylogenetic inference: 'LG+G4'
	Number of reference sequences (leaf nodes):          64
	Software used to infer phylogeny:                   'FastTree'
	Date of last update:                                '2021-03-22'
	Description:                                        'Alpha subunits of copper membrane monooxygenase enzymes'
```

For supplementary reading on how to integrate sequences from SAGs and MAGs into reference packages please visit the [TreeSAPP Wiki](https://github.com/hallamlab/TreeSAPP/wiki/Integrating-MAGs-and-SAGs-in-reference-packages).

Just to finish this off, let's map the PmoA and AmoA sequences from FunGene onto the MAG-updated reference package with `treesapp assign` to view later.

```{bash eval = FALSE}
treesapp assign \
  -n 4 \
  -m prot \
  --trim_align \
  --refpkg_dir XmoA_MAG_update/final_outputs/ \
  --fastx_input p_amoA_FunGene9.5_isolates_C65S300_uclust99.faa \
  --output p_amoA_FunGene9.5_isolates_update_assign/
```

## Annotate clades

Protein families vary in their evolutionary complexity; many, especially genes involved in scavenging and metabolizing nutrients, have often been involved in some gene duplication and/or lateral gene transfer event while others have remained anchored to their host. Moreover, closely related enzymes can vary in their activities and it is often valuable, though not necessary, to annotate such features.

Creating your own annotations involves scouring the literature to identify the phenotypes associated with each organism or taxon. For example, you would need to determine what paralogs or substrates are used by each organism in the reference package. Often though, these phenotypes are conserved across larger taxonomic groups, such as whole genera, families, orders, etc. 
If you are not familiar with the evolution of a protein family it may be helpful to find reference sequences for the activity, function, or other feature you're interested in and place them on the tree with `treesapp assign`. Then, guided by these sequences, it should be easier to annotate the clades.
Seeing as a literature review can't be accomplished within the time span of this tutorial, you can use the annotations already created for the XmoA reference package.

```{bash eval = FALSE}
wget https://raw.githubusercontent.com/hallamlab/RefPkgs/master/Nitrogen_metabolism/Nitrification/XmoA/XmoA_taxonomy-phenotype_map.tsv
```

You can view the contents of this file with the Unix command `less`:

```{bash eval = FALSE}
less XmoA_taxonomy-phenotype_map.tsv
```

```
d__Archaea	AmoA_AOA
Methylococcaceae	EmoA
g__Haliea	EmoA
f__Nitrosomonadaceae	AmoA_AOB
o__Chromatiales	AmoA_AOB
g__Methylobacter	PmoA
g__Methylacidiphilum	PmoA
```

To create a file compatible with iTOL you will need to convert the taxonomy-phenotype mapping file with `treesapp colour`. 

```{bash eval = FALSE}
treesapp colour \
  -r XmoA_MAG_update/final_outputs/XmoA_build.pkl \
  -o XmoA_colours/ \
  -n Function \
  --taxa_map XmoA_taxonomy-phenotype_map.tsv
```

The two output files---`XmoA_Function_colours_style.txt` and `XmoA_Function_colour_strip.txt`---can be used to bring colour to the JPlace file in iTOL and also be used for adding phenotype-level annotations to the classification table with `treesapp layer`. Let's make copies of the two text files in  XmoA_colours/ into the AmoA_assign/iTOL output directory and transfer them to `<path to Xmoa_sequences>`.

```{bash eval = FALSE}
cp -r XmoA_colours p_amoA_FunGene9.5_isolates_update_assign/iTOL_output/
```
    
 -  **On your computer:** Transfer `XmoA_seed_assign/iTOL_output` from the server to your local directory `Xmoa_sequences`.

```{bash eval = FALSE}
scp -r root@<server address>:/data/ts_tutorial/p_amoA_FunGene9.5_isolates_update_assign/iTOL_output <path to Xmoa_sequences>
```

-   Navigate to <https://itol.embl.de/> using a web browser and sign in using the group's username and password.

-   Upload the file `XmoA_complete_profile.jplace`. You can do this using either your computer's file browser then navigating to the `iTOL_output` directory and clicking-and-dragging the file, or iTOL's file uploader---the "Upload tree files" button. 

-   Next, navigate to the page displaying the phylogeny and click-and-drag the file `XmoA_labels.txt` from your file browser into the iTOL window. This should convert the TreeSAPP identifiers (e.g. 1_XmoA) to more useful descriptions with the organism name and accession for each leaf node.

-   The two files `XmoA_Function_colours_style.txt` and `XmoA_Function_colour_strip.txt`  can be dragged into the iTOL browser window to bring colour to the XmoA tree. The figure should look identical to FigureÂ \@ref(fig:XmoA-Classified)

![(\#fig:XmoA-Classified)XmoA phylogeny after functional annotation](images/XmoA_update_coloured.png )

-   Finally, turn on the "Phylogenetic Placements" dataset (right of the screen).
 

## Reclassify and layer annotations onto classification table

With the clade colours file in the tutorial directory, it is now simple to assign each of the classified query sequences to its respective functional guild with `treesapp layer`.

```{bash eval = FALSE}
treesapp layer \
  --colours_style XmoA_colours/XmoA_Function_colours_style.txt \
  --refpkg_dir XmoA_MAG_update/final_outputs/ \
  --treesapp_output p_amoA_FunGene9.5_isolates_update_assign/
```

The output of this command is `p_amoA_FunGene9.5_isolates_update_assign/final_outputs/extra_annotated_marker_contig_map.tsv`. It contains an extra column for the functional annotations but is otherwise identical to the original `marker_contig_map.tsv` file. You can view the first 10 lines by using the Unix command `head`:

```{bash eval = FALSE}
head p_amoA_FunGene9.5_isolates_update_assign/final_outputs/extra_annotated_marker_contig_map.tsv
```

## Use your group's reference package to classify unkown sequences then update your package

Continue to work with your group's reference package in the directory `/data/<gene name>`. Visualize the appropriate outputs with iTOL.
