# Creating a Reference Package For TreeSAPP {#treesapp-create}

## Introduction and Goals

This tutorial is meant for users who would like to analyze a gene family lacking an existing TreeSAPP reference package (refpkg). It is also useful to those who would rather start over than use `treesapp update` to improve an existing refpkg. This tutorial will cover the following subcommand of `treesapp`: `create`, `package`, `purity` and `assign`. TreeSAPP will automatically determine the taxonomic lineage assigned to each sequence if these are NCBI accessions. For ways to provide lineage information for as-of-yet unaccessioned sequences please refer to the TreeSAPP Wiki page [Building reference packages with TreeSAPP](https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#alternative-sources). TreeSAPP does not currenty support GTDB taxonomies although efforts are being made to determine feasibility of this approach. 

## Making the reference package

When building reference packages if you can source manually curated sequences---such as those hosted by SwissProt---always use those. There are plenty of sources for curated PmoA & AmoA sequences. More details on choosing databases for retrieving sequences can be found on the [TreeSAPP wiki](https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#step-1-reference-sequences).

In the [Aquiring golden data] tutorial, PmoA & AmoA sequences were downloaded from [TIGRFAM](http://tigrfams.jcvi.org/cgi-bin/index.cgi), [EggNOG](http://eggnog5.embl.de/#/app/home), and [FunGene](http://fungene.cme.msu.edu/). For now we'll just use the TIGRFAM seed sequences and EggNOG sequences as these are best curated. Let's call this our seed reference package. The steps involved are:

-   **All steps are executed on the server unless stated oterwise!!** Connect to your personal server and continue to work in the `/data/ts_tutorial` directory.

    ```{bash eval = FALSE}
    cd /data/ts_tutorial
    ```

-   **Optional:** If you could not download the reference sequences of XmoA yourself (see [Acquiring golden data]), download [`xmoa_file_list.txt`](https://github.com/hallamlab/TreeSAPP/blob/master/docs/xmoa_file_list.txt)---a list of the necessary fasta files---with `wget` to the `ts_tutorial` directory, then run `wget` again to retrieve each file within this list.

    ```{bash eval = FALSE}
    cd /data/ts_tutorial
    wget https://raw.githubusercontent.com/hallamlab/TreeSAPP/master/docs/xmoa_file_list.txt
    wget -i xmoa_file_list.txt
    ```

-   Create the input FASTA file by combining the EggNOG and TIGRFAM reference into a single file using `cat`. `cat` stands for concatentate, i.e. the files are pasted together. The output of `cat` is funneled with the `>` symbol to a `<new file>`, i.e.  `cat <file 1> <file 2> > <new file>` 

    ```{bash eval = FALSE}
    cat ENOG5028JPK_EggNOGv5.faa TIGR03080.faa arCOG08676_EggNOGv5.faa > XmoA_seed.faa
    ```

    There are 76 well curated PmoA and AmoA fasta sequences in this file.

-   Let's look briefly at the newly generated fasta file.

    ```{bash eval = FALSE}
    less XmoA_seed.faa
    ```

-   You will notice, that they are lines that start with a `>` that gives the name of a sequence followed by a line containing the actual nucleotide sequence itself. We can count and confirm the number of sequences in the fasta file by counting those `>` using `grep`. `grep` is an incredibly powerful tool to find matching text patterns (using a format called "regular expression") and is beyond the scope of this tutorial. The code below searches for a `>` at the beginning of the line (defined with `"^>"`) of the file `XmoA_seed.faa` and counts each occurrence due to the option `-c`.

    ```{bash eval = FALSE}
    grep -c "^>" XmoA_seed.faa
    ```

-   Use submodule `treesapp create` to build the XmoA reference package with different arguments (see [visualization of the `create` workflow](https://github.com/hallamlab/TreeSAPP/wiki/Building-reference-packages-with-TreeSAPP#ingredients)). To speed things along, we will use FastTree to infer the phylogeny and will skip bootstrapping with `--fast`. To remove potential redundant sequences, we will cluster the candidate reference sequences at 97% similarity using the argument `-p`. When clustering the candidate sequences TreeSAPP would normally ask which sequence to use for the representative of the cluster. This can be handy in cases when some sequences are better annotated and/or are especially important. To speed things up even more, the flag `--headless` will prevent these requests. This command will take \~2 minutes to complete.

    ```{bash eval = FALSE}
    treesapp create \
      --fast \
      --headless \
      --overwrite \
      --cluster \
      --trim_align \
      -n 4 \
      -m prot \
      -p 0.97 \
      --fastx_input XmoA_seed.faa \
      -c XmoA \
      --output XmoA_seed
    ```

     The final reference package file is located in `XmoA_seed/final_outputs/XmoA_build.pkl`. This file contains all the individual components of a reference package (multiple sequence alignment, profile HMM, phylogenetic tree, taxonomic lineages) as well as some other data. These files were bundled up using the joblib Python library. They can be accessed individually using the submodule `treesapp package`.

-   Replace the current refpkg_code `Z1111` with unique identifier `N0102`.

    ```{bash eval = FALSE}
    treesapp package edit \
      refpkg_code N0102 \
      --overwrite \
      --refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl
    ```

-   We can also modify the reference package's description while we're here.

    ```{bash eval = FALSE}
    treesapp package edit \
      description "Alpha subunits of copper membrane monooxygenase enzymes" \
      --overwrite \
      --refpkg_path XmoA_seed/final_outputs/XmoA_build.pkl
    ```

## Testing the purity of the reference package

We will determine whether there were any mis-annotated sequences that were included in our reference package with `treesapp purity`. To do this, we take a well-curated database and attempt to classify sequences in it using `treesapp assign`. The results are then analysed and displayed for the user to evaluate. The TIGRFAM database is fairly comprehensive, representing 4488 different groups in version 15, and by using just the TIGRFAM seed sequences (`TIGRFAM_seed_named.faa`) we can be fairly sure we won't be evaluating our classifications with mis-annotated sequences.

```{bash eval = FALSE}
treesapp purity \
  --trim_align \
  -m prot \
  -n 4 \
  -r XmoA_seed/final_outputs/XmoA_build.pkl \
  --extra_info TIGRFAM_info.tsv \
  -i TIGRFAM_seed_named.faa \
  --output XmoA_purity
```

The important bit of the output should look like this

```
Summarizing assignments for reference package XmoA
Ortholog        Hits    Leaves  Tree-coverage   Description
--------------------------------------------------------------------------------
TIGR03080       3       4       11.1    methane monooxygenase/ammonia monooxygenase, subunit A
```

From this summary it appears that the reference package classified three homologous sequences that were placed at leaf nodes in the tree (i.e. they're closely related) and the sequences were all from the family "TIGR03080", also known as "methane monooxygenase/ammonia monooxygenase, subunit A". In all, we are probably good to proceed!

## Looking at tree of reference package with iTOL

We have now created a phyogenetic tree for XmoA, but if we would visualize it, we would only see TreeSAPP identifiers (such as 1_XmoA, 2_XmoA) as labels on the tree which by themselves are not meaningful. In order to assign leaf labels to the refpkg iTOL tree we will use `treesapp assign` function.  

-   Assign reference sequences to reference package to generate tree with labels.

    ```{bash eval = FALSE}
    treesapp assign \
      -n 4 \
      -m prot \
      --trim_align \  
      --refpkg_dir XmoA_seed/final_outputs/ \ 
      --fastx_input XmoA_seed.faa \
      --output XmoA_seed_assign/
    ```

    The `treesapp assign` output directory (defined by the flag `--output`, i.e. in this case `XmoA_seed_assign`) contains the `iTOL_output` directory.
    
 -  **On your computer:** Transfer `XmoA_seed_assign/iTOL_output` from the server to your local directory `Xmoa_sequences`.
 
    ```{bash eval = FALSE}
     scp -r root@<server address>:/data/ts_tutorial/XmoA_seed_assign/iTOL_output <path to Xmoa_sequences>
    ```

    We can quickly and easily view the sequences placed on the phylogeny using iTOL. We will do this by uploading the JPlace file produced by TreeSAPP. This JPlace file contains both the phylogeny and the location of the placements that will be visualized.

-   Navigate to <https://itol.embl.de/> using a web browser and sign in using the group's username and password.

-   Upload the file `XmoA_complete_profile.jplace`. You can do this using either your computer's file browser then navigating to the `iTOL_output` directory and clicking-and-dragging the file, or iTOL's file uploader---the "Upload tree files" button. 

-   Next, navigate to the page displaying the phylogeny and click-and-drag the file `XmoA_labels.txt` from your file browser into the iTOL window. This should convert the TreeSAPP identifiers (e.g. 1_XmoA) to more useful descriptions with the organism name and accession for each leaf node.

-   Finally, turn on the "Phylogenetic Placements" dataset (right of the screen) and the figure should look identical to this figure below FigureÂ \@ref(fig:xmoa).

    ![(\#fig:xmoa) Phylogenetic tree of XmoA_seed refpkg. The red bubbles represent sequence placements with the size of the bubble indicating the number of placements on a branch. 76 sequences were classified and placed on 36 leaf nodes. ](images/XmoA_seed_iTOLtree.svg)

## Building your group's reference package

Now create a reference package for the gene that was assigned to your group using the **group's** server.

1) **On your computer:** create a directory `<gene_name_sequences>` in your home directory and download any reference sequences for your gene you can find on the databases.

1) **On the server:** In the directory `/data`, create a new directory `<gene name>`.

1) **On your computer:** Copy the reference sequences from your computer to the directory `/data/<gene name>` on the server.

1) **On the server:** Work in `/data/<gene name>` as you are creating your TreeSAPP outputs.

1) **On your computer:** Whenever you have TreeSAPP outputs that can be visualized with iTOL, copy the files from the server to your computer to do so.
